{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Porting genome scale metabolic models for metabolomics\n",
    "\n",
    "- to make formats compatible to mummichog\n",
    "- to link to a common compound table \n",
    "- from compound table, we generated predicted mass peaks based on formula\n",
    "\n",
    "As mummichog 3 is under development, treat this as part of development.\n",
    "\n",
    "**Use cobra to parse SBML models whereas applicable**\n",
    "\n",
    "Not all models comply with the formats in cobra. Models from USCD and Thiele labs should comply.\n",
    "\n",
    "**Base our code on metDataModel**\n",
    "\n",
    "Each model needs a list of Reactions, list of Pathways, and a list of Compounds.\n",
    "It's important to include with Compounds with all linked identifiers to other DBs (HMDB, PubChem, etc), and with formulae (usually charged form in these models) when available.\n",
    "We can alwasy update the data later. E.g. the neural formulae can be retrieved from HMDB if linked.\n",
    "Save in Python pickle and in JSON.\n",
    "\n",
    "\n",
    "Shuzhao Li, 2021-05-09, 05-10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting metDataModel\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/4a/e5/e4625fe421c74695a286bc1a289026ef1c1dff464e078c17fd9eed9167dd/metDataModel-0.3.0-py3-none-any.whl (2.1MB)\n",
      "\u001b[K     |████████████████████████████████| 2.2MB 926kB/s eta 0:00:01\n",
      "\u001b[?25hInstalling collected packages: metDataModel\n",
      "Successfully installed metDataModel-0.3.0\n"
     ]
    }
   ],
   "source": [
    "# https://github.com/shuzhao-li/metDataModel/\n",
    "!pip install metDataModel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "from metDataModel.core import Compound, Reaction, Pathway, metabolicModel\n",
    "\n",
    "# tool to parse XML - not needed if using cobra\n",
    "import xml.etree.ElementTree as ET\n",
    "# tool to parse Excel xlsx\n",
    "import xlrd\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting cobra\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/c5/04/f785d34e11b42b21c101130d60324c90f58a612f49a6f95e1747f0f05ca1/cobra-0.22.0-py2.py3-none-any.whl (2.4MB)\n",
      "\u001b[K     |████████████████████████████████| 2.4MB 2.1MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting optlang~=1.5 (from cobra)\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/12/3e/9d0b72cf5a8ff660e5787a0797906e04942081f3ad4a95f860488affff2b/optlang-1.5.2-py2.py3-none-any.whl (147kB)\n",
      "\u001b[K     |████████████████████████████████| 153kB 2.5MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting rich>=8.0 (from cobra)\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/1a/da/2a1f064dc620ab47f3f826ae085384084b71ea05c8c21d67f1dfc29189ab/rich-10.1.0-py3-none-any.whl (201kB)\n",
      "\u001b[K     |████████████████████████████████| 204kB 3.0MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting future (from cobra)\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/45/0b/38b06fd9b92dc2b68d58b75f900e97884c45bedd2ff83203d933cf5851c9/future-0.18.2.tar.gz (829kB)\n",
      "\u001b[K     |████████████████████████████████| 829kB 3.1MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: numpy~=1.13 in /opt/conda/lib/python3.7/site-packages (from cobra) (1.17.2)\n",
      "Collecting httpx~=0.14 (from cobra)\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/04/69/17b725b1bfe63228369cddf50206381eadb14bc46b933a864ae93ea1b9bf/httpx-0.18.1-py3-none-any.whl (75kB)\n",
      "\u001b[K     |████████████████████████████████| 81kB 2.0MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting pydantic~=1.6 (from cobra)\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/8f/c4/a83f0b745d824a85ee5e4c8d26329d1c687b3a9303eca762dcc373721734/pydantic-1.8.1-py3-none-any.whl (125kB)\n",
      "\u001b[K     |████████████████████████████████| 133kB 4.6MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting appdirs~=1.4 (from cobra)\n",
      "  Downloading https://files.pythonhosted.org/packages/3b/00/2344469e2084fb287c2e0b57b72910309874c3245463acd6cf5e3db69324/appdirs-1.4.4-py2.py3-none-any.whl\n",
      "Collecting swiglpk (from cobra)\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/80/48/ff3ce61567f667629268b272d4c57f980ff0a8d4bdf991a593be12384186/swiglpk-5.0.3-cp37-cp37m-manylinux2010_x86_64.whl (2.1MB)\n",
      "\u001b[K     |████████████████████████████████| 2.1MB 4.2MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting pandas~=1.0 (from cobra)\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/51/51/48f3fc47c4e2144da2806dfb6629c4dd1fa3d5a143f9652b141e979a8ca9/pandas-1.2.4-cp37-cp37m-manylinux1_x86_64.whl (9.9MB)\n",
      "\u001b[K     |████████████████████████████████| 9.9MB 51.0MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting diskcache~=5.0 (from cobra)\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/6a/5e/3deb8f9c83bead2af6f2cda97c4400516488464fede2853875a81e502953/diskcache-5.2.1-py3-none-any.whl (44kB)\n",
      "\u001b[K     |████████████████████████████████| 51kB 1.6MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting depinfo (from cobra)\n",
      "  Downloading https://files.pythonhosted.org/packages/af/8b/cee6dca4c4708705444c9cad9e783b9212cc51cab8a5e05ccfe930f53058/depinfo-1.7.0-py2.py3-none-any.whl\n",
      "Collecting importlib-resources (from cobra)\n",
      "  Downloading https://files.pythonhosted.org/packages/f0/5e/69e6a0602c1f18d390952177de648468c4a380252858b0022affc3ce7811/importlib_resources-5.1.2-py3-none-any.whl\n",
      "Collecting ruamel.yaml~=0.16 (from cobra)\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/29/4e/c3105bbbbc662f6a671a505f00ec771e93b5254f09fbb06002af9087071a/ruamel.yaml-0.17.4-py3-none-any.whl (101kB)\n",
      "\u001b[K     |████████████████████████████████| 102kB 3.5MB/s ta 0:00:011\n",
      "\u001b[?25hCollecting python-libsbml==5.19.0 (from cobra)\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/93/13/2df22b77ee9125381a00aa4dc63de1776bb84d773655d2770416e81416cd/python_libsbml-5.19.0-cp37-cp37m-manylinux1_x86_64.whl (6.2MB)\n",
      "\u001b[K     |████████████████████████████████| 6.2MB 6.3MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: six>=1.9 in /opt/conda/lib/python3.7/site-packages (from optlang~=1.5->cobra) (1.12.0)\n",
      "Requirement already satisfied: sympy>=1.0 in /opt/conda/lib/python3.7/site-packages (from optlang~=1.5->cobra) (1.4)\n",
      "Collecting colorama<0.5.0,>=0.4.0 (from rich>=8.0->cobra)\n",
      "  Downloading https://files.pythonhosted.org/packages/44/98/5b86278fbbf250d239ae0ecb724f8572af1c91f4a11edf4d36a206189440/colorama-0.4.4-py2.py3-none-any.whl\n",
      "Collecting pygments<3.0.0,>=2.6.0 (from rich>=8.0->cobra)\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/a6/c9/be11fce9810793676017f79ffab3c6cb18575844a6c7b8d4ed92f95de604/Pygments-2.9.0-py3-none-any.whl (1.0MB)\n",
      "\u001b[K     |████████████████████████████████| 1.0MB 13.9MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting typing-extensions<4.0.0,>=3.7.4 (from rich>=8.0->cobra)\n",
      "  Downloading https://files.pythonhosted.org/packages/2e/35/6c4fff5ab443b57116cb1aad46421fb719bed2825664e8fe77d66d99bcbc/typing_extensions-3.10.0.0-py3-none-any.whl\n",
      "Collecting commonmark<0.10.0,>=0.9.0 (from rich>=8.0->cobra)\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/b1/92/dfd892312d822f36c55366118b95d914e5f16de11044a27cf10a7d71bbbf/commonmark-0.9.1-py2.py3-none-any.whl (51kB)\n",
      "\u001b[K     |████████████████████████████████| 51kB 2.6MB/s  eta 0:00:01\n",
      "\u001b[?25hCollecting httpcore<0.14.0,>=0.13.0 (from httpx~=0.14->cobra)\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/7a/89/29a26a8086268a8043b1fd0228a463bab99990d77fdfaa8a2bd07117851a/httpcore-0.13.3-py3-none-any.whl (57kB)\n",
      "\u001b[K     |████████████████████████████████| 61kB 3.3MB/s eta 0:00:011\n",
      "\u001b[?25hCollecting rfc3986[idna2008]<2,>=1.3 (from httpx~=0.14->cobra)\n",
      "  Downloading https://files.pythonhosted.org/packages/c4/e5/63ca2c4edf4e00657584608bee1001302bbf8c5f569340b78304f2f446cb/rfc3986-1.5.0-py2.py3-none-any.whl\n",
      "Collecting sniffio (from httpx~=0.14->cobra)\n",
      "  Downloading https://files.pythonhosted.org/packages/52/b0/7b2e028b63d092804b6794595871f936aafa5e9322dcaaad50ebf67445b3/sniffio-1.2.0-py3-none-any.whl\n",
      "Requirement already satisfied: certifi in /opt/conda/lib/python3.7/site-packages (from httpx~=0.14->cobra) (2019.6.16)\n",
      "Requirement already satisfied: python-dateutil>=2.7.3 in /opt/conda/lib/python3.7/site-packages (from pandas~=1.0->cobra) (2.8.0)\n",
      "Requirement already satisfied: pytz>=2017.3 in /opt/conda/lib/python3.7/site-packages (from pandas~=1.0->cobra) (2019.2)\n",
      "Collecting importlib-metadata; python_version < \"3.8\" (from depinfo->cobra)\n",
      "  Downloading https://files.pythonhosted.org/packages/8e/e2/49966924c93909d47612bb47d911448140a2f6c1390aec2f4c1afbe3748f/importlib_metadata-4.0.1-py3-none-any.whl\n",
      "Collecting zipp>=0.4; python_version < \"3.8\" (from importlib-resources->cobra)\n",
      "  Downloading https://files.pythonhosted.org/packages/0f/8c/715c54e9e34c0c4820f616a913a7de3337d0cd79074dd1bed4dd840f16ae/zipp-3.4.1-py3-none-any.whl\n",
      "Collecting ruamel.yaml.clib>=0.1.2; platform_python_implementation == \"CPython\" and python_version < \"3.10\" (from ruamel.yaml~=0.16->cobra)\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/5e/6e/f652c56bbb2c3d3fca252ffc7c0358597f57a1bbdf484dac683054950c63/ruamel.yaml.clib-0.2.2-cp37-cp37m-manylinux1_x86_64.whl (547kB)\n",
      "\u001b[K     |████████████████████████████████| 552kB 11.9MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: mpmath>=0.19 in /opt/conda/lib/python3.7/site-packages (from sympy>=1.0->optlang~=1.5->cobra) (1.1.0)\n",
      "Collecting h11<0.13,>=0.11 (from httpcore<0.14.0,>=0.13.0->httpx~=0.14->cobra)\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/60/0f/7a0eeea938eaf61074f29fed9717f2010e8d0e0905d36b38d3275a1e4622/h11-0.12.0-py3-none-any.whl (54kB)\n",
      "\u001b[K     |████████████████████████████████| 61kB 2.2MB/s eta 0:00:011\n",
      "\u001b[?25hRequirement already satisfied: idna; extra == \"idna2008\" in /opt/conda/lib/python3.7/site-packages (from rfc3986[idna2008]<2,>=1.3->httpx~=0.14->cobra) (2.8)\n",
      "Building wheels for collected packages: future\n",
      "  Building wheel for future (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for future: filename=future-0.18.2-cp37-none-any.whl size=491057 sha256=5fa480e3d4a477449052cf4dd6c809a9167a799cc87bec30c93102be439d0e85\n",
      "  Stored in directory: /home/jovyan/.cache/pip/wheels/8b/99/a0/81daf51dcd359a9377b110a8a886b3895921802d2fc1b2397e\n",
      "Successfully built future\n",
      "Installing collected packages: swiglpk, optlang, colorama, pygments, typing-extensions, commonmark, rich, future, sniffio, h11, httpcore, rfc3986, httpx, pydantic, appdirs, pandas, diskcache, zipp, importlib-metadata, depinfo, importlib-resources, ruamel.yaml.clib, ruamel.yaml, python-libsbml, cobra\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Found existing installation: Pygments 2.4.2\n",
      "    Uninstalling Pygments-2.4.2:\n",
      "      Successfully uninstalled Pygments-2.4.2\n",
      "  Found existing installation: pandas 0.25.1\n",
      "    Uninstalling pandas-0.25.1:\n",
      "      Successfully uninstalled pandas-0.25.1\n",
      "Successfully installed appdirs-1.4.4 cobra-0.22.0 colorama-0.4.4 commonmark-0.9.1 depinfo-1.7.0 diskcache-5.2.1 future-0.18.2 h11-0.12.0 httpcore-0.13.3 httpx-0.18.1 importlib-metadata-4.0.1 importlib-resources-5.1.2 optlang-1.5.2 pandas-1.2.4 pydantic-1.8.1 pygments-2.9.0 python-libsbml-5.19.0 rfc3986-1.5.0 rich-10.1.0 ruamel.yaml-0.17.4 ruamel.yaml.clib-0.2.2 sniffio-1.2.0 swiglpk-5.0.3 typing-extensions-3.10.0.0 zipp-3.4.1\n"
     ]
    }
   ],
   "source": [
    "! pip install cobra"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cobra\n",
    "\n",
    "# https://cobrapy.readthedocs.io/en/latest/io.html#SBML"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>COBRAModel</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07fbf4543dc88</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>8399</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>13543</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>111</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass_reaction - 1.0*biomass_reaction_reverse_32a6c</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>Cytoplasm, Lysosome, Mitochondrion, Endoplasmic_reticulum, Extracellular, Peroxisome, Nucleus, Golgi, unknownCompartment4</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model COBRAModel at 0x7fbf4543dc88>"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# cloned from\n",
    "# https://github.com/VirtualMetabolicHuman\n",
    "# 2021-05-08\n",
    "# this is the more inclusive model. The other Recon3DModel_301 is flux constrainted.\n",
    "R3D = \"thiele/Recon/Current_Version/Recon3D_301_Reconstruction/Recon3D_301.xml\"\n",
    "\n",
    "model = cobra.io.read_sbml_model(R3D)\n",
    "\n",
    "model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['__add__',\n",
       " '__class__',\n",
       " '__delattr__',\n",
       " '__dict__',\n",
       " '__dir__',\n",
       " '__doc__',\n",
       " '__enter__',\n",
       " '__eq__',\n",
       " '__exit__',\n",
       " '__format__',\n",
       " '__ge__',\n",
       " '__getattribute__',\n",
       " '__getstate__',\n",
       " '__gt__',\n",
       " '__hash__',\n",
       " '__iadd__',\n",
       " '__init__',\n",
       " '__init_subclass__',\n",
       " '__le__',\n",
       " '__lt__',\n",
       " '__module__',\n",
       " '__ne__',\n",
       " '__new__',\n",
       " '__reduce__',\n",
       " '__reduce_ex__',\n",
       " '__repr__',\n",
       " '__setattr__',\n",
       " '__setstate__',\n",
       " '__sizeof__',\n",
       " '__str__',\n",
       " '__subclasshook__',\n",
       " '__weakref__',\n",
       " '_annotation',\n",
       " '_compartments',\n",
       " '_contexts',\n",
       " '_id',\n",
       " '_populate_solver',\n",
       " '_repr_html_',\n",
       " '_sbml',\n",
       " '_set_id_with_model',\n",
       " '_solver',\n",
       " '_tolerance',\n",
       " '_trimmed',\n",
       " '_trimmed_genes',\n",
       " '_trimmed_reactions',\n",
       " 'add_boundary',\n",
       " 'add_cons_vars',\n",
       " 'add_groups',\n",
       " 'add_metabolites',\n",
       " 'add_reaction',\n",
       " 'add_reactions',\n",
       " 'annotation',\n",
       " 'boundary',\n",
       " 'compartments',\n",
       " 'constraints',\n",
       " 'copy',\n",
       " 'demands',\n",
       " 'description',\n",
       " 'exchanges',\n",
       " 'genes',\n",
       " 'get_associated_groups',\n",
       " 'get_metabolite_compartments',\n",
       " 'groups',\n",
       " 'id',\n",
       " 'medium',\n",
       " 'merge',\n",
       " 'metabolites',\n",
       " 'name',\n",
       " 'notes',\n",
       " 'objective',\n",
       " 'objective_direction',\n",
       " 'optimize',\n",
       " 'problem',\n",
       " 'reactions',\n",
       " 'remove_cons_vars',\n",
       " 'remove_groups',\n",
       " 'remove_metabolites',\n",
       " 'remove_reactions',\n",
       " 'repair',\n",
       " 'sinks',\n",
       " 'slim_optimize',\n",
       " 'solver',\n",
       " 'summary',\n",
       " 'tolerance',\n",
       " 'variables']"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dir(model)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'hmdb': 'HMDB06225',\n",
       " 'inchi': 'InChI=1S/C28H44O3/c1-18(9-10-19(2)27(4,5)31)24-13-14-25-21(8-7-15-28(24,25)6)11-12-22-16-23(29)17-26(30)20(22)3/h9-12,18-19,23-26,29-31H,3,7-8,13-17H2,1-2,4-6H3/b10-9+,21-11+,22-12-/t18-,19+,23-,24-,25+,26+,28-/m1/s1',\n",
       " 'pubchem.compound': '9547243'}"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.metabolites[33].annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['__class__',\n",
       " '__delattr__',\n",
       " '__dict__',\n",
       " '__dir__',\n",
       " '__doc__',\n",
       " '__eq__',\n",
       " '__format__',\n",
       " '__ge__',\n",
       " '__getattribute__',\n",
       " '__getstate__',\n",
       " '__gt__',\n",
       " '__hash__',\n",
       " '__init__',\n",
       " '__init_subclass__',\n",
       " '__le__',\n",
       " '__lt__',\n",
       " '__module__',\n",
       " '__ne__',\n",
       " '__new__',\n",
       " '__reduce__',\n",
       " '__reduce_ex__',\n",
       " '__repr__',\n",
       " '__setattr__',\n",
       " '__sizeof__',\n",
       " '__str__',\n",
       " '__subclasshook__',\n",
       " '__weakref__',\n",
       " '_annotation',\n",
       " '_bound',\n",
       " '_id',\n",
       " '_model',\n",
       " '_reaction',\n",
       " '_repr_html_',\n",
       " '_set_id_with_model',\n",
       " 'annotation',\n",
       " 'charge',\n",
       " 'compartment',\n",
       " 'constraint',\n",
       " 'copy',\n",
       " 'elements',\n",
       " 'formula',\n",
       " 'formula_weight',\n",
       " 'id',\n",
       " 'model',\n",
       " 'name',\n",
       " 'notes',\n",
       " 'reactions',\n",
       " 'remove_from_model',\n",
       " 'shadow_price',\n",
       " 'summary',\n",
       " 'y']"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dir(model.metabolites[33])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['C28H44O3',\n",
       " 0,\n",
       " '1-Alpha,25-Dihydroxyvitamin D2',\n",
       " '1a25dhvitd2[m]',\n",
       " '1a25dhvitd2[m]',\n",
       " {'hmdb': 'HMDB06225',\n",
       "  'inchi': 'InChI=1S/C28H44O3/c1-18(9-10-19(2)27(4,5)31)24-13-14-25-21(8-7-15-28(24,25)6)11-12-22-16-23(29)17-26(30)20(22)3/h9-12,18-19,23-26,29-31H,3,7-8,13-17H2,1-2,4-6H3/b10-9+,21-11+,22-12-/t18-,19+,23-,24-,25+,26+,28-/m1/s1',\n",
       "  'pubchem.compound': '9547243'}]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[model.metabolites[33].formula,\n",
    "model.metabolites[33].charge,\n",
    " model.metabolites[33].name,\n",
    " model.metabolites[33].id,\n",
    " model.metabolites[33]._id,\n",
    " model.metabolites[33].annotation\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13543"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len( model.reactions )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>25VITD2Hm</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>1-Alpha-Vitamin D-25-Hydroxylase (D2)</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07fbf62abc4e0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>25hvitd2[m] + h[m] + nadph[m] + o2[m] --> 1a25dhvitd2[m] + h2o[m] + nadp[m]</p>\n",
       "                    <p style='text-align:right'>25-Hydroxyvitamin D2 + Proton + Nicotinamide Adenine Dinucleotide Phosphate - Reduced + Oxygen --> 1-Alpha,25-Dihydroxyvitamin D2 + Water + Nicotinamide Adenine Dinucleotide Phosphate</p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td>1594.1</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>0.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction 25VITD2Hm at 0x7fbf62abc4e0>"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.reactions[33]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['__add__',\n",
       " '__class__',\n",
       " '__copy__',\n",
       " '__deepcopy__',\n",
       " '__delattr__',\n",
       " '__dict__',\n",
       " '__dir__',\n",
       " '__doc__',\n",
       " '__eq__',\n",
       " '__format__',\n",
       " '__ge__',\n",
       " '__getattribute__',\n",
       " '__getstate__',\n",
       " '__gt__',\n",
       " '__hash__',\n",
       " '__iadd__',\n",
       " '__imul__',\n",
       " '__init__',\n",
       " '__init_subclass__',\n",
       " '__isub__',\n",
       " '__le__',\n",
       " '__lt__',\n",
       " '__module__',\n",
       " '__mul__',\n",
       " '__ne__',\n",
       " '__new__',\n",
       " '__radd__',\n",
       " '__reduce__',\n",
       " '__reduce_ex__',\n",
       " '__repr__',\n",
       " '__setattr__',\n",
       " '__setstate__',\n",
       " '__sizeof__',\n",
       " '__str__',\n",
       " '__sub__',\n",
       " '__subclasshook__',\n",
       " '__weakref__',\n",
       " '_annotation',\n",
       " '_associate_gene',\n",
       " '_check_bounds',\n",
       " '_dissociate_gene',\n",
       " '_gene_reaction_rule',\n",
       " '_genes',\n",
       " '_id',\n",
       " '_lower_bound',\n",
       " '_metabolites',\n",
       " '_model',\n",
       " '_repr_html_',\n",
       " '_set_id_with_model',\n",
       " '_update_awareness',\n",
       " '_upper_bound',\n",
       " 'add_metabolites',\n",
       " 'annotation',\n",
       " 'boundary',\n",
       " 'bounds',\n",
       " 'build_reaction_from_string',\n",
       " 'build_reaction_string',\n",
       " 'check_mass_balance',\n",
       " 'compartments',\n",
       " 'copy',\n",
       " 'delete',\n",
       " 'flux',\n",
       " 'flux_expression',\n",
       " 'forward_variable',\n",
       " 'functional',\n",
       " 'gene_name_reaction_rule',\n",
       " 'gene_reaction_rule',\n",
       " 'genes',\n",
       " 'get_coefficient',\n",
       " 'get_coefficients',\n",
       " 'get_compartments',\n",
       " 'id',\n",
       " 'knock_out',\n",
       " 'lower_bound',\n",
       " 'metabolites',\n",
       " 'model',\n",
       " 'name',\n",
       " 'notes',\n",
       " 'objective_coefficient',\n",
       " 'products',\n",
       " 'reactants',\n",
       " 'reaction',\n",
       " 'reduced_cost',\n",
       " 'remove_from_model',\n",
       " 'reverse_id',\n",
       " 'reverse_variable',\n",
       " 'reversibility',\n",
       " 'subsystem',\n",
       " 'subtract_metabolites',\n",
       " 'summary',\n",
       " 'update_variable_bounds',\n",
       " 'upper_bound',\n",
       " 'x',\n",
       " 'y']"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dir(model.reactions[33])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<Metabolite h2o[m] at 0x7fbf454644e0>,\n",
       " <Metabolite nadp[m] at 0x7fbf45464940>,\n",
       " <Metabolite 1a25dhvitd2[m] at 0x7fbf45464c50>]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.reactions[33].reactants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<Metabolite o2[m] at 0x7fbf454646a0>,\n",
       " <Metabolite h[m] at 0x7fbf45464860>,\n",
       " <Metabolite nadph[m] at 0x7fbf45464748>,\n",
       " <Metabolite 25hvitd2[m] at 0x7fbf45467f98>]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.reactions[33].reactants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'hmdb': 'HMDB01438', 'pubchem.compound': '22833566'}"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.reactions[33].reactants[3].annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'25hvitd2[m]'"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.reactions[33].reactants[3].id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "''"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.reactions[33].subsystem"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Group group34 at 0x7fbf42da0e10>"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.groups[33]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['KIND_TYPES',\n",
       " '__class__',\n",
       " '__delattr__',\n",
       " '__dict__',\n",
       " '__dir__',\n",
       " '__doc__',\n",
       " '__eq__',\n",
       " '__format__',\n",
       " '__ge__',\n",
       " '__getattribute__',\n",
       " '__getstate__',\n",
       " '__gt__',\n",
       " '__hash__',\n",
       " '__init__',\n",
       " '__init_subclass__',\n",
       " '__le__',\n",
       " '__len__',\n",
       " '__lt__',\n",
       " '__module__',\n",
       " '__ne__',\n",
       " '__new__',\n",
       " '__reduce__',\n",
       " '__reduce_ex__',\n",
       " '__repr__',\n",
       " '__setattr__',\n",
       " '__sizeof__',\n",
       " '__str__',\n",
       " '__subclasshook__',\n",
       " '__weakref__',\n",
       " '_annotation',\n",
       " '_id',\n",
       " '_kind',\n",
       " '_members',\n",
       " '_model',\n",
       " '_set_id_with_model',\n",
       " 'add_members',\n",
       " 'annotation',\n",
       " 'id',\n",
       " 'kind',\n",
       " 'members',\n",
       " 'name',\n",
       " 'notes',\n",
       " 'remove_members']"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dir(model.groups[33])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Glycine, serine, alanine, and threonine metabolism',\n",
       " [<Reaction 2AMACHYD at 0x7fbf62abccf8>,\n",
       "  <Reaction AACTOOR at 0x7fbf62985fd0>,\n",
       "  <Reaction ALASm at 0x7fbf62862c18>,\n",
       "  <Reaction AOBUTDsm at 0x7fbf627cae48>,\n",
       "  <Reaction BETALDHxm at 0x7fbf626c74a8>,\n",
       "  <Reaction BHMT at 0x7fbf6273c710>,\n",
       "  <Reaction CHOLD2m at 0x7fbf4537e6a0>,\n",
       "  <Reaction DMGDHm at 0x7fbf45227b70>,\n",
       "  <Reaction GCC2am at 0x7fbf44d17208>,\n",
       "  <Reaction GCC2bim at 0x7fbf44d17780>,\n",
       "  <Reaction GCC2cm at 0x7fbf44d17b70>,\n",
       "  <Reaction GCCam at 0x7fbf44cf7828>,\n",
       "  <Reaction GCCbim at 0x7fbf44d223c8>,\n",
       "  <Reaction GCCcm at 0x7fbf44d0def0>,\n",
       "  <Reaction GHMT2rm at 0x7fbf44cda0f0>,\n",
       "  <Reaction GLYATm at 0x7fbf44bf9d68>,\n",
       "  <Reaction GLYOp at 0x7fbf44c07940>,\n",
       "  <Reaction GNMT at 0x7fbf44c1c5f8>,\n",
       "  <Reaction PGCD at 0x7fbf4468c828>,\n",
       "  <Reaction SARCOXp at 0x7fbf4441ae10>,\n",
       "  <Reaction SERHL at 0x7fbf443d0eb8>,\n",
       "  <Reaction SPTix at 0x7fbf44332eb8>,\n",
       "  <Reaction r0160 at 0x7fbf44167eb8>,\n",
       "  <Reaction r0552 at 0x7fbf43ff1ef0>,\n",
       "  <Reaction r0553 at 0x7fbf43ff1e10>,\n",
       "  <Reaction RE2111M at 0x7fbf43296b00>,\n",
       "  <Reaction RE2117M at 0x7fbf432a2390>,\n",
       "  <Reaction RE2427M at 0x7fbf431d66a0>,\n",
       "  <Reaction RE2428M at 0x7fbf431d6b38>,\n",
       "  <Reaction RE2429M at 0x7fbf431d6da0>,\n",
       "  <Reaction GLYACm at 0x7fbf423c6a20>,\n",
       "  <Reaction THRACm at 0x7fbf423999b0>,\n",
       "  <Reaction ACHOMm at 0x7fbf42347828>,\n",
       "  <Reaction GHMT2r at 0x7fbf41dd6e80>,\n",
       "  <Reaction PSERT at 0x7fbf41d50a58>,\n",
       "  <Reaction PSP_L at 0x7fbf41d5ac18>,\n",
       "  <Reaction THRS at 0x7fbf41d08748>,\n",
       "  <Reaction THRD_L at 0x7fbf41c08208>,\n",
       "  <Reaction OBDHc at 0x7fbf41bb8a20>,\n",
       "  <Reaction PPIOGLYc at 0x7fbf41b804a8>,\n",
       "  <Reaction TIGGLYc at 0x7fbf41b932e8>,\n",
       "  <Reaction HEXGLYc at 0x7fbf41b3f828>,\n",
       "  <Reaction SUBGLYc at 0x7fbf41b54198>,\n",
       "  <Reaction HMR_4284 at 0x7fbf40754e10>,\n",
       "  <Reaction HMR_4466 at 0x7fbf407620f0>,\n",
       "  <Reaction HMR_4696 at 0x7fbf40762160>,\n",
       "  <Reaction HMR_4700 at 0x7fbf406f89b0>,\n",
       "  <Reaction HMR_7703 at 0x7fbf40200748>,\n",
       "  <Reaction HMR_9718 at 0x7fbf3ff6fac8>],\n",
       " {'sbo': 'SBO:0000633'}]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[\n",
    "    model.groups[33].name,\n",
    "    model.groups[33].members,\n",
    "    model.groups[33].annotation,\n",
    "]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Summary\n",
    "\n",
    "- metabolites and reactions are quite standard.\n",
    "- subsystem may not be defined in these SBML files.\n",
    "- use \"group\" as pathway\n",
    "\n",
    "Targeted structure in Azimuth \n",
    "\n",
    "```\n",
    "│   ├── RECON3D (document)\n",
    "│   │   ├── Reactions (subcollection)\n",
    "│   │   ├── Compounds (subcollection)\n",
    "│   │   └── Pathways (subcollection)\n",
    "```\n",
    "\n",
    "We will export one .py and one JSON per model."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1a25dhvitd2'"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# from metDataModel.core import Compound, Reaction, Pathway, metabolicModel\n",
    "\n",
    "def metabolite2compound(M):\n",
    "    # convert cobra Metabolite to metDataModel Compound\n",
    "    Cpd = Compound()\n",
    "    Cpd.src_id = M.id\n",
    "    Cpd.id = M.id.split(\"[\")[0]\n",
    "    Cpd.name = M.name\n",
    "    Cpd.charge = M.charge\n",
    "    Cpd.charged_formula = M.formula\n",
    "    Cpd.db_ids = M.annotation\n",
    "    return Cpd\n",
    "\n",
    "metabolite2compound(model.metabolites[33]).id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total,  8399 4140\n",
      "real,  4140 2631 2943\n"
     ]
    }
   ],
   "source": [
    "# list of Compounds\n",
    "myCpds = []\n",
    "anno = {}\n",
    "for M in model.metabolites:\n",
    "    anno[M.id.split(\"[\")[0]] = M.annotation\n",
    "    myCpds.append(metabolite2compound(M))\n",
    "    \n",
    "print(\"total, \", len(myCpds), len(anno))\n",
    "\n",
    "unique = set([C.id for C in myCpds])\n",
    "formula = set([C.charged_formula for C in myCpds if C.charged_formula])\n",
    "# a little tricker to test on dictionaries\n",
    "annotation = [v for v in anno.values() if v]\n",
    "\n",
    "print(\"real, \", len(unique), len(formula), len(annotation))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Numbers in RECON3D\n",
    "\n",
    "len(unique), len(formula), len(annotation) = \n",
    "4140, 2631, 2943"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Now do ATLAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'' is not a valid SBML 'SId'.\n",
      "Adding exchange reaction EX_m00001x with default bounds for boundary metabolite: m00001x.\n",
      "Adding exchange reaction EX_m00002x with default bounds for boundary metabolite: m00002x.\n",
      "Adding exchange reaction EX_m00032x with default bounds for boundary metabolite: m00032x.\n",
      "Adding exchange reaction EX_m00035x with default bounds for boundary metabolite: m00035x.\n",
      "Adding exchange reaction EX_m00097x with default bounds for boundary metabolite: m00097x.\n",
      "Adding exchange reaction EX_m00157x with default bounds for boundary metabolite: m00157x.\n",
      "Adding exchange reaction EX_m00179x with default bounds for boundary metabolite: m00179x.\n",
      "Adding exchange reaction EX_m00204x with default bounds for boundary metabolite: m00204x.\n",
      "Adding exchange reaction EX_m00228x with default bounds for boundary metabolite: m00228x.\n",
      "Adding exchange reaction EX_m00242x with default bounds for boundary metabolite: m00242x.\n",
      "Adding exchange reaction EX_m00266x with default bounds for boundary metabolite: m00266x.\n",
      "Adding exchange reaction EX_m00267x with default bounds for boundary metabolite: m00267x.\n",
      "Adding exchange reaction EX_m00268x with default bounds for boundary metabolite: m00268x.\n",
      "Adding exchange reaction EX_m00269x with default bounds for boundary metabolite: m00269x.\n",
      "Adding exchange reaction EX_m00325x with default bounds for boundary metabolite: m00325x.\n",
      "Adding exchange reaction EX_m00353x with default bounds for boundary metabolite: m00353x.\n",
      "Adding exchange reaction EX_m00371x with default bounds for boundary metabolite: m00371x.\n",
      "Adding exchange reaction EX_m00402x with default bounds for boundary metabolite: m00402x.\n",
      "Adding exchange reaction EX_m00403x with default bounds for boundary metabolite: m00403x.\n",
      "Adding exchange reaction EX_m00432x with default bounds for boundary metabolite: m00432x.\n",
      "Adding exchange reaction EX_m00516x with default bounds for boundary metabolite: m00516x.\n",
      "Adding exchange reaction EX_m00536x with default bounds for boundary metabolite: m00536x.\n",
      "Adding exchange reaction EX_m00545x with default bounds for boundary metabolite: m00545x.\n",
      "Adding exchange reaction EX_m00549x with default bounds for boundary metabolite: m00549x.\n",
      "Adding exchange reaction EX_m00577x with default bounds for boundary metabolite: m00577x.\n",
      "Adding exchange reaction EX_m00580x with default bounds for boundary metabolite: m00580x.\n",
      "Adding exchange reaction EX_m00613x with default bounds for boundary metabolite: m00613x.\n",
      "Adding exchange reaction EX_m00620x with default bounds for boundary metabolite: m00620x.\n",
      "Adding exchange reaction EX_m00626x with default bounds for boundary metabolite: m00626x.\n",
      "Adding exchange reaction EX_m00648x with default bounds for boundary metabolite: m00648x.\n",
      "Adding exchange reaction EX_m00665x with default bounds for boundary metabolite: m00665x.\n",
      "Adding exchange reaction EX_m00730x with default bounds for boundary metabolite: m00730x.\n",
      "Adding exchange reaction EX_m00734x with default bounds for boundary metabolite: m00734x.\n",
      "Adding exchange reaction EX_m00744x with default bounds for boundary metabolite: m00744x.\n",
      "Adding exchange reaction EX_m00771x with default bounds for boundary metabolite: m00771x.\n",
      "Adding exchange reaction EX_m00816x with default bounds for boundary metabolite: m00816x.\n",
      "Adding exchange reaction EX_m00932x with default bounds for boundary metabolite: m00932x.\n",
      "Adding exchange reaction EX_m00970x with default bounds for boundary metabolite: m00970x.\n",
      "Adding exchange reaction EX_m00986x with default bounds for boundary metabolite: m00986x.\n",
      "Adding exchange reaction EX_m00998x with default bounds for boundary metabolite: m00998x.\n",
      "Adding exchange reaction EX_m01003x with default bounds for boundary metabolite: m01003x.\n",
      "Adding exchange reaction EX_m01006x with default bounds for boundary metabolite: m01006x.\n",
      "Adding exchange reaction EX_m01007x with default bounds for boundary metabolite: m01007x.\n",
      "Adding exchange reaction EX_m01019x with default bounds for boundary metabolite: m01019x.\n",
      "Adding exchange reaction EX_m01020x with default bounds for boundary metabolite: m01020x.\n",
      "Adding exchange reaction EX_m01021x with default bounds for boundary metabolite: m01021x.\n",
      "Adding exchange reaction EX_m01033x with default bounds for boundary metabolite: m01033x.\n",
      "Adding exchange reaction EX_m01069x with default bounds for boundary metabolite: m01069x.\n",
      "Adding exchange reaction EX_m01070x with default bounds for boundary metabolite: m01070x.\n",
      "Adding exchange reaction EX_m01071x with default bounds for boundary metabolite: m01071x.\n",
      "Adding exchange reaction EX_m01098x with default bounds for boundary metabolite: m01098x.\n",
      "Adding exchange reaction EX_m01100x with default bounds for boundary metabolite: m01100x.\n",
      "Adding exchange reaction EX_m01107x with default bounds for boundary metabolite: m01107x.\n",
      "Adding exchange reaction EX_m01109x with default bounds for boundary metabolite: m01109x.\n",
      "Adding exchange reaction EX_m01111x with default bounds for boundary metabolite: m01111x.\n",
      "Adding exchange reaction EX_m01115x with default bounds for boundary metabolite: m01115x.\n",
      "Adding exchange reaction EX_m01158x with default bounds for boundary metabolite: m01158x.\n",
      "Adding exchange reaction EX_m01159x with default bounds for boundary metabolite: m01159x.\n",
      "Adding exchange reaction EX_m01163x with default bounds for boundary metabolite: m01163x.\n",
      "Adding exchange reaction EX_m01174x with default bounds for boundary metabolite: m01174x.\n",
      "Adding exchange reaction EX_m01244x with default bounds for boundary metabolite: m01244x.\n",
      "Adding exchange reaction EX_m01245x with default bounds for boundary metabolite: m01245x.\n",
      "Adding exchange reaction EX_m01249x with default bounds for boundary metabolite: m01249x.\n",
      "Adding exchange reaction EX_m01252x with default bounds for boundary metabolite: m01252x.\n",
      "Adding exchange reaction EX_m01253x with default bounds for boundary metabolite: m01253x.\n",
      "Adding exchange reaction EX_m01256x with default bounds for boundary metabolite: m01256x.\n",
      "Adding exchange reaction EX_m01260x with default bounds for boundary metabolite: m01260x.\n",
      "Adding exchange reaction EX_m01279x with default bounds for boundary metabolite: m01279x.\n",
      "Adding exchange reaction EX_m01280x with default bounds for boundary metabolite: m01280x.\n",
      "Adding exchange reaction EX_m01285x with default bounds for boundary metabolite: m01285x.\n",
      "Adding exchange reaction EX_m01286x with default bounds for boundary metabolite: m01286x.\n",
      "Adding exchange reaction EX_m01287x with default bounds for boundary metabolite: m01287x.\n",
      "Adding exchange reaction EX_m01288x with default bounds for boundary metabolite: m01288x.\n",
      "Adding exchange reaction EX_m01289x with default bounds for boundary metabolite: m01289x.\n",
      "Adding exchange reaction EX_m01290x with default bounds for boundary metabolite: m01290x.\n",
      "Adding exchange reaction EX_m01296x with default bounds for boundary metabolite: m01296x.\n",
      "Adding exchange reaction EX_m01298x with default bounds for boundary metabolite: m01298x.\n",
      "Adding exchange reaction EX_m01303x with default bounds for boundary metabolite: m01303x.\n",
      "Adding exchange reaction EX_m01306x with default bounds for boundary metabolite: m01306x.\n",
      "Adding exchange reaction EX_m01307x with default bounds for boundary metabolite: m01307x.\n",
      "Adding exchange reaction EX_m01308x with default bounds for boundary metabolite: m01308x.\n",
      "Adding exchange reaction EX_m01309x with default bounds for boundary metabolite: m01309x.\n",
      "Adding exchange reaction EX_m01326x with default bounds for boundary metabolite: m01326x.\n",
      "Adding exchange reaction EX_m01327x with default bounds for boundary metabolite: m01327x.\n",
      "Adding exchange reaction EX_m01330x with default bounds for boundary metabolite: m01330x.\n",
      "Adding exchange reaction EX_m01334x with default bounds for boundary metabolite: m01334x.\n",
      "Adding exchange reaction EX_m01338x with default bounds for boundary metabolite: m01338x.\n",
      "Adding exchange reaction EX_m01339x with default bounds for boundary metabolite: m01339x.\n",
      "Adding exchange reaction EX_m01344x with default bounds for boundary metabolite: m01344x.\n",
      "Adding exchange reaction EX_m01356x with default bounds for boundary metabolite: m01356x.\n",
      "Adding exchange reaction EX_m01361x with default bounds for boundary metabolite: m01361x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_m01365x with default bounds for boundary metabolite: m01365x.\n",
      "Adding exchange reaction EX_m01368x with default bounds for boundary metabolite: m01368x.\n",
      "Adding exchange reaction EX_m01369x with default bounds for boundary metabolite: m01369x.\n",
      "Adding exchange reaction EX_m01370x with default bounds for boundary metabolite: m01370x.\n",
      "Adding exchange reaction EX_m01374x with default bounds for boundary metabolite: m01374x.\n",
      "Adding exchange reaction EX_m01383x with default bounds for boundary metabolite: m01383x.\n",
      "Adding exchange reaction EX_m01385x with default bounds for boundary metabolite: m01385x.\n",
      "Adding exchange reaction EX_m01393x with default bounds for boundary metabolite: m01393x.\n",
      "Adding exchange reaction EX_m01396x with default bounds for boundary metabolite: m01396x.\n",
      "Adding exchange reaction EX_m01397x with default bounds for boundary metabolite: m01397x.\n",
      "Adding exchange reaction EX_m01398x with default bounds for boundary metabolite: m01398x.\n",
      "Adding exchange reaction EX_m01400x with default bounds for boundary metabolite: m01400x.\n",
      "Adding exchange reaction EX_m01401x with default bounds for boundary metabolite: m01401x.\n",
      "Adding exchange reaction EX_m01403x with default bounds for boundary metabolite: m01403x.\n",
      "Adding exchange reaction EX_m01410x with default bounds for boundary metabolite: m01410x.\n",
      "Adding exchange reaction EX_m01413x with default bounds for boundary metabolite: m01413x.\n",
      "Adding exchange reaction EX_m01415x with default bounds for boundary metabolite: m01415x.\n",
      "Adding exchange reaction EX_m01418x with default bounds for boundary metabolite: m01418x.\n",
      "Adding exchange reaction EX_m01419x with default bounds for boundary metabolite: m01419x.\n",
      "Adding exchange reaction EX_m01433x with default bounds for boundary metabolite: m01433x.\n",
      "Adding exchange reaction EX_m01438x with default bounds for boundary metabolite: m01438x.\n",
      "Adding exchange reaction EX_m01442x with default bounds for boundary metabolite: m01442x.\n",
      "Adding exchange reaction EX_m01445x with default bounds for boundary metabolite: m01445x.\n",
      "Adding exchange reaction EX_m01450x with default bounds for boundary metabolite: m01450x.\n",
      "Adding exchange reaction EX_m01513x with default bounds for boundary metabolite: m01513x.\n",
      "Adding exchange reaction EX_m01517x with default bounds for boundary metabolite: m01517x.\n",
      "Adding exchange reaction EX_m01525x with default bounds for boundary metabolite: m01525x.\n",
      "Adding exchange reaction EX_m01533x with default bounds for boundary metabolite: m01533x.\n",
      "Adding exchange reaction EX_m01543x with default bounds for boundary metabolite: m01543x.\n",
      "Adding exchange reaction EX_m01554x with default bounds for boundary metabolite: m01554x.\n",
      "Adding exchange reaction EX_m01569x with default bounds for boundary metabolite: m01569x.\n",
      "Adding exchange reaction EX_m01570x with default bounds for boundary metabolite: m01570x.\n",
      "Adding exchange reaction EX_m01587x with default bounds for boundary metabolite: m01587x.\n",
      "Adding exchange reaction EX_m01588x with default bounds for boundary metabolite: m01588x.\n",
      "Adding exchange reaction EX_m01590x with default bounds for boundary metabolite: m01590x.\n",
      "Adding exchange reaction EX_m01595x with default bounds for boundary metabolite: m01595x.\n",
      "Adding exchange reaction EX_m01596x with default bounds for boundary metabolite: m01596x.\n",
      "Adding exchange reaction EX_m01603x with default bounds for boundary metabolite: m01603x.\n",
      "Adding exchange reaction EX_m01604x with default bounds for boundary metabolite: m01604x.\n",
      "Adding exchange reaction EX_m01610x with default bounds for boundary metabolite: m01610x.\n",
      "Adding exchange reaction EX_m01612x with default bounds for boundary metabolite: m01612x.\n",
      "Adding exchange reaction EX_m01613x with default bounds for boundary metabolite: m01613x.\n",
      "Adding exchange reaction EX_m01614x with default bounds for boundary metabolite: m01614x.\n",
      "Adding exchange reaction EX_m01615x with default bounds for boundary metabolite: m01615x.\n",
      "Adding exchange reaction EX_m01617x with default bounds for boundary metabolite: m01617x.\n",
      "Adding exchange reaction EX_m01619x with default bounds for boundary metabolite: m01619x.\n",
      "Adding exchange reaction EX_m01621x with default bounds for boundary metabolite: m01621x.\n",
      "Adding exchange reaction EX_m01626x with default bounds for boundary metabolite: m01626x.\n",
      "Adding exchange reaction EX_m01628x with default bounds for boundary metabolite: m01628x.\n",
      "Adding exchange reaction EX_m01629x with default bounds for boundary metabolite: m01629x.\n",
      "Adding exchange reaction EX_m01630x with default bounds for boundary metabolite: m01630x.\n",
      "Adding exchange reaction EX_m01632x with default bounds for boundary metabolite: m01632x.\n",
      "Adding exchange reaction EX_m01633x with default bounds for boundary metabolite: m01633x.\n",
      "Adding exchange reaction EX_m01638x with default bounds for boundary metabolite: m01638x.\n",
      "Adding exchange reaction EX_m01640x with default bounds for boundary metabolite: m01640x.\n",
      "Adding exchange reaction EX_m01641x with default bounds for boundary metabolite: m01641x.\n",
      "Adding exchange reaction EX_m01647x with default bounds for boundary metabolite: m01647x.\n",
      "Adding exchange reaction EX_m01648x with default bounds for boundary metabolite: m01648x.\n",
      "Adding exchange reaction EX_m01652x with default bounds for boundary metabolite: m01652x.\n",
      "Adding exchange reaction EX_m01655x with default bounds for boundary metabolite: m01655x.\n",
      "Adding exchange reaction EX_m01659x with default bounds for boundary metabolite: m01659x.\n",
      "Adding exchange reaction EX_m01665x with default bounds for boundary metabolite: m01665x.\n",
      "Adding exchange reaction EX_m01666x with default bounds for boundary metabolite: m01666x.\n",
      "Adding exchange reaction EX_m01668x with default bounds for boundary metabolite: m01668x.\n",
      "Adding exchange reaction EX_m01669x with default bounds for boundary metabolite: m01669x.\n",
      "Adding exchange reaction EX_m01671x with default bounds for boundary metabolite: m01671x.\n",
      "Adding exchange reaction EX_m01672x with default bounds for boundary metabolite: m01672x.\n",
      "Adding exchange reaction EX_m01673x with default bounds for boundary metabolite: m01673x.\n",
      "Adding exchange reaction EX_m01682x with default bounds for boundary metabolite: m01682x.\n",
      "Adding exchange reaction EX_m01687x with default bounds for boundary metabolite: m01687x.\n",
      "Adding exchange reaction EX_m01695x with default bounds for boundary metabolite: m01695x.\n",
      "Adding exchange reaction EX_m01700x with default bounds for boundary metabolite: m01700x.\n",
      "Adding exchange reaction EX_m01704x with default bounds for boundary metabolite: m01704x.\n",
      "Adding exchange reaction EX_m01711x with default bounds for boundary metabolite: m01711x.\n",
      "Adding exchange reaction EX_m01712x with default bounds for boundary metabolite: m01712x.\n",
      "Adding exchange reaction EX_m01714x with default bounds for boundary metabolite: m01714x.\n",
      "Adding exchange reaction EX_m01716x with default bounds for boundary metabolite: m01716x.\n",
      "Adding exchange reaction EX_m01730x with default bounds for boundary metabolite: m01730x.\n",
      "Adding exchange reaction EX_m01736x with default bounds for boundary metabolite: m01736x.\n",
      "Adding exchange reaction EX_m01737x with default bounds for boundary metabolite: m01737x.\n",
      "Adding exchange reaction EX_m01740x with default bounds for boundary metabolite: m01740x.\n",
      "Adding exchange reaction EX_m01742x with default bounds for boundary metabolite: m01742x.\n",
      "Adding exchange reaction EX_m01743x with default bounds for boundary metabolite: m01743x.\n",
      "Adding exchange reaction EX_m01744x with default bounds for boundary metabolite: m01744x.\n",
      "Adding exchange reaction EX_m01745x with default bounds for boundary metabolite: m01745x.\n",
      "Adding exchange reaction EX_m01758x with default bounds for boundary metabolite: m01758x.\n",
      "Adding exchange reaction EX_m01765x with default bounds for boundary metabolite: m01765x.\n",
      "Adding exchange reaction EX_m01768x with default bounds for boundary metabolite: m01768x.\n",
      "Adding exchange reaction EX_m01786x with default bounds for boundary metabolite: m01786x.\n",
      "Adding exchange reaction EX_m01787x with default bounds for boundary metabolite: m01787x.\n",
      "Adding exchange reaction EX_m01788x with default bounds for boundary metabolite: m01788x.\n",
      "Adding exchange reaction EX_m01789x with default bounds for boundary metabolite: m01789x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_m01795x with default bounds for boundary metabolite: m01795x.\n",
      "Adding exchange reaction EX_m01796x with default bounds for boundary metabolite: m01796x.\n",
      "Adding exchange reaction EX_m01797x with default bounds for boundary metabolite: m01797x.\n",
      "Adding exchange reaction EX_m01799x with default bounds for boundary metabolite: m01799x.\n",
      "Adding exchange reaction EX_m01800x with default bounds for boundary metabolite: m01800x.\n",
      "Adding exchange reaction EX_m02458x with default bounds for boundary metabolite: m02458x.\n",
      "Adding exchange reaction EX_m01821x with default bounds for boundary metabolite: m01821x.\n",
      "Adding exchange reaction EX_m01822x with default bounds for boundary metabolite: m01822x.\n",
      "Adding exchange reaction EX_m01830x with default bounds for boundary metabolite: m01830x.\n",
      "Adding exchange reaction EX_m01833x with default bounds for boundary metabolite: m01833x.\n",
      "Adding exchange reaction EX_m01840x with default bounds for boundary metabolite: m01840x.\n",
      "Adding exchange reaction EX_m01850x with default bounds for boundary metabolite: m01850x.\n",
      "Adding exchange reaction EX_m01851x with default bounds for boundary metabolite: m01851x.\n",
      "Adding exchange reaction EX_m01852x with default bounds for boundary metabolite: m01852x.\n",
      "Adding exchange reaction EX_m01853x with default bounds for boundary metabolite: m01853x.\n",
      "Adding exchange reaction EX_m01859x with default bounds for boundary metabolite: m01859x.\n",
      "Adding exchange reaction EX_m01861x with default bounds for boundary metabolite: m01861x.\n",
      "Adding exchange reaction EX_m01870x with default bounds for boundary metabolite: m01870x.\n",
      "Adding exchange reaction EX_m01872x with default bounds for boundary metabolite: m01872x.\n",
      "Adding exchange reaction EX_m01874x with default bounds for boundary metabolite: m01874x.\n",
      "Adding exchange reaction EX_m01881x with default bounds for boundary metabolite: m01881x.\n",
      "Adding exchange reaction EX_m01883x with default bounds for boundary metabolite: m01883x.\n",
      "Adding exchange reaction EX_m01884x with default bounds for boundary metabolite: m01884x.\n",
      "Adding exchange reaction EX_m01887x with default bounds for boundary metabolite: m01887x.\n",
      "Adding exchange reaction EX_m01910x with default bounds for boundary metabolite: m01910x.\n",
      "Adding exchange reaction EX_m01913x with default bounds for boundary metabolite: m01913x.\n",
      "Adding exchange reaction EX_m01914x with default bounds for boundary metabolite: m01914x.\n",
      "Adding exchange reaction EX_m01915x with default bounds for boundary metabolite: m01915x.\n",
      "Adding exchange reaction EX_m01916x with default bounds for boundary metabolite: m01916x.\n",
      "Adding exchange reaction EX_m01917x with default bounds for boundary metabolite: m01917x.\n",
      "Adding exchange reaction EX_m01919x with default bounds for boundary metabolite: m01919x.\n",
      "Adding exchange reaction EX_m01935x with default bounds for boundary metabolite: m01935x.\n",
      "Adding exchange reaction EX_m01938x with default bounds for boundary metabolite: m01938x.\n",
      "Adding exchange reaction EX_m01944x with default bounds for boundary metabolite: m01944x.\n",
      "Adding exchange reaction EX_m01945x with default bounds for boundary metabolite: m01945x.\n",
      "Adding exchange reaction EX_m01948x with default bounds for boundary metabolite: m01948x.\n",
      "Adding exchange reaction EX_m01955x with default bounds for boundary metabolite: m01955x.\n",
      "Adding exchange reaction EX_m01959x with default bounds for boundary metabolite: m01959x.\n",
      "Adding exchange reaction EX_m01962x with default bounds for boundary metabolite: m01962x.\n",
      "Adding exchange reaction EX_m01965x with default bounds for boundary metabolite: m01965x.\n",
      "Adding exchange reaction EX_m01966x with default bounds for boundary metabolite: m01966x.\n",
      "Adding exchange reaction EX_m01974x with default bounds for boundary metabolite: m01974x.\n",
      "Adding exchange reaction EX_m01975x with default bounds for boundary metabolite: m01975x.\n",
      "Adding exchange reaction EX_m01982x with default bounds for boundary metabolite: m01982x.\n",
      "Adding exchange reaction EX_m01983x with default bounds for boundary metabolite: m01983x.\n",
      "Adding exchange reaction EX_m01986x with default bounds for boundary metabolite: m01986x.\n",
      "Adding exchange reaction EX_m01987x with default bounds for boundary metabolite: m01987x.\n",
      "Adding exchange reaction EX_m01988x with default bounds for boundary metabolite: m01988x.\n",
      "Adding exchange reaction EX_m01989x with default bounds for boundary metabolite: m01989x.\n",
      "Adding exchange reaction EX_m03161x with default bounds for boundary metabolite: m03161x.\n",
      "Adding exchange reaction EX_m01999x with default bounds for boundary metabolite: m01999x.\n",
      "Adding exchange reaction EX_m02001x with default bounds for boundary metabolite: m02001x.\n",
      "Adding exchange reaction EX_m02016x with default bounds for boundary metabolite: m02016x.\n",
      "Adding exchange reaction EX_m02018x with default bounds for boundary metabolite: m02018x.\n",
      "Adding exchange reaction EX_m02019x with default bounds for boundary metabolite: m02019x.\n",
      "Adding exchange reaction EX_m02022x with default bounds for boundary metabolite: m02022x.\n",
      "Adding exchange reaction EX_m02023x with default bounds for boundary metabolite: m02023x.\n",
      "Adding exchange reaction EX_m02024x with default bounds for boundary metabolite: m02024x.\n",
      "Adding exchange reaction EX_m02026x with default bounds for boundary metabolite: m02026x.\n",
      "Adding exchange reaction EX_m02027x with default bounds for boundary metabolite: m02027x.\n",
      "Adding exchange reaction EX_m02028x with default bounds for boundary metabolite: m02028x.\n",
      "Adding exchange reaction EX_m02034x with default bounds for boundary metabolite: m02034x.\n",
      "Adding exchange reaction EX_m02037x with default bounds for boundary metabolite: m02037x.\n",
      "Adding exchange reaction EX_m02038x with default bounds for boundary metabolite: m02038x.\n",
      "Adding exchange reaction EX_m02039x with default bounds for boundary metabolite: m02039x.\n",
      "Adding exchange reaction EX_m02040x with default bounds for boundary metabolite: m02040x.\n",
      "Adding exchange reaction EX_m02041x with default bounds for boundary metabolite: m02041x.\n",
      "Adding exchange reaction EX_m02042x with default bounds for boundary metabolite: m02042x.\n",
      "Adding exchange reaction EX_m02046x with default bounds for boundary metabolite: m02046x.\n",
      "Adding exchange reaction EX_m02047x with default bounds for boundary metabolite: m02047x.\n",
      "Adding exchange reaction EX_m02048x with default bounds for boundary metabolite: m02048x.\n",
      "Adding exchange reaction EX_m02049x with default bounds for boundary metabolite: m02049x.\n",
      "Adding exchange reaction EX_m02050x with default bounds for boundary metabolite: m02050x.\n",
      "Adding exchange reaction EX_m02054x with default bounds for boundary metabolite: m02054x.\n",
      "Adding exchange reaction EX_m02104x with default bounds for boundary metabolite: m02104x.\n",
      "Adding exchange reaction EX_m02105x with default bounds for boundary metabolite: m02105x.\n",
      "Adding exchange reaction EX_m02108x with default bounds for boundary metabolite: m02108x.\n",
      "Adding exchange reaction EX_m02118x with default bounds for boundary metabolite: m02118x.\n",
      "Adding exchange reaction EX_m02119x with default bounds for boundary metabolite: m02119x.\n",
      "Adding exchange reaction EX_m02120x with default bounds for boundary metabolite: m02120x.\n",
      "Adding exchange reaction EX_m02125x with default bounds for boundary metabolite: m02125x.\n",
      "Adding exchange reaction EX_m02136x with default bounds for boundary metabolite: m02136x.\n",
      "Adding exchange reaction EX_m02137x with default bounds for boundary metabolite: m02137x.\n",
      "Adding exchange reaction EX_m02139x with default bounds for boundary metabolite: m02139x.\n",
      "Adding exchange reaction EX_m02141x with default bounds for boundary metabolite: m02141x.\n",
      "Adding exchange reaction EX_m02145x with default bounds for boundary metabolite: m02145x.\n",
      "Adding exchange reaction EX_m02147x with default bounds for boundary metabolite: m02147x.\n",
      "Adding exchange reaction EX_m02150x with default bounds for boundary metabolite: m02150x.\n",
      "Adding exchange reaction EX_m02153x with default bounds for boundary metabolite: m02153x.\n",
      "Adding exchange reaction EX_m02155x with default bounds for boundary metabolite: m02155x.\n",
      "Adding exchange reaction EX_m02159x with default bounds for boundary metabolite: m02159x.\n",
      "Adding exchange reaction EX_m02161x with default bounds for boundary metabolite: m02161x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_m02164x with default bounds for boundary metabolite: m02164x.\n",
      "Adding exchange reaction EX_m02167x with default bounds for boundary metabolite: m02167x.\n",
      "Adding exchange reaction EX_m02170x with default bounds for boundary metabolite: m02170x.\n",
      "Adding exchange reaction EX_m02171x with default bounds for boundary metabolite: m02171x.\n",
      "Adding exchange reaction EX_m02174x with default bounds for boundary metabolite: m02174x.\n",
      "Adding exchange reaction EX_m02182x with default bounds for boundary metabolite: m02182x.\n",
      "Adding exchange reaction EX_m02184x with default bounds for boundary metabolite: m02184x.\n",
      "Adding exchange reaction EX_m02191x with default bounds for boundary metabolite: m02191x.\n",
      "Adding exchange reaction EX_m02193x with default bounds for boundary metabolite: m02193x.\n",
      "Adding exchange reaction EX_m02198x with default bounds for boundary metabolite: m02198x.\n",
      "Adding exchange reaction EX_m02199x with default bounds for boundary metabolite: m02199x.\n",
      "Adding exchange reaction EX_m02200x with default bounds for boundary metabolite: m02200x.\n",
      "Adding exchange reaction EX_m02237x with default bounds for boundary metabolite: m02237x.\n",
      "Adding exchange reaction EX_m02278x with default bounds for boundary metabolite: m02278x.\n",
      "Adding exchange reaction EX_m02288x with default bounds for boundary metabolite: m02288x.\n",
      "Adding exchange reaction EX_m02303x with default bounds for boundary metabolite: m02303x.\n",
      "Adding exchange reaction EX_m02325x with default bounds for boundary metabolite: m02325x.\n",
      "Adding exchange reaction EX_m02331x with default bounds for boundary metabolite: m02331x.\n",
      "Adding exchange reaction EX_m02332x with default bounds for boundary metabolite: m02332x.\n",
      "Adding exchange reaction EX_m02337x with default bounds for boundary metabolite: m02337x.\n",
      "Adding exchange reaction EX_m02338x with default bounds for boundary metabolite: m02338x.\n",
      "Adding exchange reaction EX_m02348x with default bounds for boundary metabolite: m02348x.\n",
      "Adding exchange reaction EX_m02352x with default bounds for boundary metabolite: m02352x.\n",
      "Adding exchange reaction EX_m02353x with default bounds for boundary metabolite: m02353x.\n",
      "Adding exchange reaction EX_m02354x with default bounds for boundary metabolite: m02354x.\n",
      "Adding exchange reaction EX_m02357x with default bounds for boundary metabolite: m02357x.\n",
      "Adding exchange reaction EX_m02360x with default bounds for boundary metabolite: m02360x.\n",
      "Adding exchange reaction EX_m02362x with default bounds for boundary metabolite: m02362x.\n",
      "Adding exchange reaction EX_m02364x with default bounds for boundary metabolite: m02364x.\n",
      "Adding exchange reaction EX_m02366x with default bounds for boundary metabolite: m02366x.\n",
      "Adding exchange reaction EX_m02369x with default bounds for boundary metabolite: m02369x.\n",
      "Adding exchange reaction EX_m02370x with default bounds for boundary metabolite: m02370x.\n",
      "Adding exchange reaction EX_m02384x with default bounds for boundary metabolite: m02384x.\n",
      "Adding exchange reaction EX_m02386x with default bounds for boundary metabolite: m02386x.\n",
      "Adding exchange reaction EX_m02387x with default bounds for boundary metabolite: m02387x.\n",
      "Adding exchange reaction EX_m02389x with default bounds for boundary metabolite: m02389x.\n",
      "Adding exchange reaction EX_m02394x with default bounds for boundary metabolite: m02394x.\n",
      "Adding exchange reaction EX_m02403x with default bounds for boundary metabolite: m02403x.\n",
      "Adding exchange reaction EX_m02407x with default bounds for boundary metabolite: m02407x.\n",
      "Adding exchange reaction EX_m02414x with default bounds for boundary metabolite: m02414x.\n",
      "Adding exchange reaction EX_m02418x with default bounds for boundary metabolite: m02418x.\n",
      "Adding exchange reaction EX_m02426x with default bounds for boundary metabolite: m02426x.\n",
      "Adding exchange reaction EX_m02434x with default bounds for boundary metabolite: m02434x.\n",
      "Adding exchange reaction EX_m02440x with default bounds for boundary metabolite: m02440x.\n",
      "Adding exchange reaction EX_m02445x with default bounds for boundary metabolite: m02445x.\n",
      "Adding exchange reaction EX_m02450x with default bounds for boundary metabolite: m02450x.\n",
      "Adding exchange reaction EX_m02452x with default bounds for boundary metabolite: m02452x.\n",
      "Adding exchange reaction EX_m02453x with default bounds for boundary metabolite: m02453x.\n",
      "Adding exchange reaction EX_m02462x with default bounds for boundary metabolite: m02462x.\n",
      "Adding exchange reaction EX_m02470x with default bounds for boundary metabolite: m02470x.\n",
      "Adding exchange reaction EX_m02471x with default bounds for boundary metabolite: m02471x.\n",
      "Adding exchange reaction EX_m02475x with default bounds for boundary metabolite: m02475x.\n",
      "Adding exchange reaction EX_m02477x with default bounds for boundary metabolite: m02477x.\n",
      "Adding exchange reaction EX_m02510x with default bounds for boundary metabolite: m02510x.\n",
      "Adding exchange reaction EX_m02516x with default bounds for boundary metabolite: m02516x.\n",
      "Adding exchange reaction EX_m02519x with default bounds for boundary metabolite: m02519x.\n",
      "Adding exchange reaction EX_m02524x with default bounds for boundary metabolite: m02524x.\n",
      "Adding exchange reaction EX_m02527x with default bounds for boundary metabolite: m02527x.\n",
      "Adding exchange reaction EX_m02552x with default bounds for boundary metabolite: m02552x.\n",
      "Adding exchange reaction EX_m02554x with default bounds for boundary metabolite: m02554x.\n",
      "Adding exchange reaction EX_m02556x with default bounds for boundary metabolite: m02556x.\n",
      "Adding exchange reaction EX_m02560x with default bounds for boundary metabolite: m02560x.\n",
      "Adding exchange reaction EX_m02561x with default bounds for boundary metabolite: m02561x.\n",
      "Adding exchange reaction EX_m02578x with default bounds for boundary metabolite: m02578x.\n",
      "Adding exchange reaction EX_m02583x with default bounds for boundary metabolite: m02583x.\n",
      "Adding exchange reaction EX_m02586x with default bounds for boundary metabolite: m02586x.\n",
      "Adding exchange reaction EX_m02587x with default bounds for boundary metabolite: m02587x.\n",
      "Adding exchange reaction EX_m02588x with default bounds for boundary metabolite: m02588x.\n",
      "Adding exchange reaction EX_m02609x with default bounds for boundary metabolite: m02609x.\n",
      "Adding exchange reaction EX_m02614x with default bounds for boundary metabolite: m02614x.\n",
      "Adding exchange reaction EX_m02617x with default bounds for boundary metabolite: m02617x.\n",
      "Adding exchange reaction EX_m02620x with default bounds for boundary metabolite: m02620x.\n",
      "Adding exchange reaction EX_m02630x with default bounds for boundary metabolite: m02630x.\n",
      "Adding exchange reaction EX_m02631x with default bounds for boundary metabolite: m02631x.\n",
      "Adding exchange reaction EX_m02642x with default bounds for boundary metabolite: m02642x.\n",
      "Adding exchange reaction EX_m02653x with default bounds for boundary metabolite: m02653x.\n",
      "Adding exchange reaction EX_m02658x with default bounds for boundary metabolite: m02658x.\n",
      "Adding exchange reaction EX_m02659x with default bounds for boundary metabolite: m02659x.\n",
      "Adding exchange reaction EX_m02661x with default bounds for boundary metabolite: m02661x.\n",
      "Adding exchange reaction EX_m02672x with default bounds for boundary metabolite: m02672x.\n",
      "Adding exchange reaction EX_m02673x with default bounds for boundary metabolite: m02673x.\n",
      "Adding exchange reaction EX_m02680x with default bounds for boundary metabolite: m02680x.\n",
      "Adding exchange reaction EX_m02691x with default bounds for boundary metabolite: m02691x.\n",
      "Adding exchange reaction EX_m02692x with default bounds for boundary metabolite: m02692x.\n",
      "Adding exchange reaction EX_m02712x with default bounds for boundary metabolite: m02712x.\n",
      "Adding exchange reaction EX_m02722x with default bounds for boundary metabolite: m02722x.\n",
      "Adding exchange reaction EX_m02723x with default bounds for boundary metabolite: m02723x.\n",
      "Adding exchange reaction EX_m02724x with default bounds for boundary metabolite: m02724x.\n",
      "Adding exchange reaction EX_m02740x with default bounds for boundary metabolite: m02740x.\n",
      "Adding exchange reaction EX_m02744x with default bounds for boundary metabolite: m02744x.\n",
      "Adding exchange reaction EX_m02746x with default bounds for boundary metabolite: m02746x.\n",
      "Adding exchange reaction EX_m02751x with default bounds for boundary metabolite: m02751x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_m02754x with default bounds for boundary metabolite: m02754x.\n",
      "Adding exchange reaction EX_m02769x with default bounds for boundary metabolite: m02769x.\n",
      "Adding exchange reaction EX_m02770x with default bounds for boundary metabolite: m02770x.\n",
      "Adding exchange reaction EX_m02772x with default bounds for boundary metabolite: m02772x.\n",
      "Adding exchange reaction EX_m02783x with default bounds for boundary metabolite: m02783x.\n",
      "Adding exchange reaction EX_m02785x with default bounds for boundary metabolite: m02785x.\n",
      "Adding exchange reaction EX_m02786x with default bounds for boundary metabolite: m02786x.\n",
      "Adding exchange reaction EX_m02789x with default bounds for boundary metabolite: m02789x.\n",
      "Adding exchange reaction EX_m02813x with default bounds for boundary metabolite: m02813x.\n",
      "Adding exchange reaction EX_m02814x with default bounds for boundary metabolite: m02814x.\n",
      "Adding exchange reaction EX_m02815x with default bounds for boundary metabolite: m02815x.\n",
      "Adding exchange reaction EX_m02817x with default bounds for boundary metabolite: m02817x.\n",
      "Adding exchange reaction EX_m02819x with default bounds for boundary metabolite: m02819x.\n",
      "Adding exchange reaction EX_m02833x with default bounds for boundary metabolite: m02833x.\n",
      "Adding exchange reaction EX_m02834x with default bounds for boundary metabolite: m02834x.\n",
      "Adding exchange reaction EX_m02836x with default bounds for boundary metabolite: m02836x.\n",
      "Adding exchange reaction EX_m02841x with default bounds for boundary metabolite: m02841x.\n",
      "Adding exchange reaction EX_m02842x with default bounds for boundary metabolite: m02842x.\n",
      "Adding exchange reaction EX_m02843x with default bounds for boundary metabolite: m02843x.\n",
      "Adding exchange reaction EX_m02880x with default bounds for boundary metabolite: m02880x.\n",
      "Adding exchange reaction EX_m02885x with default bounds for boundary metabolite: m02885x.\n",
      "Adding exchange reaction EX_m02896x with default bounds for boundary metabolite: m02896x.\n",
      "Adding exchange reaction EX_m02897x with default bounds for boundary metabolite: m02897x.\n",
      "Adding exchange reaction EX_m02902x with default bounds for boundary metabolite: m02902x.\n",
      "Adding exchange reaction EX_m02903x with default bounds for boundary metabolite: m02903x.\n",
      "Adding exchange reaction EX_m02907x with default bounds for boundary metabolite: m02907x.\n",
      "Adding exchange reaction EX_m02909x with default bounds for boundary metabolite: m02909x.\n",
      "Adding exchange reaction EX_m02924x with default bounds for boundary metabolite: m02924x.\n",
      "Adding exchange reaction EX_m02926x with default bounds for boundary metabolite: m02926x.\n",
      "Adding exchange reaction EX_m02928x with default bounds for boundary metabolite: m02928x.\n",
      "Adding exchange reaction EX_m02930x with default bounds for boundary metabolite: m02930x.\n",
      "Adding exchange reaction EX_m02931x with default bounds for boundary metabolite: m02931x.\n",
      "Adding exchange reaction EX_m02936x with default bounds for boundary metabolite: m02936x.\n",
      "Adding exchange reaction EX_m02937x with default bounds for boundary metabolite: m02937x.\n",
      "Adding exchange reaction EX_m02943x with default bounds for boundary metabolite: m02943x.\n",
      "Adding exchange reaction EX_m02945x with default bounds for boundary metabolite: m02945x.\n",
      "Adding exchange reaction EX_m02946x with default bounds for boundary metabolite: m02946x.\n",
      "Adding exchange reaction EX_m02949x with default bounds for boundary metabolite: m02949x.\n",
      "Adding exchange reaction EX_m02957x with default bounds for boundary metabolite: m02957x.\n",
      "Adding exchange reaction EX_m02961x with default bounds for boundary metabolite: m02961x.\n",
      "Adding exchange reaction EX_m02962x with default bounds for boundary metabolite: m02962x.\n",
      "Adding exchange reaction EX_m02963x with default bounds for boundary metabolite: m02963x.\n",
      "Adding exchange reaction EX_m02967x with default bounds for boundary metabolite: m02967x.\n",
      "Adding exchange reaction EX_m02968x with default bounds for boundary metabolite: m02968x.\n",
      "Adding exchange reaction EX_m02969x with default bounds for boundary metabolite: m02969x.\n",
      "Adding exchange reaction EX_m02980x with default bounds for boundary metabolite: m02980x.\n",
      "Adding exchange reaction EX_m02982x with default bounds for boundary metabolite: m02982x.\n",
      "Adding exchange reaction EX_m02983x with default bounds for boundary metabolite: m02983x.\n",
      "Adding exchange reaction EX_m02985x with default bounds for boundary metabolite: m02985x.\n",
      "Adding exchange reaction EX_m02986x with default bounds for boundary metabolite: m02986x.\n",
      "Adding exchange reaction EX_m02991x with default bounds for boundary metabolite: m02991x.\n",
      "Adding exchange reaction EX_m02993x with default bounds for boundary metabolite: m02993x.\n",
      "Adding exchange reaction EX_m02994x with default bounds for boundary metabolite: m02994x.\n",
      "Adding exchange reaction EX_m02996x with default bounds for boundary metabolite: m02996x.\n",
      "Adding exchange reaction EX_m02997x with default bounds for boundary metabolite: m02997x.\n",
      "Adding exchange reaction EX_m02998x with default bounds for boundary metabolite: m02998x.\n",
      "Adding exchange reaction EX_m03001x with default bounds for boundary metabolite: m03001x.\n",
      "Adding exchange reaction EX_m03039x with default bounds for boundary metabolite: m03039x.\n",
      "Adding exchange reaction EX_m03044x with default bounds for boundary metabolite: m03044x.\n",
      "Adding exchange reaction EX_m03052x with default bounds for boundary metabolite: m03052x.\n",
      "Adding exchange reaction EX_m03089x with default bounds for boundary metabolite: m03089x.\n",
      "Adding exchange reaction EX_m03096x with default bounds for boundary metabolite: m03096x.\n",
      "Adding exchange reaction EX_m03098x with default bounds for boundary metabolite: m03098x.\n",
      "Adding exchange reaction EX_m03100x with default bounds for boundary metabolite: m03100x.\n",
      "Adding exchange reaction EX_m03101x with default bounds for boundary metabolite: m03101x.\n",
      "Adding exchange reaction EX_m03106x with default bounds for boundary metabolite: m03106x.\n",
      "Adding exchange reaction EX_m03113x with default bounds for boundary metabolite: m03113x.\n",
      "Adding exchange reaction EX_m03114x with default bounds for boundary metabolite: m03114x.\n",
      "Adding exchange reaction EX_m03117x with default bounds for boundary metabolite: m03117x.\n",
      "Adding exchange reaction EX_m03118x with default bounds for boundary metabolite: m03118x.\n",
      "Adding exchange reaction EX_m03120x with default bounds for boundary metabolite: m03120x.\n",
      "Adding exchange reaction EX_m03121x with default bounds for boundary metabolite: m03121x.\n",
      "Adding exchange reaction EX_m03123x with default bounds for boundary metabolite: m03123x.\n",
      "Adding exchange reaction EX_m03130x with default bounds for boundary metabolite: m03130x.\n",
      "Adding exchange reaction EX_m03131x with default bounds for boundary metabolite: m03131x.\n",
      "Adding exchange reaction EX_m03134x with default bounds for boundary metabolite: m03134x.\n",
      "Adding exchange reaction EX_m03135x with default bounds for boundary metabolite: m03135x.\n",
      "Adding exchange reaction EX_m03138x with default bounds for boundary metabolite: m03138x.\n",
      "Adding exchange reaction EX_m03141x with default bounds for boundary metabolite: m03141x.\n",
      "Adding exchange reaction EX_m03142x with default bounds for boundary metabolite: m03142x.\n",
      "Adding exchange reaction EX_m03146x with default bounds for boundary metabolite: m03146x.\n",
      "Adding exchange reaction EX_m03147x with default bounds for boundary metabolite: m03147x.\n",
      "Adding exchange reaction EX_m03151x with default bounds for boundary metabolite: m03151x.\n",
      "Adding exchange reaction EX_m03154x with default bounds for boundary metabolite: m03154x.\n",
      "Adding exchange reaction EX_m03155x with default bounds for boundary metabolite: m03155x.\n",
      "Adding exchange reaction EX_m03157x with default bounds for boundary metabolite: m03157x.\n",
      "Adding exchange reaction EX_temp001x with default bounds for boundary metabolite: temp001x.\n",
      "Adding exchange reaction EX_m10000x with default bounds for boundary metabolite: m10000x.\n",
      "Adding exchange reaction EX_m10001x with default bounds for boundary metabolite: m10001x.\n",
      "Adding exchange reaction EX_m10002x with default bounds for boundary metabolite: m10002x.\n",
      "Adding exchange reaction EX_m10003x with default bounds for boundary metabolite: m10003x.\n",
      "Adding exchange reaction EX_m01451x with default bounds for boundary metabolite: m01451x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_m00003x with default bounds for boundary metabolite: m00003x.\n",
      "Adding exchange reaction EX_m00008x with default bounds for boundary metabolite: m00008x.\n",
      "Adding exchange reaction EX_m00010x with default bounds for boundary metabolite: m00010x.\n",
      "Adding exchange reaction EX_m00017x with default bounds for boundary metabolite: m00017x.\n",
      "Adding exchange reaction EX_m00019x with default bounds for boundary metabolite: m00019x.\n",
      "Adding exchange reaction EX_m00021x with default bounds for boundary metabolite: m00021x.\n",
      "Adding exchange reaction EX_m00028x with default bounds for boundary metabolite: m00028x.\n",
      "Adding exchange reaction EX_m00094x with default bounds for boundary metabolite: m00094x.\n",
      "Adding exchange reaction EX_m00104x with default bounds for boundary metabolite: m00104x.\n",
      "Adding exchange reaction EX_m00105x with default bounds for boundary metabolite: m00105x.\n",
      "Adding exchange reaction EX_m00111x with default bounds for boundary metabolite: m00111x.\n",
      "Adding exchange reaction EX_m00114x with default bounds for boundary metabolite: m00114x.\n",
      "Adding exchange reaction EX_m00115x with default bounds for boundary metabolite: m00115x.\n",
      "Adding exchange reaction EX_m00117x with default bounds for boundary metabolite: m00117x.\n",
      "Adding exchange reaction EX_CE2516_x with default bounds for boundary metabolite: CE2516_x.\n",
      "Adding exchange reaction EX_m00126x with default bounds for boundary metabolite: m00126x.\n",
      "Adding exchange reaction EX_m00128x with default bounds for boundary metabolite: m00128x.\n",
      "Adding exchange reaction EX_m00132x with default bounds for boundary metabolite: m00132x.\n",
      "Adding exchange reaction EX_m00135x with default bounds for boundary metabolite: m00135x.\n",
      "Adding exchange reaction EX_m00167x with default bounds for boundary metabolite: m00167x.\n",
      "Adding exchange reaction EX_m00169x with default bounds for boundary metabolite: m00169x.\n",
      "Adding exchange reaction EX_glyc_S_x with default bounds for boundary metabolite: glyc_S_x.\n",
      "Adding exchange reaction EX_hexdtr_x with default bounds for boundary metabolite: hexdtr_x.\n",
      "Adding exchange reaction EX_1glyc_hs_x with default bounds for boundary metabolite: 1glyc_hs_x.\n",
      "Adding exchange reaction EX_1ohmdz_x with default bounds for boundary metabolite: 1ohmdz_x.\n",
      "Adding exchange reaction EX_1hmdgluc_x with default bounds for boundary metabolite: 1hmdgluc_x.\n",
      "Adding exchange reaction EX_12dgr120_x with default bounds for boundary metabolite: 12dgr120_x.\n",
      "Adding exchange reaction EX_m00231x with default bounds for boundary metabolite: m00231x.\n",
      "Adding exchange reaction EX_m00240x with default bounds for boundary metabolite: m00240x.\n",
      "Adding exchange reaction EX_m00241x with default bounds for boundary metabolite: m00241x.\n",
      "Adding exchange reaction EX_m00234x with default bounds for boundary metabolite: m00234x.\n",
      "Adding exchange reaction EX_m00248x with default bounds for boundary metabolite: m00248x.\n",
      "Adding exchange reaction EX_14hmdz_x with default bounds for boundary metabolite: 14hmdz_x.\n",
      "Adding exchange reaction EX_magarachi_hs_x with default bounds for boundary metabolite: magarachi_hs_x.\n",
      "Adding exchange reaction EX_pcholar_hs_x with default bounds for boundary metabolite: pcholar_hs_x.\n",
      "Adding exchange reaction EX_pear_hs_x with default bounds for boundary metabolite: pear_hs_x.\n",
      "Adding exchange reaction EX_pailar_hs_x with default bounds for boundary metabolite: pailar_hs_x.\n",
      "Adding exchange reaction EX_pe12_hs_x with default bounds for boundary metabolite: pe12_hs_x.\n",
      "Adding exchange reaction EX_pcholn203_hs_x with default bounds for boundary metabolite: pcholn203_hs_x.\n",
      "Adding exchange reaction EX_pedh203_hs_x with default bounds for boundary metabolite: pedh203_hs_x.\n",
      "Adding exchange reaction EX_pcholdoc_hs_x with default bounds for boundary metabolite: pcholdoc_hs_x.\n",
      "Adding exchange reaction EX_pe226_hs_x with default bounds for boundary metabolite: pe226_hs_x.\n",
      "Adding exchange reaction EX_pcholn226_hs_x with default bounds for boundary metabolite: pcholn226_hs_x.\n",
      "Adding exchange reaction EX_pcholn2254_hs_x with default bounds for boundary metabolite: pcholn2254_hs_x.\n",
      "Adding exchange reaction EX_pcholn225_hs_x with default bounds for boundary metabolite: pcholn225_hs_x.\n",
      "Adding exchange reaction EX_pe224_hs_x with default bounds for boundary metabolite: pe224_hs_x.\n",
      "Adding exchange reaction EX_pcholn224_hs_x with default bounds for boundary metabolite: pcholn224_hs_x.\n",
      "Adding exchange reaction EX_pcholeic_hs_x with default bounds for boundary metabolite: pcholeic_hs_x.\n",
      "Adding exchange reaction EX_pcholn205_hs_x with default bounds for boundary metabolite: pcholn205_hs_x.\n",
      "Adding exchange reaction EX_pcholn204_hs_x with default bounds for boundary metabolite: pcholn204_hs_x.\n",
      "Adding exchange reaction EX_pcholet_hs_x with default bounds for boundary metabolite: pcholet_hs_x.\n",
      "Adding exchange reaction EX_pe203_hs_x with default bounds for boundary metabolite: pe203_hs_x.\n",
      "Adding exchange reaction EX_pcholn201_hs_x with default bounds for boundary metabolite: pcholn201_hs_x.\n",
      "Adding exchange reaction EX_xolest183_hs_x with default bounds for boundary metabolite: xolest183_hs_x.\n",
      "Adding exchange reaction EX_pcholhep_hs_x with default bounds for boundary metabolite: pcholhep_hs_x.\n",
      "Adding exchange reaction EX_pe17_hs_x with default bounds for boundary metabolite: pe17_hs_x.\n",
      "Adding exchange reaction EX_pe161_hs_x with default bounds for boundary metabolite: pe161_hs_x.\n",
      "Adding exchange reaction EX_pcholn24_hs_x with default bounds for boundary metabolite: pcholn24_hs_x.\n",
      "Adding exchange reaction EX_xolest182_hs_x with default bounds for boundary metabolite: xolest182_hs_x.\n",
      "Adding exchange reaction EX_maglinl_hs_x with default bounds for boundary metabolite: maglinl_hs_x.\n",
      "Adding exchange reaction EX_pchollinl_hs_x with default bounds for boundary metabolite: pchollinl_hs_x.\n",
      "Adding exchange reaction EX_pelinl_hs_x with default bounds for boundary metabolite: pelinl_hs_x.\n",
      "Adding exchange reaction EX_pcholmyr_hs_x with default bounds for boundary metabolite: pcholmyr_hs_x.\n",
      "Adding exchange reaction EX_pe14_hs_x with default bounds for boundary metabolite: pe14_hs_x.\n",
      "Adding exchange reaction EX_pcholn19_hs_x with default bounds for boundary metabolite: pcholn19_hs_x.\n",
      "Adding exchange reaction EX_pcholn1836_hs_x with default bounds for boundary metabolite: pcholn1836_hs_x.\n",
      "Adding exchange reaction EX_pcholn183_hs_x with default bounds for boundary metabolite: pcholn183_hs_x.\n",
      "Adding exchange reaction EX_magole_hs_x with default bounds for boundary metabolite: magole_hs_x.\n",
      "Adding exchange reaction EX_pcholole_hs_x with default bounds for boundary metabolite: pcholole_hs_x.\n",
      "Adding exchange reaction EX_peole_hs_x with default bounds for boundary metabolite: peole_hs_x.\n",
      "Adding exchange reaction EX_pcholpalme_hs_x with default bounds for boundary metabolite: pcholpalme_hs_x.\n",
      "Adding exchange reaction EX_magpalm_hs_x with default bounds for boundary metabolite: magpalm_hs_x.\n",
      "Adding exchange reaction EX_pcholpalm_hs_x with default bounds for boundary metabolite: pcholpalm_hs_x.\n",
      "Adding exchange reaction EX_pepalm_hs_x with default bounds for boundary metabolite: pepalm_hs_x.\n",
      "Adding exchange reaction EX_pailpalm_hs_x with default bounds for boundary metabolite: pailpalm_hs_x.\n",
      "Adding exchange reaction EX_pcholn15_hs_x with default bounds for boundary metabolite: pcholn15_hs_x.\n",
      "Adding exchange reaction EX_pe15_hs_x with default bounds for boundary metabolite: pe15_hs_x.\n",
      "Adding exchange reaction EX_magste_hs_x with default bounds for boundary metabolite: magste_hs_x.\n",
      "Adding exchange reaction EX_pcholste_hs_x with default bounds for boundary metabolite: pcholste_hs_x.\n",
      "Adding exchange reaction EX_peste_hs_x with default bounds for boundary metabolite: peste_hs_x.\n",
      "Adding exchange reaction EX_pailste_hs_x with default bounds for boundary metabolite: pailste_hs_x.\n",
      "Adding exchange reaction EX_xolest205_hs_x with default bounds for boundary metabolite: xolest205_hs_x.\n",
      "Adding exchange reaction EX_pe13_hs_x with default bounds for boundary metabolite: pe13_hs_x.\n",
      "Adding exchange reaction EX_xolest181_hs_x with default bounds for boundary metabolite: xolest181_hs_x.\n",
      "Adding exchange reaction EX_m00510x with default bounds for boundary metabolite: m00510x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_m00503x with default bounds for boundary metabolite: m00503x.\n",
      "Adding exchange reaction EX_m00519x with default bounds for boundary metabolite: m00519x.\n",
      "Adding exchange reaction EX_1hibup_S_x with default bounds for boundary metabolite: 1hibup_S_x.\n",
      "Adding exchange reaction EX_1hibupglu_S_x with default bounds for boundary metabolite: 1hibupglu_S_x.\n",
      "Adding exchange reaction EX_m00260x with default bounds for boundary metabolite: m00260x.\n",
      "Adding exchange reaction EX_m00265x with default bounds for boundary metabolite: m00265x.\n",
      "Adding exchange reaction EX_m00270x with default bounds for boundary metabolite: m00270x.\n",
      "Adding exchange reaction EX_m00279x with default bounds for boundary metabolite: m00279x.\n",
      "Adding exchange reaction EX_m00291x with default bounds for boundary metabolite: m00291x.\n",
      "Adding exchange reaction EX_m00293x with default bounds for boundary metabolite: m00293x.\n",
      "Adding exchange reaction EX_m00294x with default bounds for boundary metabolite: m00294x.\n",
      "Adding exchange reaction EX_m00295x with default bounds for boundary metabolite: m00295x.\n",
      "Adding exchange reaction EX_m00305x with default bounds for boundary metabolite: m00305x.\n",
      "Adding exchange reaction EX_m00308x with default bounds for boundary metabolite: m00308x.\n",
      "Adding exchange reaction EX_m00309x with default bounds for boundary metabolite: m00309x.\n",
      "Adding exchange reaction EX_m00314x with default bounds for boundary metabolite: m00314x.\n",
      "Adding exchange reaction EX_m00315x with default bounds for boundary metabolite: m00315x.\n",
      "Adding exchange reaction EX_12dhchol_x with default bounds for boundary metabolite: 12dhchol_x.\n",
      "Adding exchange reaction EX_12htacr_x with default bounds for boundary metabolite: 12htacr_x.\n",
      "Adding exchange reaction EX_m00324x with default bounds for boundary metabolite: m00324x.\n",
      "Adding exchange reaction EX_m00337x with default bounds for boundary metabolite: m00337x.\n",
      "Adding exchange reaction EX_1513tacr_x with default bounds for boundary metabolite: 1513tacr_x.\n",
      "Adding exchange reaction EX_m00341x with default bounds for boundary metabolite: m00341x.\n",
      "Adding exchange reaction EX_1331tacr_x with default bounds for boundary metabolite: 1331tacr_x.\n",
      "Adding exchange reaction EX_13dmt_x with default bounds for boundary metabolite: 13dmt_x.\n",
      "Adding exchange reaction EX_m00350x with default bounds for boundary metabolite: m00350x.\n",
      "Adding exchange reaction EX_m00365x with default bounds for boundary metabolite: m00365x.\n",
      "Adding exchange reaction EX_m00366x with default bounds for boundary metabolite: m00366x.\n",
      "Adding exchange reaction EX_m00373x with default bounds for boundary metabolite: m00373x.\n",
      "Adding exchange reaction EX_m00376x with default bounds for boundary metabolite: m00376x.\n",
      "Adding exchange reaction EX_m00378x with default bounds for boundary metabolite: m00378x.\n",
      "Adding exchange reaction EX_m00380x with default bounds for boundary metabolite: m00380x.\n",
      "Adding exchange reaction EX_1531tacr_x with default bounds for boundary metabolite: 1531tacr_x.\n",
      "Adding exchange reaction EX_15dmt_x with default bounds for boundary metabolite: 15dmt_x.\n",
      "Adding exchange reaction EX_15kprostgf2_x with default bounds for boundary metabolite: 15kprostgf2_x.\n",
      "Adding exchange reaction EX_m00384x with default bounds for boundary metabolite: m00384x.\n",
      "Adding exchange reaction EX_m00385x with default bounds for boundary metabolite: m00385x.\n",
      "Adding exchange reaction EX_m00399x with default bounds for boundary metabolite: m00399x.\n",
      "Adding exchange reaction EX_m00400x with default bounds for boundary metabolite: m00400x.\n",
      "Adding exchange reaction EX_m00408x with default bounds for boundary metabolite: m00408x.\n",
      "Adding exchange reaction EX_m00407x with default bounds for boundary metabolite: m00407x.\n",
      "Adding exchange reaction EX_m00409x with default bounds for boundary metabolite: m00409x.\n",
      "Adding exchange reaction EX_m00428x with default bounds for boundary metabolite: m00428x.\n",
      "Adding exchange reaction EX_and19one_x with default bounds for boundary metabolite: and19one_x.\n",
      "Adding exchange reaction EX_m00570x with default bounds for boundary metabolite: m00570x.\n",
      "Adding exchange reaction EX_m00576x with default bounds for boundary metabolite: m00576x.\n",
      "Adding exchange reaction EX_2h3mv_x with default bounds for boundary metabolite: 2h3mv_x.\n",
      "Adding exchange reaction EX_2hydog_x with default bounds for boundary metabolite: 2hydog_x.\n",
      "Adding exchange reaction EX_2hxic_L_x with default bounds for boundary metabolite: 2hxic_L_x.\n",
      "Adding exchange reaction EX_2hiv_x with default bounds for boundary metabolite: 2hiv_x.\n",
      "Adding exchange reaction EX_adpoh_x with default bounds for boundary metabolite: adpoh_x.\n",
      "Adding exchange reaction EX_pchol2linl_hs_x with default bounds for boundary metabolite: pchol2linl_hs_x.\n",
      "Adding exchange reaction EX_pe2linl_hs_x with default bounds for boundary metabolite: pe2linl_hs_x.\n",
      "Adding exchange reaction EX_2m3hbu_x with default bounds for boundary metabolite: 2m3hbu_x.\n",
      "Adding exchange reaction EX_2m3hvac_x with default bounds for boundary metabolite: 2m3hvac_x.\n",
      "Adding exchange reaction EX_pchol2ole_hs_x with default bounds for boundary metabolite: pchol2ole_hs_x.\n",
      "Adding exchange reaction EX_pchol2palm_hs_x with default bounds for boundary metabolite: pchol2palm_hs_x.\n",
      "Adding exchange reaction EX_pchol2ste_hs_x with default bounds for boundary metabolite: pchol2ste_hs_x.\n",
      "Adding exchange reaction EX_m00635x with default bounds for boundary metabolite: m00635x.\n",
      "Adding exchange reaction EX_2hibupglu_S_x with default bounds for boundary metabolite: 2hibupglu_S_x.\n",
      "Adding exchange reaction EX_2hibup_R_x with default bounds for boundary metabolite: 2hibup_R_x.\n",
      "Adding exchange reaction EX_2hibup_S_x with default bounds for boundary metabolite: 2hibup_S_x.\n",
      "Adding exchange reaction EX_2hatvacid_x with default bounds for boundary metabolite: 2hatvacid_x.\n",
      "Adding exchange reaction EX_2hatvacidgluc_x with default bounds for boundary metabolite: 2hatvacidgluc_x.\n",
      "Adding exchange reaction EX_2hatvlac_x with default bounds for boundary metabolite: 2hatvlac_x.\n",
      "Adding exchange reaction EX_2hatvlacgluc_x with default bounds for boundary metabolite: 2hatvlacgluc_x.\n",
      "Adding exchange reaction EX_m00649x with default bounds for boundary metabolite: m00649x.\n",
      "Adding exchange reaction EX_m00650x with default bounds for boundary metabolite: m00650x.\n",
      "Adding exchange reaction EX_m00654x with default bounds for boundary metabolite: m00654x.\n",
      "Adding exchange reaction EX_m00656x with default bounds for boundary metabolite: m00656x.\n",
      "Adding exchange reaction EX_m00659x with default bounds for boundary metabolite: m00659x.\n",
      "Adding exchange reaction EX_m00660x with default bounds for boundary metabolite: m00660x.\n",
      "Adding exchange reaction EX_m00664x with default bounds for boundary metabolite: m00664x.\n",
      "Adding exchange reaction EX_m00669x with default bounds for boundary metabolite: m00669x.\n",
      "Adding exchange reaction EX_m00670x with default bounds for boundary metabolite: m00670x.\n",
      "Adding exchange reaction EX_m00671x with default bounds for boundary metabolite: m00671x.\n",
      "Adding exchange reaction EX_m00674x with default bounds for boundary metabolite: m00674x.\n",
      "Adding exchange reaction EX_m00585x with default bounds for boundary metabolite: m00585x.\n",
      "Adding exchange reaction EX_m00599x with default bounds for boundary metabolite: m00599x.\n",
      "Adding exchange reaction EX_21hprgnlone_x with default bounds for boundary metabolite: 21hprgnlone_x.\n",
      "Adding exchange reaction EX_m00604x with default bounds for boundary metabolite: m00604x.\n",
      "Adding exchange reaction EX_m00610x with default bounds for boundary metabolite: m00610x.\n",
      "Adding exchange reaction EX_m00619x with default bounds for boundary metabolite: m00619x.\n",
      "Adding exchange reaction EX_m00623x with default bounds for boundary metabolite: m00623x.\n",
      "Adding exchange reaction EX_3hlvstacid_x with default bounds for boundary metabolite: 3hlvstacid_x.\n",
      "Adding exchange reaction EX_35dsmv_x with default bounds for boundary metabolite: 35dsmv_x.\n",
      "Adding exchange reaction EX_3hpvs_x with default bounds for boundary metabolite: 3hpvs_x.\n",
      "Adding exchange reaction EX_3hpvstet_x with default bounds for boundary metabolite: 3hpvstet_x.\n",
      "Adding exchange reaction EX_35dhpvs_x with default bounds for boundary metabolite: 35dhpvs_x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_isolvstacid_x with default bounds for boundary metabolite: isolvstacid_x.\n",
      "Adding exchange reaction EX_3hsmvacid_x with default bounds for boundary metabolite: 3hsmvacid_x.\n",
      "Adding exchange reaction EX_34dhpe_x with default bounds for boundary metabolite: 34dhpe_x.\n",
      "Adding exchange reaction EX_m00727x with default bounds for boundary metabolite: m00727x.\n",
      "Adding exchange reaction EX_m00729x with default bounds for boundary metabolite: m00729x.\n",
      "Adding exchange reaction EX_m00739x with default bounds for boundary metabolite: m00739x.\n",
      "Adding exchange reaction EX_3hpppn_x with default bounds for boundary metabolite: 3hpppn_x.\n",
      "Adding exchange reaction EX_3hpppnohgluc_x with default bounds for boundary metabolite: 3hpppnohgluc_x.\n",
      "Adding exchange reaction EX_3dhcdchol_x with default bounds for boundary metabolite: 3dhcdchol_x.\n",
      "Adding exchange reaction EX_3dhchol_x with default bounds for boundary metabolite: 3dhchol_x.\n",
      "Adding exchange reaction EX_3ohsubac_x with default bounds for boundary metabolite: 3ohsubac_x.\n",
      "Adding exchange reaction EX_3bcrn_x with default bounds for boundary metabolite: 3bcrn_x.\n",
      "Adding exchange reaction EX_3tetd7ecoacrn_x with default bounds for boundary metabolite: 3tetd7ecoacrn_x.\n",
      "Adding exchange reaction EX_3ttetddcoacrn_x with default bounds for boundary metabolite: 3ttetddcoacrn_x.\n",
      "Adding exchange reaction EX_3thexddcoacrn_x with default bounds for boundary metabolite: 3thexddcoacrn_x.\n",
      "Adding exchange reaction EX_3h3mglt_x with default bounds for boundary metabolite: 3h3mglt_x.\n",
      "Adding exchange reaction EX_3hadpac_x with default bounds for boundary metabolite: 3hadpac_x.\n",
      "Adding exchange reaction EX_3ohglutac_x with default bounds for boundary metabolite: 3ohglutac_x.\n",
      "Adding exchange reaction EX_3ivcrn_x with default bounds for boundary metabolite: 3ivcrn_x.\n",
      "Adding exchange reaction EX_3octdece1crn_x with default bounds for boundary metabolite: 3octdece1crn_x.\n",
      "Adding exchange reaction EX_3ohsebac_x with default bounds for boundary metabolite: 3ohsebac_x.\n",
      "Adding exchange reaction EX_3tdcrn_x with default bounds for boundary metabolite: 3tdcrn_x.\n",
      "Adding exchange reaction EX_3hcinnm_x with default bounds for boundary metabolite: 3hcinnm_x.\n",
      "Adding exchange reaction EX_3deccrn_x with default bounds for boundary metabolite: 3deccrn_x.\n",
      "Adding exchange reaction EX_3ddcrn_x with default bounds for boundary metabolite: 3ddcrn_x.\n",
      "Adding exchange reaction EX_3hexdcrn_x with default bounds for boundary metabolite: 3hexdcrn_x.\n",
      "Adding exchange reaction EX_3hdececrn_x with default bounds for boundary metabolite: 3hdececrn_x.\n",
      "Adding exchange reaction EX_3hivac_x with default bounds for boundary metabolite: 3hivac_x.\n",
      "Adding exchange reaction EX_3octdec2crn_x with default bounds for boundary metabolite: 3octdec2crn_x.\n",
      "Adding exchange reaction EX_3octdeccrn_x with default bounds for boundary metabolite: 3octdeccrn_x.\n",
      "Adding exchange reaction EX_3hppa_x with default bounds for boundary metabolite: 3hppa_x.\n",
      "Adding exchange reaction EX_3mglutac_x with default bounds for boundary metabolite: 3mglutac_x.\n",
      "Adding exchange reaction EX_3mglutr_x with default bounds for boundary metabolite: 3mglutr_x.\n",
      "Adding exchange reaction EX_3mtp_x with default bounds for boundary metabolite: 3mtp_x.\n",
      "Adding exchange reaction EX_3mhis_x with default bounds for boundary metabolite: 3mhis_x.\n",
      "Adding exchange reaction EX_m00830x with default bounds for boundary metabolite: m00830x.\n",
      "Adding exchange reaction EX_m00921x with default bounds for boundary metabolite: m00921x.\n",
      "Adding exchange reaction EX_3ispvs_x with default bounds for boundary metabolite: 3ispvs_x.\n",
      "Adding exchange reaction EX_m00766x with default bounds for boundary metabolite: m00766x.\n",
      "Adding exchange reaction EX_3dhdchol_x with default bounds for boundary metabolite: 3dhdchol_x.\n",
      "Adding exchange reaction EX_3dhlchol_x with default bounds for boundary metabolite: 3dhlchol_x.\n",
      "Adding exchange reaction EX_3hibup_R_x with default bounds for boundary metabolite: 3hibup_R_x.\n",
      "Adding exchange reaction EX_3hibup_S_x with default bounds for boundary metabolite: 3hibup_S_x.\n",
      "Adding exchange reaction EX_3hibupglu_S_x with default bounds for boundary metabolite: 3hibupglu_S_x.\n",
      "Adding exchange reaction EX_m00788x with default bounds for boundary metabolite: m00788x.\n",
      "Adding exchange reaction EX_3ohacmp_x with default bounds for boundary metabolite: 3ohacmp_x.\n",
      "Adding exchange reaction EX_m00775x with default bounds for boundary metabolite: m00775x.\n",
      "Adding exchange reaction EX_m00784x with default bounds for boundary metabolite: m00784x.\n",
      "Adding exchange reaction EX_m00807x with default bounds for boundary metabolite: m00807x.\n",
      "Adding exchange reaction EX_56dhpvs_x with default bounds for boundary metabolite: 56dhpvs_x.\n",
      "Adding exchange reaction EX_m00810x with default bounds for boundary metabolite: m00810x.\n",
      "Adding exchange reaction EX_m00812x with default bounds for boundary metabolite: m00812x.\n",
      "Adding exchange reaction EX_m00821x with default bounds for boundary metabolite: m00821x.\n",
      "Adding exchange reaction EX_m00824x with default bounds for boundary metabolite: m00824x.\n",
      "Adding exchange reaction EX_m00825x with default bounds for boundary metabolite: m00825x.\n",
      "Adding exchange reaction EX_m00866x with default bounds for boundary metabolite: m00866x.\n",
      "Adding exchange reaction EX_m00913x with default bounds for boundary metabolite: m00913x.\n",
      "Adding exchange reaction EX_m00916x with default bounds for boundary metabolite: m00916x.\n",
      "Adding exchange reaction EX_m00922x with default bounds for boundary metabolite: m00922x.\n",
      "Adding exchange reaction EX_m00923x with default bounds for boundary metabolite: m00923x.\n",
      "Adding exchange reaction EX_31dmt_x with default bounds for boundary metabolite: 31dmt_x.\n",
      "Adding exchange reaction EX_epoxtac_x with default bounds for boundary metabolite: epoxtac_x.\n",
      "Adding exchange reaction EX_m00745x with default bounds for boundary metabolite: m00745x.\n",
      "Adding exchange reaction EX_m00752x with default bounds for boundary metabolite: m00752x.\n",
      "Adding exchange reaction EX_m00758x with default bounds for boundary metabolite: m00758x.\n",
      "Adding exchange reaction EX_4ohbut_x with default bounds for boundary metabolite: 4ohbut_x.\n",
      "Adding exchange reaction EX_4ohmdz_x with default bounds for boundary metabolite: 4ohmdz_x.\n",
      "Adding exchange reaction EX_4hmdgluc_x with default bounds for boundary metabolite: 4hmdgluc_x.\n",
      "Adding exchange reaction EX_4glu56dihdind_x with default bounds for boundary metabolite: 4glu56dihdind_x.\n",
      "Adding exchange reaction EX_m00952x with default bounds for boundary metabolite: m00952x.\n",
      "Adding exchange reaction EX_m00969x with default bounds for boundary metabolite: m00969x.\n",
      "Adding exchange reaction EX_m00971x with default bounds for boundary metabolite: m00971x.\n",
      "Adding exchange reaction EX_4bhglz_x with default bounds for boundary metabolite: 4bhglz_x.\n",
      "Adding exchange reaction EX_m00981x with default bounds for boundary metabolite: m00981x.\n",
      "Adding exchange reaction EX_m00988x with default bounds for boundary metabolite: m00988x.\n",
      "Adding exchange reaction EX_m00990x with default bounds for boundary metabolite: m00990x.\n",
      "Adding exchange reaction EX_4hatvacid_x with default bounds for boundary metabolite: 4hatvacid_x.\n",
      "Adding exchange reaction EX_4hatvlac_x with default bounds for boundary metabolite: 4hatvlac_x.\n",
      "Adding exchange reaction EX_m00995x with default bounds for boundary metabolite: m00995x.\n",
      "Adding exchange reaction EX_m01004x with default bounds for boundary metabolite: m01004x.\n",
      "Adding exchange reaction EX_m01005x with default bounds for boundary metabolite: m01005x.\n",
      "Adding exchange reaction EX_m01013x with default bounds for boundary metabolite: m01013x.\n",
      "Adding exchange reaction EX_m01039x with default bounds for boundary metabolite: m01039x.\n",
      "Adding exchange reaction EX_m01040x with default bounds for boundary metabolite: m01040x.\n",
      "Adding exchange reaction EX_m01042x with default bounds for boundary metabolite: m01042x.\n",
      "Adding exchange reaction EX_m01045x with default bounds for boundary metabolite: m01045x.\n",
      "Adding exchange reaction EX_m01047x with default bounds for boundary metabolite: m01047x.\n",
      "Adding exchange reaction EX_m01050x with default bounds for boundary metabolite: m01050x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_m01052x with default bounds for boundary metabolite: m01052x.\n",
      "Adding exchange reaction EX_m01054x with default bounds for boundary metabolite: m01054x.\n",
      "Adding exchange reaction EX_56eppvs_x with default bounds for boundary metabolite: 56eppvs_x.\n",
      "Adding exchange reaction EX_5eipenc_x with default bounds for boundary metabolite: 5eipenc_x.\n",
      "Adding exchange reaction EX_5ohhexa_x with default bounds for boundary metabolite: 5ohhexa_x.\n",
      "Adding exchange reaction EX_5cysdopa_x with default bounds for boundary metabolite: 5cysdopa_x.\n",
      "Adding exchange reaction EX_5cysgly34dhphe_x with default bounds for boundary metabolite: 5cysgly34dhphe_x.\n",
      "Adding exchange reaction EX_m01134x with default bounds for boundary metabolite: m01134x.\n",
      "Adding exchange reaction EX_m01139x with default bounds for boundary metabolite: m01139x.\n",
      "Adding exchange reaction EX_m01138x with default bounds for boundary metabolite: m01138x.\n",
      "Adding exchange reaction EX_m01073x with default bounds for boundary metabolite: m01073x.\n",
      "Adding exchange reaction EX_m01074x with default bounds for boundary metabolite: m01074x.\n",
      "Adding exchange reaction EX_m01101x with default bounds for boundary metabolite: m01101x.\n",
      "Adding exchange reaction EX_5ohfvs_x with default bounds for boundary metabolite: 5ohfvs_x.\n",
      "Adding exchange reaction EX_5ohfvsglu_x with default bounds for boundary metabolite: 5ohfvsglu_x.\n",
      "Adding exchange reaction EX_m01103x with default bounds for boundary metabolite: m01103x.\n",
      "Adding exchange reaction EX_m01110x with default bounds for boundary metabolite: m01110x.\n",
      "Adding exchange reaction EX_m01113x with default bounds for boundary metabolite: m01113x.\n",
      "Adding exchange reaction EX_m01116x with default bounds for boundary metabolite: m01116x.\n",
      "Adding exchange reaction EX_m01127x with default bounds for boundary metabolite: m01127x.\n",
      "Adding exchange reaction EX_m01064x with default bounds for boundary metabolite: m01064x.\n",
      "Adding exchange reaction EX_m01065x with default bounds for boundary metabolite: m01065x.\n",
      "Adding exchange reaction EX_m01072x with default bounds for boundary metabolite: m01072x.\n",
      "Adding exchange reaction EX_m01087x with default bounds for boundary metabolite: m01087x.\n",
      "Adding exchange reaction EX_m01093x with default bounds for boundary metabolite: m01093x.\n",
      "Adding exchange reaction EX_6csmvacid_x with default bounds for boundary metabolite: 6csmvacid_x.\n",
      "Adding exchange reaction EX_6hsmvacid_x with default bounds for boundary metabolite: 6hsmvacid_x.\n",
      "Adding exchange reaction EX_6hlvst_x with default bounds for boundary metabolite: 6hlvst_x.\n",
      "Adding exchange reaction EX_6hmsmvacid_x with default bounds for boundary metabolite: 6hmsmvacid_x.\n",
      "Adding exchange reaction EX_6melvst_x with default bounds for boundary metabolite: 6melvst_x.\n",
      "Adding exchange reaction EX_6melvacid_x with default bounds for boundary metabolite: 6melvacid_x.\n",
      "Adding exchange reaction EX_m01154x with default bounds for boundary metabolite: m01154x.\n",
      "Adding exchange reaction EX_m01155x with default bounds for boundary metabolite: m01155x.\n",
      "Adding exchange reaction EX_6ahglz_x with default bounds for boundary metabolite: 6ahglz_x.\n",
      "Adding exchange reaction EX_6bhglz_x with default bounds for boundary metabolite: 6bhglz_x.\n",
      "Adding exchange reaction EX_6bhglzglc_x with default bounds for boundary metabolite: 6bhglzglc_x.\n",
      "Adding exchange reaction EX_6epvs_x with default bounds for boundary metabolite: 6epvs_x.\n",
      "Adding exchange reaction EX_6ohfvs_x with default bounds for boundary metabolite: 6ohfvs_x.\n",
      "Adding exchange reaction EX_6ohfvsglu_x with default bounds for boundary metabolite: 6ohfvsglu_x.\n",
      "Adding exchange reaction EX_m01161x with default bounds for boundary metabolite: m01161x.\n",
      "Adding exchange reaction EX_m01168x with default bounds for boundary metabolite: m01168x.\n",
      "Adding exchange reaction EX_m01172x with default bounds for boundary metabolite: m01172x.\n",
      "Adding exchange reaction EX_7dhcdchol_x with default bounds for boundary metabolite: 7dhcdchol_x.\n",
      "Adding exchange reaction EX_7dhchol_x with default bounds for boundary metabolite: 7dhchol_x.\n",
      "Adding exchange reaction EX_7ohocata_x with default bounds for boundary metabolite: 7ohocata_x.\n",
      "Adding exchange reaction EX_7klitchol_x with default bounds for boundary metabolite: 7klitchol_x.\n",
      "Adding exchange reaction EX_7ahglz_x with default bounds for boundary metabolite: 7ahglz_x.\n",
      "Adding exchange reaction EX_7bhglz_x with default bounds for boundary metabolite: 7bhglz_x.\n",
      "Adding exchange reaction EX_7bhglzglc_x with default bounds for boundary metabolite: 7bhglzglc_x.\n",
      "Adding exchange reaction EX_7hpvs_x with default bounds for boundary metabolite: 7hpvs_x.\n",
      "Adding exchange reaction EX_m01197x with default bounds for boundary metabolite: m01197x.\n",
      "Adding exchange reaction EX_thexdd_x with default bounds for boundary metabolite: thexdd_x.\n",
      "Adding exchange reaction EX_m01178x with default bounds for boundary metabolite: m01178x.\n",
      "Adding exchange reaction EX_m01182x with default bounds for boundary metabolite: m01182x.\n",
      "Adding exchange reaction EX_m01207x with default bounds for boundary metabolite: m01207x.\n",
      "Adding exchange reaction EX_m01209x with default bounds for boundary metabolite: m01209x.\n",
      "Adding exchange reaction EX_CE1297_x with default bounds for boundary metabolite: CE1297_x.\n",
      "Adding exchange reaction EX_m01216x with default bounds for boundary metabolite: m01216x.\n",
      "Adding exchange reaction EX_m01220x with default bounds for boundary metabolite: m01220x.\n",
      "Adding exchange reaction EX_m01231x with default bounds for boundary metabolite: m01231x.\n",
      "Adding exchange reaction EX_m01232x with default bounds for boundary metabolite: m01232x.\n",
      "Adding exchange reaction EX_m01235x with default bounds for boundary metabolite: m01235x.\n",
      "Adding exchange reaction EX_m01238x with default bounds for boundary metabolite: m01238x.\n",
      "Adding exchange reaction EX_m01304x with default bounds for boundary metabolite: m01304x.\n",
      "Adding exchange reaction EX_am1csa_x with default bounds for boundary metabolite: am1csa_x.\n",
      "Adding exchange reaction EX_am19cs_x with default bounds for boundary metabolite: am19cs_x.\n",
      "Adding exchange reaction EX_am1acs_x with default bounds for boundary metabolite: am1acs_x.\n",
      "Adding exchange reaction EX_am1a4ncs_x with default bounds for boundary metabolite: am1a4ncs_x.\n",
      "Adding exchange reaction EX_am1alcs_x with default bounds for boundary metabolite: am1alcs_x.\n",
      "Adding exchange reaction EX_am1accs_x with default bounds for boundary metabolite: am1accs_x.\n",
      "Adding exchange reaction EX_am1ccs_x with default bounds for boundary metabolite: am1ccs_x.\n",
      "Adding exchange reaction EX_am1cglc_x with default bounds for boundary metabolite: am1cglc_x.\n",
      "Adding exchange reaction EX_am1c4n9cs_x with default bounds for boundary metabolite: am1c4n9cs_x.\n",
      "Adding exchange reaction EX_am1c9cs_x with default bounds for boundary metabolite: am1c9cs_x.\n",
      "Adding exchange reaction EX_am4ncs_x with default bounds for boundary metabolite: am4ncs_x.\n",
      "Adding exchange reaction EX_am4n9cs_x with default bounds for boundary metabolite: am4n9cs_x.\n",
      "Adding exchange reaction EX_am9csa_x with default bounds for boundary metabolite: am9csa_x.\n",
      "Adding exchange reaction EX_m01371x with default bounds for boundary metabolite: m01371x.\n",
      "Adding exchange reaction EX_acmpglu_x with default bounds for boundary metabolite: acmpglu_x.\n",
      "Adding exchange reaction EX_acile_L_x with default bounds for boundary metabolite: acile_L_x.\n",
      "Adding exchange reaction EX_acgly_x with default bounds for boundary metabolite: acgly_x.\n",
      "Adding exchange reaction EX_aclys_x with default bounds for boundary metabolite: aclys_x.\n",
      "Adding exchange reaction EX_acthr_L_x with default bounds for boundary metabolite: acthr_L_x.\n",
      "Adding exchange reaction EX_achom_L_x with default bounds for boundary metabolite: achom_L_x.\n",
      "Adding exchange reaction EX_adpac_x with default bounds for boundary metabolite: adpac_x.\n",
      "Adding exchange reaction EX_c6dc_x with default bounds for boundary metabolite: c6dc_x.\n",
      "Adding exchange reaction EX_alaargcys_x with default bounds for boundary metabolite: alaargcys_x.\n",
      "Adding exchange reaction EX_alaarggly_x with default bounds for boundary metabolite: alaarggly_x.\n",
      "Adding exchange reaction EX_alaasnleu_x with default bounds for boundary metabolite: alaasnleu_x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_alaglylys_x with default bounds for boundary metabolite: alaglylys_x.\n",
      "Adding exchange reaction EX_alahisala_x with default bounds for boundary metabolite: alahisala_x.\n",
      "Adding exchange reaction EX_alalysthr_x with default bounds for boundary metabolite: alalysthr_x.\n",
      "Adding exchange reaction EX_argalaphe_x with default bounds for boundary metabolite: argalaphe_x.\n",
      "Adding exchange reaction EX_argalaala_x with default bounds for boundary metabolite: argalaala_x.\n",
      "Adding exchange reaction EX_argalathr_x with default bounds for boundary metabolite: argalathr_x.\n",
      "Adding exchange reaction EX_argarg_x with default bounds for boundary metabolite: argarg_x.\n",
      "Adding exchange reaction EX_argarglys_x with default bounds for boundary metabolite: argarglys_x.\n",
      "Adding exchange reaction EX_argargmet_x with default bounds for boundary metabolite: argargmet_x.\n",
      "Adding exchange reaction EX_argcysgly_x with default bounds for boundary metabolite: argcysgly_x.\n",
      "Adding exchange reaction EX_argcysser_x with default bounds for boundary metabolite: argcysser_x.\n",
      "Adding exchange reaction EX_arggluglu_x with default bounds for boundary metabolite: arggluglu_x.\n",
      "Adding exchange reaction EX_argglupro_x with default bounds for boundary metabolite: argglupro_x.\n",
      "Adding exchange reaction EX_argglygly_x with default bounds for boundary metabolite: argglygly_x.\n",
      "Adding exchange reaction EX_arghisthr_x with default bounds for boundary metabolite: arghisthr_x.\n",
      "Adding exchange reaction EX_argleuphe_x with default bounds for boundary metabolite: argleuphe_x.\n",
      "Adding exchange reaction EX_arglysasp_x with default bounds for boundary metabolite: arglysasp_x.\n",
      "Adding exchange reaction EX_argphearg_x with default bounds for boundary metabolite: argphearg_x.\n",
      "Adding exchange reaction EX_argpromet_x with default bounds for boundary metabolite: argpromet_x.\n",
      "Adding exchange reaction EX_argprothr_x with default bounds for boundary metabolite: argprothr_x.\n",
      "Adding exchange reaction EX_argserser_x with default bounds for boundary metabolite: argserser_x.\n",
      "Adding exchange reaction EX_argvalcys_x with default bounds for boundary metabolite: argvalcys_x.\n",
      "Adding exchange reaction EX_argvaltrp_x with default bounds for boundary metabolite: argvaltrp_x.\n",
      "Adding exchange reaction EX_argtyrval_x with default bounds for boundary metabolite: argtyrval_x.\n",
      "Adding exchange reaction EX_aspalaarg_x with default bounds for boundary metabolite: aspalaarg_x.\n",
      "Adding exchange reaction EX_asnasnarg_x with default bounds for boundary metabolite: asnasnarg_x.\n",
      "Adding exchange reaction EX_asncyscys_x with default bounds for boundary metabolite: asncyscys_x.\n",
      "Adding exchange reaction EX_asnmetpro_x with default bounds for boundary metabolite: asnmetpro_x.\n",
      "Adding exchange reaction EX_asnpheasp_x with default bounds for boundary metabolite: asnpheasp_x.\n",
      "Adding exchange reaction EX_asnphecys_x with default bounds for boundary metabolite: asnphecys_x.\n",
      "Adding exchange reaction EX_asntyrgly_x with default bounds for boundary metabolite: asntyrgly_x.\n",
      "Adding exchange reaction EX_asntyrphe_x with default bounds for boundary metabolite: asntyrphe_x.\n",
      "Adding exchange reaction EX_asntyrthr_x with default bounds for boundary metabolite: asntyrthr_x.\n",
      "Adding exchange reaction EX_aspasnglu_x with default bounds for boundary metabolite: aspasnglu_x.\n",
      "Adding exchange reaction EX_aspglu_x with default bounds for boundary metabolite: aspglu_x.\n",
      "Adding exchange reaction EX_aspglupro_x with default bounds for boundary metabolite: aspglupro_x.\n",
      "Adding exchange reaction EX_aspglutrp_x with default bounds for boundary metabolite: aspglutrp_x.\n",
      "Adding exchange reaction EX_asphiscys_x with default bounds for boundary metabolite: asphiscys_x.\n",
      "Adding exchange reaction EX_asphispro_x with default bounds for boundary metabolite: asphispro_x.\n",
      "Adding exchange reaction EX_asplysglu_x with default bounds for boundary metabolite: asplysglu_x.\n",
      "Adding exchange reaction EX_asplyshis_x with default bounds for boundary metabolite: asplyshis_x.\n",
      "Adding exchange reaction EX_aspmetasp_x with default bounds for boundary metabolite: aspmetasp_x.\n",
      "Adding exchange reaction EX_aspprolys_x with default bounds for boundary metabolite: aspprolys_x.\n",
      "Adding exchange reaction EX_aspvalasn_x with default bounds for boundary metabolite: aspvalasn_x.\n",
      "Adding exchange reaction EX_glgchlo_x with default bounds for boundary metabolite: glgchlo_x.\n",
      "Adding exchange reaction EX_gltcho_x with default bounds for boundary metabolite: gltcho_x.\n",
      "Adding exchange reaction EX_gltdechol_x with default bounds for boundary metabolite: gltdechol_x.\n",
      "Adding exchange reaction EX_bglc_x with default bounds for boundary metabolite: bglc_x.\n",
      "Adding exchange reaction EX_bvite_x with default bounds for boundary metabolite: bvite_x.\n",
      "Adding exchange reaction EX_didecaeth_x with default bounds for boundary metabolite: didecaeth_x.\n",
      "Adding exchange reaction EX_tetdecaeth_x with default bounds for boundary metabolite: tetdecaeth_x.\n",
      "Adding exchange reaction EX_m01424x with default bounds for boundary metabolite: m01424x.\n",
      "Adding exchange reaction EX_m01428x with default bounds for boundary metabolite: m01428x.\n",
      "Adding exchange reaction EX_m01592x with default bounds for boundary metabolite: m01592x.\n",
      "Adding exchange reaction EX_m01623x with default bounds for boundary metabolite: m01623x.\n",
      "Adding exchange reaction EX_C02528_x with default bounds for boundary metabolite: C02528_x.\n",
      "Adding exchange reaction EX_cdca24g_x with default bounds for boundary metabolite: cdca24g_x.\n",
      "Adding exchange reaction EX_cdca3g_x with default bounds for boundary metabolite: cdca3g_x.\n",
      "Adding exchange reaction EX_xolest_hs_x with default bounds for boundary metabolite: xolest_hs_x.\n",
      "Adding exchange reaction EX_xolest204_hs_x with default bounds for boundary metabolite: xolest204_hs_x.\n",
      "Adding exchange reaction EX_xolest226_hs_x with default bounds for boundary metabolite: xolest226_hs_x.\n",
      "Adding exchange reaction EX_ca3s_x with default bounds for boundary metabolite: ca3s_x.\n",
      "Adding exchange reaction EX_ca24g_x with default bounds for boundary metabolite: ca24g_x.\n",
      "Adding exchange reaction EX_tetdeca511ac_x with default bounds for boundary metabolite: tetdeca511ac_x.\n",
      "Adding exchange reaction EX_eidi1114ac_x with default bounds for boundary metabolite: eidi1114ac_x.\n",
      "Adding exchange reaction EX_m01597x with default bounds for boundary metabolite: m01597x.\n",
      "Adding exchange reaction EX_coprost_x with default bounds for boundary metabolite: coprost_x.\n",
      "Adding exchange reaction EX_m01624x with default bounds for boundary metabolite: m01624x.\n",
      "Adding exchange reaction EX_cynt_x with default bounds for boundary metabolite: cynt_x.\n",
      "Adding exchange reaction EX_CE1950_x with default bounds for boundary metabolite: CE1950_x.\n",
      "Adding exchange reaction EX_cysasnmet_x with default bounds for boundary metabolite: cysasnmet_x.\n",
      "Adding exchange reaction EX_cysaspphe_x with default bounds for boundary metabolite: cysaspphe_x.\n",
      "Adding exchange reaction EX_cyscys_x with default bounds for boundary metabolite: cyscys_x.\n",
      "Adding exchange reaction EX_cysglnmet_x with default bounds for boundary metabolite: cysglnmet_x.\n",
      "Adding exchange reaction EX_cysgluhis_x with default bounds for boundary metabolite: cysgluhis_x.\n",
      "Adding exchange reaction EX_cysglutrp_x with default bounds for boundary metabolite: cysglutrp_x.\n",
      "Adding exchange reaction EX_cysleuthr_x with default bounds for boundary metabolite: cysleuthr_x.\n",
      "Adding exchange reaction EX_cyssermet_x with default bounds for boundary metabolite: cyssermet_x.\n",
      "Adding exchange reaction EX_cystyrasn_x with default bounds for boundary metabolite: cystyrasn_x.\n",
      "Adding exchange reaction EX_alaala_x with default bounds for boundary metabolite: alaala_x.\n",
      "Adding exchange reaction EX_abt_D_x with default bounds for boundary metabolite: abt_D_x.\n",
      "Adding exchange reaction EX_galam_x with default bounds for boundary metabolite: galam_x.\n",
      "Adding exchange reaction EX_m01679x with default bounds for boundary metabolite: m01679x.\n",
      "Adding exchange reaction EX_m01681x with default bounds for boundary metabolite: m01681x.\n",
      "Adding exchange reaction EX_m01683x with default bounds for boundary metabolite: m01683x.\n",
      "Adding exchange reaction EX_m01759x with default bounds for boundary metabolite: m01759x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_m01689x with default bounds for boundary metabolite: m01689x.\n",
      "Adding exchange reaction EX_m01690x with default bounds for boundary metabolite: m01690x.\n",
      "Adding exchange reaction EX_m01741x with default bounds for boundary metabolite: m01741x.\n",
      "Adding exchange reaction EX_decdicrn_x with default bounds for boundary metabolite: decdicrn_x.\n",
      "Adding exchange reaction EX_c10crn_x with default bounds for boundary metabolite: c10crn_x.\n",
      "Adding exchange reaction EX_c101crn_x with default bounds for boundary metabolite: c101crn_x.\n",
      "Adding exchange reaction EX_dca3s_x with default bounds for boundary metabolite: dca3s_x.\n",
      "Adding exchange reaction EX_dca24g_x with default bounds for boundary metabolite: dca24g_x.\n",
      "Adding exchange reaction EX_dca3g_x with default bounds for boundary metabolite: dca3g_x.\n",
      "Adding exchange reaction EX_diholineth_x with default bounds for boundary metabolite: diholineth_x.\n",
      "Adding exchange reaction EX_docohxeth_x with default bounds for boundary metabolite: docohxeth_x.\n",
      "Adding exchange reaction EX_docosdiac_x with default bounds for boundary metabolite: docosdiac_x.\n",
      "Adding exchange reaction EX_docteteth_x with default bounds for boundary metabolite: docteteth_x.\n",
      "Adding exchange reaction EX_dodecanac_x with default bounds for boundary metabolite: dodecanac_x.\n",
      "Adding exchange reaction EX_c12dc_x with default bounds for boundary metabolite: c12dc_x.\n",
      "Adding exchange reaction EX_ddece1crn_x with default bounds for boundary metabolite: ddece1crn_x.\n",
      "Adding exchange reaction EX_dopa4sf_x with default bounds for boundary metabolite: dopa4sf_x.\n",
      "Adding exchange reaction EX_dopa3glcur_x with default bounds for boundary metabolite: dopa3glcur_x.\n",
      "Adding exchange reaction EX_dopa4glcur_x with default bounds for boundary metabolite: dopa4glcur_x.\n",
      "Adding exchange reaction EX_m01784x with default bounds for boundary metabolite: m01784x.\n",
      "Adding exchange reaction EX_ethmalac_x with default bounds for boundary metabolite: ethmalac_x.\n",
      "Adding exchange reaction EX_m01801x with default bounds for boundary metabolite: m01801x.\n",
      "Adding exchange reaction EX_m01802x with default bounds for boundary metabolite: m01802x.\n",
      "Adding exchange reaction EX_m01828x with default bounds for boundary metabolite: m01828x.\n",
      "Adding exchange reaction EX_9_cis_retfa_x with default bounds for boundary metabolite: 9_cis_retfa_x.\n",
      "Adding exchange reaction EX_m01904x with default bounds for boundary metabolite: m01904x.\n",
      "Adding exchange reaction EX_m01941x with default bounds for boundary metabolite: m01941x.\n",
      "Adding exchange reaction EX_m01943x with default bounds for boundary metabolite: m01943x.\n",
      "Adding exchange reaction EX_m01946x with default bounds for boundary metabolite: m01946x.\n",
      "Adding exchange reaction EX_m01947x with default bounds for boundary metabolite: m01947x.\n",
      "Adding exchange reaction EX_m02008x with default bounds for boundary metabolite: m02008x.\n",
      "Adding exchange reaction EX_m02009x with default bounds for boundary metabolite: m02009x.\n",
      "Adding exchange reaction EX_m02010x with default bounds for boundary metabolite: m02010x.\n",
      "Adding exchange reaction EX_m02011x with default bounds for boundary metabolite: m02011x.\n",
      "Adding exchange reaction EX_m02014x with default bounds for boundary metabolite: m02014x.\n",
      "Adding exchange reaction EX_m02015x with default bounds for boundary metabolite: m02015x.\n",
      "Adding exchange reaction EX_m02030x with default bounds for boundary metabolite: m02030x.\n",
      "Adding exchange reaction EX_galgluside_hs_x with default bounds for boundary metabolite: galgluside_hs_x.\n",
      "Adding exchange reaction EX_gm1_hs_x with default bounds for boundary metabolite: gm1_hs_x.\n",
      "Adding exchange reaction EX_glz_x with default bounds for boundary metabolite: glz_x.\n",
      "Adding exchange reaction EX_glutcon_x with default bounds for boundary metabolite: glutcon_x.\n",
      "Adding exchange reaction EX_gluargleu_x with default bounds for boundary metabolite: gluargleu_x.\n",
      "Adding exchange reaction EX_glnasngln_x with default bounds for boundary metabolite: glnasngln_x.\n",
      "Adding exchange reaction EX_gluasnleu_x with default bounds for boundary metabolite: gluasnleu_x.\n",
      "Adding exchange reaction EX_glnhishis_x with default bounds for boundary metabolite: glnhishis_x.\n",
      "Adding exchange reaction EX_glnhislys_x with default bounds for boundary metabolite: glnhislys_x.\n",
      "Adding exchange reaction EX_glnlyslys_x with default bounds for boundary metabolite: glnlyslys_x.\n",
      "Adding exchange reaction EX_glnlystrp_x with default bounds for boundary metabolite: glnlystrp_x.\n",
      "Adding exchange reaction EX_glnproglu_x with default bounds for boundary metabolite: glnproglu_x.\n",
      "Adding exchange reaction EX_glntrpglu_x with default bounds for boundary metabolite: glntrpglu_x.\n",
      "Adding exchange reaction EX_glntyrleu_x with default bounds for boundary metabolite: glntyrleu_x.\n",
      "Adding exchange reaction EX_gluglu_x with default bounds for boundary metabolite: gluglu_x.\n",
      "Adding exchange reaction EX_gluilelys_x with default bounds for boundary metabolite: gluilelys_x.\n",
      "Adding exchange reaction EX_gluleu_x with default bounds for boundary metabolite: gluleu_x.\n",
      "Adding exchange reaction EX_glumet_x with default bounds for boundary metabolite: glumet_x.\n",
      "Adding exchange reaction EX_glumethis_x with default bounds for boundary metabolite: glumethis_x.\n",
      "Adding exchange reaction EX_gluthr_x with default bounds for boundary metabolite: gluthr_x.\n",
      "Adding exchange reaction EX_gluthrlys_x with default bounds for boundary metabolite: gluthrlys_x.\n",
      "Adding exchange reaction EX_glutrpala_x with default bounds for boundary metabolite: glutrpala_x.\n",
      "Adding exchange reaction EX_glutar_x with default bounds for boundary metabolite: glutar_x.\n",
      "Adding exchange reaction EX_c5dc_x with default bounds for boundary metabolite: c5dc_x.\n",
      "Adding exchange reaction EX_glyc2p_x with default bounds for boundary metabolite: glyc2p_x.\n",
      "Adding exchange reaction EX_gcdca3s_x with default bounds for boundary metabolite: gcdca3s_x.\n",
      "Adding exchange reaction EX_gca3s_x with default bounds for boundary metabolite: gca3s_x.\n",
      "Adding exchange reaction EX_gdca3s_x with default bounds for boundary metabolite: gdca3s_x.\n",
      "Adding exchange reaction EX_glygn4_x with default bounds for boundary metabolite: glygn4_x.\n",
      "Adding exchange reaction EX_glygn5_x with default bounds for boundary metabolite: glygn5_x.\n",
      "Adding exchange reaction EX_gudca3s_x with default bounds for boundary metabolite: gudca3s_x.\n",
      "Adding exchange reaction EX_glygly_x with default bounds for boundary metabolite: glygly_x.\n",
      "Adding exchange reaction EX_glyhisasn_x with default bounds for boundary metabolite: glyhisasn_x.\n",
      "Adding exchange reaction EX_glyhislys_x with default bounds for boundary metabolite: glyhislys_x.\n",
      "Adding exchange reaction EX_glylyscys_x with default bounds for boundary metabolite: glylyscys_x.\n",
      "Adding exchange reaction EX_glylysphe_x with default bounds for boundary metabolite: glylysphe_x.\n",
      "Adding exchange reaction EX_glyphe_x with default bounds for boundary metabolite: glyphe_x.\n",
      "Adding exchange reaction EX_glytyrlys_x with default bounds for boundary metabolite: glytyrlys_x.\n",
      "Adding exchange reaction EX_glyvalhis_x with default bounds for boundary metabolite: glyvalhis_x.\n",
      "Adding exchange reaction EX_glyleu_x with default bounds for boundary metabolite: glyleu_x.\n",
      "Adding exchange reaction EX_glypro_x with default bounds for boundary metabolite: glypro_x.\n",
      "Adding exchange reaction EX_glysar_x with default bounds for boundary metabolite: glysar_x.\n",
      "Adding exchange reaction EX_HC02154_x with default bounds for boundary metabolite: HC02154_x.\n",
      "Adding exchange reaction EX_gumdchac_x with default bounds for boundary metabolite: gumdchac_x.\n",
      "Adding exchange reaction EX_gumgchol_x with default bounds for boundary metabolite: gumgchol_x.\n",
      "Adding exchange reaction EX_gumtchol_x with default bounds for boundary metabolite: gumtchol_x.\n",
      "Adding exchange reaction EX_gum_x with default bounds for boundary metabolite: gum_x.\n",
      "Adding exchange reaction EX_hepdeceth_x with default bounds for boundary metabolite: hepdeceth_x.\n",
      "Adding exchange reaction EX_hexdiac_x with default bounds for boundary metabolite: hexdiac_x.\n",
      "Adding exchange reaction EX_c16dc_x with default bounds for boundary metabolite: c16dc_x.\n",
      "Adding exchange reaction EX_hexdeceeth_x with default bounds for boundary metabolite: hexdeceeth_x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_hxa_x with default bounds for boundary metabolite: hxa_x.\n",
      "Adding exchange reaction EX_c6crn_x with default bounds for boundary metabolite: c6crn_x.\n",
      "Adding exchange reaction EX_hexgly_x with default bounds for boundary metabolite: hexgly_x.\n",
      "Adding exchange reaction EX_hdl_hs_x with default bounds for boundary metabolite: hdl_hs_x.\n",
      "Adding exchange reaction EX_hisargcys_x with default bounds for boundary metabolite: hisargcys_x.\n",
      "Adding exchange reaction EX_hisargser_x with default bounds for boundary metabolite: hisargser_x.\n",
      "Adding exchange reaction EX_hisasp_x with default bounds for boundary metabolite: hisasp_x.\n",
      "Adding exchange reaction EX_hiscyscys_x with default bounds for boundary metabolite: hiscyscys_x.\n",
      "Adding exchange reaction EX_hisglu_x with default bounds for boundary metabolite: hisglu_x.\n",
      "Adding exchange reaction EX_hisglnala_x with default bounds for boundary metabolite: hisglnala_x.\n",
      "Adding exchange reaction EX_hisglugln_x with default bounds for boundary metabolite: hisglugln_x.\n",
      "Adding exchange reaction EX_hishislys_x with default bounds for boundary metabolite: hishislys_x.\n",
      "Adding exchange reaction EX_hislysala_x with default bounds for boundary metabolite: hislysala_x.\n",
      "Adding exchange reaction EX_hislysglu_x with default bounds for boundary metabolite: hislysglu_x.\n",
      "Adding exchange reaction EX_hislysile_x with default bounds for boundary metabolite: hislysile_x.\n",
      "Adding exchange reaction EX_hisglylys_x with default bounds for boundary metabolite: hisglylys_x.\n",
      "Adding exchange reaction EX_hislysthr_x with default bounds for boundary metabolite: hislysthr_x.\n",
      "Adding exchange reaction EX_hislysval_x with default bounds for boundary metabolite: hislysval_x.\n",
      "Adding exchange reaction EX_hismet_x with default bounds for boundary metabolite: hismet_x.\n",
      "Adding exchange reaction EX_hismetgln_x with default bounds for boundary metabolite: hismetgln_x.\n",
      "Adding exchange reaction EX_hisphearg_x with default bounds for boundary metabolite: hisphearg_x.\n",
      "Adding exchange reaction EX_hisprolys_x with default bounds for boundary metabolite: hisprolys_x.\n",
      "Adding exchange reaction EX_histrphis_x with default bounds for boundary metabolite: histrphis_x.\n",
      "Adding exchange reaction EX_hmcarn_x with default bounds for boundary metabolite: hmcarn_x.\n",
      "Adding exchange reaction EX_hmcr_x with default bounds for boundary metabolite: hmcr_x.\n",
      "Adding exchange reaction EX_hca24g_x with default bounds for boundary metabolite: hca24g_x.\n",
      "Adding exchange reaction EX_hca6g_x with default bounds for boundary metabolite: hca6g_x.\n",
      "Adding exchange reaction EX_hyochol_x with default bounds for boundary metabolite: hyochol_x.\n",
      "Adding exchange reaction EX_hdca24g_x with default bounds for boundary metabolite: hdca24g_x.\n",
      "Adding exchange reaction EX_hdca6g_x with default bounds for boundary metabolite: hdca6g_x.\n",
      "Adding exchange reaction EX_indole_x with default bounds for boundary metabolite: indole_x.\n",
      "Adding exchange reaction EX_inds_x with default bounds for boundary metabolite: inds_x.\n",
      "Adding exchange reaction EX_idl_hs_x with default bounds for boundary metabolite: idl_hs_x.\n",
      "Adding exchange reaction EX_isochol_x with default bounds for boundary metabolite: isochol_x.\n",
      "Adding exchange reaction EX_icdchol_x with default bounds for boundary metabolite: icdchol_x.\n",
      "Adding exchange reaction EX_ileasp_x with default bounds for boundary metabolite: ileasp_x.\n",
      "Adding exchange reaction EX_ileglnglu_x with default bounds for boundary metabolite: ileglnglu_x.\n",
      "Adding exchange reaction EX_ileglyarg_x with default bounds for boundary metabolite: ileglyarg_x.\n",
      "Adding exchange reaction EX_ileprolys_x with default bounds for boundary metabolite: ileprolys_x.\n",
      "Adding exchange reaction EX_ileserarg_x with default bounds for boundary metabolite: ileserarg_x.\n",
      "Adding exchange reaction EX_iletrptyr_x with default bounds for boundary metabolite: iletrptyr_x.\n",
      "Adding exchange reaction EX_ileargile_x with default bounds for boundary metabolite: ileargile_x.\n",
      "Adding exchange reaction EX_ileasnhis_x with default bounds for boundary metabolite: ileasnhis_x.\n",
      "Adding exchange reaction EX_ivcrn_x with default bounds for boundary metabolite: ivcrn_x.\n",
      "Adding exchange reaction EX_m02322x with default bounds for boundary metabolite: m02322x.\n",
      "Adding exchange reaction EX_m02326x with default bounds for boundary metabolite: m02326x.\n",
      "Adding exchange reaction EX_Lhcystin_x with default bounds for boundary metabolite: Lhcystin_x.\n",
      "Adding exchange reaction EX_progly_x with default bounds for boundary metabolite: progly_x.\n",
      "Adding exchange reaction EX_m02349x with default bounds for boundary metabolite: m02349x.\n",
      "Adding exchange reaction EX_m02350x with default bounds for boundary metabolite: m02350x.\n",
      "Adding exchange reaction EX_m02381x with default bounds for boundary metabolite: m02381x.\n",
      "Adding exchange reaction EX_m02409x with default bounds for boundary metabolite: m02409x.\n",
      "Adding exchange reaction EX_m02411x with default bounds for boundary metabolite: m02411x.\n",
      "Adding exchange reaction EX_m02413x with default bounds for boundary metabolite: m02413x.\n",
      "Adding exchange reaction EX_m02425x with default bounds for boundary metabolite: m02425x.\n",
      "Adding exchange reaction EX_ddeccrn_x with default bounds for boundary metabolite: ddeccrn_x.\n",
      "Adding exchange reaction EX_fucfucgalacglcgalgluside_hs_x with default bounds for boundary metabolite: fucfucgalacglcgalgluside_hs_x.\n",
      "Adding exchange reaction EX_leualaarg_x with default bounds for boundary metabolite: leualaarg_x.\n",
      "Adding exchange reaction EX_leuasnasp_x with default bounds for boundary metabolite: leuasnasp_x.\n",
      "Adding exchange reaction EX_leuasplys_x with default bounds for boundary metabolite: leuasplys_x.\n",
      "Adding exchange reaction EX_leugly_x with default bounds for boundary metabolite: leugly_x.\n",
      "Adding exchange reaction EX_leuleutrp_x with default bounds for boundary metabolite: leuleutrp_x.\n",
      "Adding exchange reaction EX_leupro_x with default bounds for boundary metabolite: leupro_x.\n",
      "Adding exchange reaction EX_leuproarg_x with default bounds for boundary metabolite: leuproarg_x.\n",
      "Adding exchange reaction EX_leusertrp_x with default bounds for boundary metabolite: leusertrp_x.\n",
      "Adding exchange reaction EX_leutrp_x with default bounds for boundary metabolite: leutrp_x.\n",
      "Adding exchange reaction EX_leutrparg_x with default bounds for boundary metabolite: leutrparg_x.\n",
      "Adding exchange reaction EX_leutyrtyr_x with default bounds for boundary metabolite: leutyrtyr_x.\n",
      "Adding exchange reaction EX_leuval_x with default bounds for boundary metabolite: leuval_x.\n",
      "Adding exchange reaction EX_leuleu_x with default bounds for boundary metabolite: leuleu_x.\n",
      "Adding exchange reaction EX_fucfuc12gal14acglcgalgluside_hs_x with default bounds for boundary metabolite: fucfuc12gal14acglcgalgluside_hs_x.\n",
      "Adding exchange reaction EX_m02382x with default bounds for boundary metabolite: m02382x.\n",
      "Adding exchange reaction EX_lineth_x with default bounds for boundary metabolite: lineth_x.\n",
      "Adding exchange reaction EX_lca3s_x with default bounds for boundary metabolite: lca3s_x.\n",
      "Adding exchange reaction EX_lca24g_x with default bounds for boundary metabolite: lca24g_x.\n",
      "Adding exchange reaction EX_lca3g_x with default bounds for boundary metabolite: lca3g_x.\n",
      "Adding exchange reaction EX_lstn_x with default bounds for boundary metabolite: lstn_x.\n",
      "Adding exchange reaction EX_lst4exp_x with default bounds for boundary metabolite: lst4exp_x.\n",
      "Adding exchange reaction EX_lstnm1_x with default bounds for boundary metabolite: lstnm1_x.\n",
      "Adding exchange reaction EX_lstnm2_x with default bounds for boundary metabolite: lstnm2_x.\n",
      "Adding exchange reaction EX_lstnm4_x with default bounds for boundary metabolite: lstnm4_x.\n",
      "Adding exchange reaction EX_lstnm5_x with default bounds for boundary metabolite: lstnm5_x.\n",
      "Adding exchange reaction EX_lstn1gluc_x with default bounds for boundary metabolite: lstn1gluc_x.\n",
      "Adding exchange reaction EX_lstnm7_x with default bounds for boundary metabolite: lstnm7_x.\n",
      "Adding exchange reaction EX_ldl_hs_x with default bounds for boundary metabolite: ldl_hs_x.\n",
      "Adding exchange reaction EX_pcholn261_hs_x with default bounds for boundary metabolite: pcholn261_hs_x.\n",
      "Adding exchange reaction EX_pcholn28_hs_x with default bounds for boundary metabolite: pcholn28_hs_x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_pcholn281_hs_x with default bounds for boundary metabolite: pcholn281_hs_x.\n",
      "Adding exchange reaction EX_lysargleu_x with default bounds for boundary metabolite: lysargleu_x.\n",
      "Adding exchange reaction EX_lyscyshis_x with default bounds for boundary metabolite: lyscyshis_x.\n",
      "Adding exchange reaction EX_lysglnphe_x with default bounds for boundary metabolite: lysglnphe_x.\n",
      "Adding exchange reaction EX_lysgluglu_x with default bounds for boundary metabolite: lysgluglu_x.\n",
      "Adding exchange reaction EX_lyslyslys_x with default bounds for boundary metabolite: lyslyslys_x.\n",
      "Adding exchange reaction EX_lyspheile_x with default bounds for boundary metabolite: lyspheile_x.\n",
      "Adding exchange reaction EX_lystrparg_x with default bounds for boundary metabolite: lystrparg_x.\n",
      "Adding exchange reaction EX_lystyrile_x with default bounds for boundary metabolite: lystyrile_x.\n",
      "Adding exchange reaction EX_lysvalphe_x with default bounds for boundary metabolite: lysvalphe_x.\n",
      "Adding exchange reaction EX_lysvaltrp_x with default bounds for boundary metabolite: lysvaltrp_x.\n",
      "Adding exchange reaction EX_c3dc_x with default bounds for boundary metabolite: c3dc_x.\n",
      "Adding exchange reaction EX_malthp_x with default bounds for boundary metabolite: malthp_x.\n",
      "Adding exchange reaction EX_malthx_x with default bounds for boundary metabolite: malthx_x.\n",
      "Adding exchange reaction EX_maltpt_x with default bounds for boundary metabolite: maltpt_x.\n",
      "Adding exchange reaction EX_maltttr_x with default bounds for boundary metabolite: maltttr_x.\n",
      "Adding exchange reaction EX_mqn10_x with default bounds for boundary metabolite: mqn10_x.\n",
      "Adding exchange reaction EX_mqn11_x with default bounds for boundary metabolite: mqn11_x.\n",
      "Adding exchange reaction EX_mqn7_x with default bounds for boundary metabolite: mqn7_x.\n",
      "Adding exchange reaction EX_mqn8_x with default bounds for boundary metabolite: mqn8_x.\n",
      "Adding exchange reaction EX_mqn9_x with default bounds for boundary metabolite: mqn9_x.\n",
      "Adding exchange reaction EX_ch4s_x with default bounds for boundary metabolite: ch4s_x.\n",
      "Adding exchange reaction EX_metargleu_x with default bounds for boundary metabolite: metargleu_x.\n",
      "Adding exchange reaction EX_metasntyr_x with default bounds for boundary metabolite: metasntyr_x.\n",
      "Adding exchange reaction EX_metglntyr_x with default bounds for boundary metabolite: metglntyr_x.\n",
      "Adding exchange reaction EX_metglyarg_x with default bounds for boundary metabolite: metglyarg_x.\n",
      "Adding exchange reaction EX_methislys_x with default bounds for boundary metabolite: methislys_x.\n",
      "Adding exchange reaction EX_metmetile_x with default bounds for boundary metabolite: metmetile_x.\n",
      "Adding exchange reaction EX_metphearg_x with default bounds for boundary metabolite: metphearg_x.\n",
      "Adding exchange reaction EX_mettrpphe_x with default bounds for boundary metabolite: mettrpphe_x.\n",
      "Adding exchange reaction EX_methsucc_x with default bounds for boundary metabolite: methsucc_x.\n",
      "Adding exchange reaction EX_mhglz_x with default bounds for boundary metabolite: mhglz_x.\n",
      "Adding exchange reaction EX_mvlac_x with default bounds for boundary metabolite: mvlac_x.\n",
      "Adding exchange reaction EX_m02482x with default bounds for boundary metabolite: m02482x.\n",
      "Adding exchange reaction EX_mdz_x with default bounds for boundary metabolite: mdz_x.\n",
      "Adding exchange reaction EX_mdzglc_x with default bounds for boundary metabolite: mdzglc_x.\n",
      "Adding exchange reaction EX_m02497x with default bounds for boundary metabolite: m02497x.\n",
      "Adding exchange reaction EX_actyr_x with default bounds for boundary metabolite: actyr_x.\n",
      "Adding exchange reaction EX_acleu_L_x with default bounds for boundary metabolite: acleu_L_x.\n",
      "Adding exchange reaction EX_nacvanala_x with default bounds for boundary metabolite: nacvanala_x.\n",
      "Adding exchange reaction EX_m02530x with default bounds for boundary metabolite: m02530x.\n",
      "Adding exchange reaction EX_m02531x with default bounds for boundary metabolite: m02531x.\n",
      "Adding exchange reaction EX_m02532x with default bounds for boundary metabolite: m02532x.\n",
      "Adding exchange reaction EX_m02534x with default bounds for boundary metabolite: m02534x.\n",
      "Adding exchange reaction EX_m02536x with default bounds for boundary metabolite: m02536x.\n",
      "Adding exchange reaction EX_m02550x with default bounds for boundary metabolite: m02550x.\n",
      "Adding exchange reaction EX_m02551x with default bounds for boundary metabolite: m02551x.\n",
      "Adding exchange reaction EX_m02525x with default bounds for boundary metabolite: m02525x.\n",
      "Adding exchange reaction EX_m02540x with default bounds for boundary metabolite: m02540x.\n",
      "Adding exchange reaction EX_m02543x with default bounds for boundary metabolite: m02543x.\n",
      "Adding exchange reaction EX_m02546x with default bounds for boundary metabolite: m02546x.\n",
      "Adding exchange reaction EX_m02559x with default bounds for boundary metabolite: m02559x.\n",
      "Adding exchange reaction EX_desfvs_x with default bounds for boundary metabolite: desfvs_x.\n",
      "Adding exchange reaction EX_ndersv_x with default bounds for boundary metabolite: ndersv_x.\n",
      "Adding exchange reaction EX_m02574x with default bounds for boundary metabolite: m02574x.\n",
      "Adding exchange reaction EX_m02601x with default bounds for boundary metabolite: m02601x.\n",
      "Adding exchange reaction EX_m02603x with default bounds for boundary metabolite: m02603x.\n",
      "Adding exchange reaction EX_m02503x with default bounds for boundary metabolite: m02503x.\n",
      "Adding exchange reaction EX_m02517x with default bounds for boundary metabolite: m02517x.\n",
      "Adding exchange reaction EX_m02518x with default bounds for boundary metabolite: m02518x.\n",
      "Adding exchange reaction EX_m02553x with default bounds for boundary metabolite: m02553x.\n",
      "Adding exchange reaction EX_m02579x with default bounds for boundary metabolite: m02579x.\n",
      "Adding exchange reaction EX_nfd_x with default bounds for boundary metabolite: nfd_x.\n",
      "Adding exchange reaction EX_nfdnpy_x with default bounds for boundary metabolite: nfdnpy_x.\n",
      "Adding exchange reaction EX_aracheth_x with default bounds for boundary metabolite: aracheth_x.\n",
      "Adding exchange reaction EX_m02634x with default bounds for boundary metabolite: m02634x.\n",
      "Adding exchange reaction EX_m02635x with default bounds for boundary metabolite: m02635x.\n",
      "Adding exchange reaction EX_m02654x with default bounds for boundary metabolite: m02654x.\n",
      "Adding exchange reaction EX_m02657x with default bounds for boundary metabolite: m02657x.\n",
      "Adding exchange reaction EX_m02633x with default bounds for boundary metabolite: m02633x.\n",
      "Adding exchange reaction EX_c81crn_x with default bounds for boundary metabolite: c81crn_x.\n",
      "Adding exchange reaction EX_oleth_x with default bounds for boundary metabolite: oleth_x.\n",
      "Adding exchange reaction EX_pcresol_x with default bounds for boundary metabolite: pcresol_x.\n",
      "Adding exchange reaction EX_pcs_x with default bounds for boundary metabolite: pcs_x.\n",
      "Adding exchange reaction EX_m02684x with default bounds for boundary metabolite: m02684x.\n",
      "Adding exchange reaction EX_m02685x with default bounds for boundary metabolite: m02685x.\n",
      "Adding exchange reaction EX_m02696x with default bounds for boundary metabolite: m02696x.\n",
      "Adding exchange reaction EX_m02715x with default bounds for boundary metabolite: m02715x.\n",
      "Adding exchange reaction EX_m02750x with default bounds for boundary metabolite: m02750x.\n",
      "Adding exchange reaction EX_m02759x with default bounds for boundary metabolite: m02759x.\n",
      "Adding exchange reaction EX_m02806x with default bounds for boundary metabolite: m02806x.\n",
      "Adding exchange reaction EX_m02808x with default bounds for boundary metabolite: m02808x.\n",
      "Adding exchange reaction EX_pmeth_x with default bounds for boundary metabolite: pmeth_x.\n",
      "Adding exchange reaction EX_pect_x with default bounds for boundary metabolite: pect_x.\n",
      "Adding exchange reaction EX_pectindchac_x with default bounds for boundary metabolite: pectindchac_x.\n",
      "Adding exchange reaction EX_pectingchol_x with default bounds for boundary metabolite: pectingchol_x.\n",
      "Adding exchange reaction EX_pectintchol_x with default bounds for boundary metabolite: pectintchol_x.\n",
      "Adding exchange reaction EX_pendecaeth_x with default bounds for boundary metabolite: pendecaeth_x.\n",
      "Adding exchange reaction EX_pheacgly_x with default bounds for boundary metabolite: pheacgly_x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_phacgly_x with default bounds for boundary metabolite: phacgly_x.\n",
      "Adding exchange reaction EX_pheasnmet_x with default bounds for boundary metabolite: pheasnmet_x.\n",
      "Adding exchange reaction EX_pheasp_x with default bounds for boundary metabolite: pheasp_x.\n",
      "Adding exchange reaction EX_pheglnphe_x with default bounds for boundary metabolite: pheglnphe_x.\n",
      "Adding exchange reaction EX_pheleu_x with default bounds for boundary metabolite: pheleu_x.\n",
      "Adding exchange reaction EX_pheleuasp_x with default bounds for boundary metabolite: pheleuasp_x.\n",
      "Adding exchange reaction EX_pheleuhis_x with default bounds for boundary metabolite: pheleuhis_x.\n",
      "Adding exchange reaction EX_phelysala_x with default bounds for boundary metabolite: phelysala_x.\n",
      "Adding exchange reaction EX_phelyspro_x with default bounds for boundary metabolite: phelyspro_x.\n",
      "Adding exchange reaction EX_phephe_x with default bounds for boundary metabolite: phephe_x.\n",
      "Adding exchange reaction EX_phepheasn_x with default bounds for boundary metabolite: phepheasn_x.\n",
      "Adding exchange reaction EX_phephethr_x with default bounds for boundary metabolite: phephethr_x.\n",
      "Adding exchange reaction EX_pheproarg_x with default bounds for boundary metabolite: pheproarg_x.\n",
      "Adding exchange reaction EX_phesertrp_x with default bounds for boundary metabolite: phesertrp_x.\n",
      "Adding exchange reaction EX_phethrlys_x with default bounds for boundary metabolite: phethrlys_x.\n",
      "Adding exchange reaction EX_phetrpleu_x with default bounds for boundary metabolite: phetrpleu_x.\n",
      "Adding exchange reaction EX_phetyr_x with default bounds for boundary metabolite: phetyr_x.\n",
      "Adding exchange reaction EX_phetyrgln_x with default bounds for boundary metabolite: phetyrgln_x.\n",
      "Adding exchange reaction EX_phetyrlys_x with default bounds for boundary metabolite: phetyrlys_x.\n",
      "Adding exchange reaction EX_phlac_x with default bounds for boundary metabolite: phlac_x.\n",
      "Adding exchange reaction EX_phyt_x with default bounds for boundary metabolite: phyt_x.\n",
      "Adding exchange reaction EX_proargasp_x with default bounds for boundary metabolite: proargasp_x.\n",
      "Adding exchange reaction EX_proargcys_x with default bounds for boundary metabolite: proargcys_x.\n",
      "Adding exchange reaction EX_proasncys_x with default bounds for boundary metabolite: proasncys_x.\n",
      "Adding exchange reaction EX_procys_x with default bounds for boundary metabolite: procys_x.\n",
      "Adding exchange reaction EX_proglulys_x with default bounds for boundary metabolite: proglulys_x.\n",
      "Adding exchange reaction EX_proglnpro_x with default bounds for boundary metabolite: proglnpro_x.\n",
      "Adding exchange reaction EX_prohis_x with default bounds for boundary metabolite: prohis_x.\n",
      "Adding exchange reaction EX_prohistyr_x with default bounds for boundary metabolite: prohistyr_x.\n",
      "Adding exchange reaction EX_proleuarg_x with default bounds for boundary metabolite: proleuarg_x.\n",
      "Adding exchange reaction EX_prolyspro_x with default bounds for boundary metabolite: prolyspro_x.\n",
      "Adding exchange reaction EX_prophe_x with default bounds for boundary metabolite: prophe_x.\n",
      "Adding exchange reaction EX_proproarg_x with default bounds for boundary metabolite: proproarg_x.\n",
      "Adding exchange reaction EX_propropro_x with default bounds for boundary metabolite: propropro_x.\n",
      "Adding exchange reaction EX_protrplys_x with default bounds for boundary metabolite: protrplys_x.\n",
      "Adding exchange reaction EX_protrpthr_x with default bounds for boundary metabolite: protrpthr_x.\n",
      "Adding exchange reaction EX_provalgln_x with default bounds for boundary metabolite: provalgln_x.\n",
      "Adding exchange reaction EX_ppiogly_x with default bounds for boundary metabolite: ppiogly_x.\n",
      "Adding exchange reaction EX_psylchol_x with default bounds for boundary metabolite: psylchol_x.\n",
      "Adding exchange reaction EX_psyl_x with default bounds for boundary metabolite: psyl_x.\n",
      "Adding exchange reaction EX_psyltchol_x with default bounds for boundary metabolite: psyltchol_x.\n",
      "Adding exchange reaction EX_psyltdechol_x with default bounds for boundary metabolite: psyltdechol_x.\n",
      "Adding exchange reaction EX_caribup_R_x with default bounds for boundary metabolite: caribup_R_x.\n",
      "Adding exchange reaction EX_slfcys_x with default bounds for boundary metabolite: slfcys_x.\n",
      "Adding exchange reaction EX_m02870x with default bounds for boundary metabolite: m02870x.\n",
      "Adding exchange reaction EX_caribup_s_x with default bounds for boundary metabolite: caribup_s_x.\n",
      "Adding exchange reaction EX_caribupglu_S_x with default bounds for boundary metabolite: caribupglu_S_x.\n",
      "Adding exchange reaction EX_m02900x with default bounds for boundary metabolite: m02900x.\n",
      "Adding exchange reaction EX_m02901x with default bounds for boundary metabolite: m02901x.\n",
      "Adding exchange reaction EX_m02871x with default bounds for boundary metabolite: m02871x.\n",
      "Adding exchange reaction EX_m02877x with default bounds for boundary metabolite: m02877x.\n",
      "Adding exchange reaction EX_m02908x with default bounds for boundary metabolite: m02908x.\n",
      "Adding exchange reaction EX_sebacid_x with default bounds for boundary metabolite: sebacid_x.\n",
      "Adding exchange reaction EX_c10dc_x with default bounds for boundary metabolite: c10dc_x.\n",
      "Adding exchange reaction EX_serargala_x with default bounds for boundary metabolite: serargala_x.\n",
      "Adding exchange reaction EX_serargtrp_x with default bounds for boundary metabolite: serargtrp_x.\n",
      "Adding exchange reaction EX_sercysarg_x with default bounds for boundary metabolite: sercysarg_x.\n",
      "Adding exchange reaction EX_serglyglu_x with default bounds for boundary metabolite: serglyglu_x.\n",
      "Adding exchange reaction EX_serlyshis_x with default bounds for boundary metabolite: serlyshis_x.\n",
      "Adding exchange reaction EX_serphelys_x with default bounds for boundary metabolite: serphelys_x.\n",
      "Adding exchange reaction EX_sertrphis_x with default bounds for boundary metabolite: sertrphis_x.\n",
      "Adding exchange reaction EX_sphmyln1824_hs_x with default bounds for boundary metabolite: sphmyln1824_hs_x.\n",
      "Adding exchange reaction EX_sphmyln180241_hs_x with default bounds for boundary metabolite: sphmyln180241_hs_x.\n",
      "Adding exchange reaction EX_sphmyln1825_hs_x with default bounds for boundary metabolite: sphmyln1825_hs_x.\n",
      "Adding exchange reaction EX_sphmyln18114_hs_x with default bounds for boundary metabolite: sphmyln18114_hs_x.\n",
      "Adding exchange reaction EX_sphmyln18115_hs_x with default bounds for boundary metabolite: sphmyln18115_hs_x.\n",
      "Adding exchange reaction EX_sphmyln18116_hs_x with default bounds for boundary metabolite: sphmyln18116_hs_x.\n",
      "Adding exchange reaction EX_sphmyln181161_hs_x with default bounds for boundary metabolite: sphmyln181161_hs_x.\n",
      "Adding exchange reaction EX_sphmyln18117_hs_x with default bounds for boundary metabolite: sphmyln18117_hs_x.\n",
      "Adding exchange reaction EX_sphmyln18118_hs_x with default bounds for boundary metabolite: sphmyln18118_hs_x.\n",
      "Adding exchange reaction EX_sphmyln181181_hs_x with default bounds for boundary metabolite: sphmyln181181_hs_x.\n",
      "Adding exchange reaction EX_sphmyln18120_hs_x with default bounds for boundary metabolite: sphmyln18120_hs_x.\n",
      "Adding exchange reaction EX_sphmyln181201_hs_x with default bounds for boundary metabolite: sphmyln181201_hs_x.\n",
      "Adding exchange reaction EX_sphmyln18121_hs_x with default bounds for boundary metabolite: sphmyln18121_hs_x.\n",
      "Adding exchange reaction EX_sphmyln18122_hs_x with default bounds for boundary metabolite: sphmyln18122_hs_x.\n",
      "Adding exchange reaction EX_sphmyln181221_hs_x with default bounds for boundary metabolite: sphmyln181221_hs_x.\n",
      "Adding exchange reaction EX_sphmyln18123_hs_x with default bounds for boundary metabolite: sphmyln18123_hs_x.\n",
      "Adding exchange reaction EX_steeth_x with default bounds for boundary metabolite: steeth_x.\n",
      "Adding exchange reaction EX_subeac_x with default bounds for boundary metabolite: subeac_x.\n",
      "Adding exchange reaction EX_c8dc_x with default bounds for boundary metabolite: c8dc_x.\n",
      "Adding exchange reaction EX_subgly_x with default bounds for boundary metabolite: subgly_x.\n",
      "Adding exchange reaction EX_c4dc_x with default bounds for boundary metabolite: c4dc_x.\n",
      "Adding exchange reaction EX_sucaceto_x with default bounds for boundary metabolite: sucaceto_x.\n",
      "Adding exchange reaction EX_m02959x with default bounds for boundary metabolite: m02959x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_m02956x with default bounds for boundary metabolite: m02956x.\n",
      "Adding exchange reaction EX_tacr_x with default bounds for boundary metabolite: tacr_x.\n",
      "Adding exchange reaction EX_tcdca3s_x with default bounds for boundary metabolite: tcdca3s_x.\n",
      "Adding exchange reaction EX_tca3s_x with default bounds for boundary metabolite: tca3s_x.\n",
      "Adding exchange reaction EX_tdca3s_x with default bounds for boundary metabolite: tdca3s_x.\n",
      "Adding exchange reaction EX_thyochol_x with default bounds for boundary metabolite: thyochol_x.\n",
      "Adding exchange reaction EX_tudca3s_x with default bounds for boundary metabolite: tudca3s_x.\n",
      "Adding exchange reaction EX_tetdec2crn_x with default bounds for boundary metabolite: tetdec2crn_x.\n",
      "Adding exchange reaction EX_tetdece1crn_x with default bounds for boundary metabolite: tetdece1crn_x.\n",
      "Adding exchange reaction EX_thrargtyr_x with default bounds for boundary metabolite: thrargtyr_x.\n",
      "Adding exchange reaction EX_thrasntyr_x with default bounds for boundary metabolite: thrasntyr_x.\n",
      "Adding exchange reaction EX_thrglnglu_x with default bounds for boundary metabolite: thrglnglu_x.\n",
      "Adding exchange reaction EX_thrglntyr_x with default bounds for boundary metabolite: thrglntyr_x.\n",
      "Adding exchange reaction EX_thrhishis_x with default bounds for boundary metabolite: thrhishis_x.\n",
      "Adding exchange reaction EX_thrilearg_x with default bounds for boundary metabolite: thrilearg_x.\n",
      "Adding exchange reaction EX_thrmetarg_x with default bounds for boundary metabolite: thrmetarg_x.\n",
      "Adding exchange reaction EX_thrphearg_x with default bounds for boundary metabolite: thrphearg_x.\n",
      "Adding exchange reaction EX_thrserarg_x with default bounds for boundary metabolite: thrserarg_x.\n",
      "Adding exchange reaction EX_thrthrarg_x with default bounds for boundary metabolite: thrthrarg_x.\n",
      "Adding exchange reaction EX_thrtyrmet_x with default bounds for boundary metabolite: thrtyrmet_x.\n",
      "Adding exchange reaction EX_txb2_x with default bounds for boundary metabolite: txb2_x.\n",
      "Adding exchange reaction EX_c51crn_x with default bounds for boundary metabolite: c51crn_x.\n",
      "Adding exchange reaction EX_tiggly_x with default bounds for boundary metabolite: tiggly_x.\n",
      "Adding exchange reaction EX_tmd_x with default bounds for boundary metabolite: tmd_x.\n",
      "Adding exchange reaction EX_tmdm1_x with default bounds for boundary metabolite: tmdm1_x.\n",
      "Adding exchange reaction EX_tmdm3_x with default bounds for boundary metabolite: tmdm3_x.\n",
      "Adding exchange reaction EX_tmdm5_x with default bounds for boundary metabolite: tmdm5_x.\n",
      "Adding exchange reaction EX_eic21114tr_x with default bounds for boundary metabolite: eic21114tr_x.\n",
      "Adding exchange reaction EX_hpdece_x with default bounds for boundary metabolite: hpdece_x.\n",
      "Adding exchange reaction EX_trideceth_x with default bounds for boundary metabolite: trideceth_x.\n",
      "Adding exchange reaction EX_trpalapro_x with default bounds for boundary metabolite: trpalapro_x.\n",
      "Adding exchange reaction EX_trpargala_x with default bounds for boundary metabolite: trpargala_x.\n",
      "Adding exchange reaction EX_trpaspasp_x with default bounds for boundary metabolite: trpaspasp_x.\n",
      "Adding exchange reaction EX_trpglngln_x with default bounds for boundary metabolite: trpglngln_x.\n",
      "Adding exchange reaction EX_trpglugly_x with default bounds for boundary metabolite: trpglugly_x.\n",
      "Adding exchange reaction EX_trpgluleu_x with default bounds for boundary metabolite: trpgluleu_x.\n",
      "Adding exchange reaction EX_trpglupro_x with default bounds for boundary metabolite: trpglupro_x.\n",
      "Adding exchange reaction EX_trpglutyr_x with default bounds for boundary metabolite: trpglutyr_x.\n",
      "Adding exchange reaction EX_trpglyasp_x with default bounds for boundary metabolite: trpglyasp_x.\n",
      "Adding exchange reaction EX_trpglyleu_x with default bounds for boundary metabolite: trpglyleu_x.\n",
      "Adding exchange reaction EX_trpglyphe_x with default bounds for boundary metabolite: trpglyphe_x.\n",
      "Adding exchange reaction EX_trpglyval_x with default bounds for boundary metabolite: trpglyval_x.\n",
      "Adding exchange reaction EX_trphismet_x with default bounds for boundary metabolite: trphismet_x.\n",
      "Adding exchange reaction EX_trpilelys_x with default bounds for boundary metabolite: trpilelys_x.\n",
      "Adding exchange reaction EX_trpiletrp_x with default bounds for boundary metabolite: trpiletrp_x.\n",
      "Adding exchange reaction EX_trpleuval_x with default bounds for boundary metabolite: trpleuval_x.\n",
      "Adding exchange reaction EX_trplys_x with default bounds for boundary metabolite: trplys_x.\n",
      "Adding exchange reaction EX_trpmetarg_x with default bounds for boundary metabolite: trpmetarg_x.\n",
      "Adding exchange reaction EX_trpmetval_x with default bounds for boundary metabolite: trpmetval_x.\n",
      "Adding exchange reaction EX_trpphe_x with default bounds for boundary metabolite: trpphe_x.\n",
      "Adding exchange reaction EX_trpprogly_x with default bounds for boundary metabolite: trpprogly_x.\n",
      "Adding exchange reaction EX_trpproleu_x with default bounds for boundary metabolite: trpproleu_x.\n",
      "Adding exchange reaction EX_trpproval_x with default bounds for boundary metabolite: trpproval_x.\n",
      "Adding exchange reaction EX_trpsertyr_x with default bounds for boundary metabolite: trpsertyr_x.\n",
      "Adding exchange reaction EX_trpthrglu_x with default bounds for boundary metabolite: trpthrglu_x.\n",
      "Adding exchange reaction EX_trpthrile_x with default bounds for boundary metabolite: trpthrile_x.\n",
      "Adding exchange reaction EX_trpthrtyr_x with default bounds for boundary metabolite: trpthrtyr_x.\n",
      "Adding exchange reaction EX_trptyrgln_x with default bounds for boundary metabolite: trptyrgln_x.\n",
      "Adding exchange reaction EX_trptyrtyr_x with default bounds for boundary metabolite: trptyrtyr_x.\n",
      "Adding exchange reaction EX_trpvalasp_x with default bounds for boundary metabolite: trpvalasp_x.\n",
      "Adding exchange reaction EX_tyrala_x with default bounds for boundary metabolite: tyrala_x.\n",
      "Adding exchange reaction EX_tyralaphe_x with default bounds for boundary metabolite: tyralaphe_x.\n",
      "Adding exchange reaction EX_tyrargglu_x with default bounds for boundary metabolite: tyrargglu_x.\n",
      "Adding exchange reaction EX_tyrargser_x with default bounds for boundary metabolite: tyrargser_x.\n",
      "Adding exchange reaction EX_tyrasparg_x with default bounds for boundary metabolite: tyrasparg_x.\n",
      "Adding exchange reaction EX_tyrcysgly_x with default bounds for boundary metabolite: tyrcysgly_x.\n",
      "Adding exchange reaction EX_tyrcysthr_x with default bounds for boundary metabolite: tyrcysthr_x.\n",
      "Adding exchange reaction EX_tyrglu_x with default bounds for boundary metabolite: tyrglu_x.\n",
      "Adding exchange reaction EX_tyrleuarg_x with default bounds for boundary metabolite: tyrleuarg_x.\n",
      "Adding exchange reaction EX_tyrphetyr_x with default bounds for boundary metabolite: tyrphetyr_x.\n",
      "Adding exchange reaction EX_tyrthr_x with default bounds for boundary metabolite: tyrthr_x.\n",
      "Adding exchange reaction EX_tyrtrpphe_x with default bounds for boundary metabolite: tyrtrpphe_x.\n",
      "Adding exchange reaction EX_tyrtyr_x with default bounds for boundary metabolite: tyrtyr_x.\n",
      "Adding exchange reaction EX_tyrvalmet_x with default bounds for boundary metabolite: tyrvalmet_x.\n",
      "Adding exchange reaction EX_m03107x with default bounds for boundary metabolite: m03107x.\n",
      "Adding exchange reaction EX_m03108x with default bounds for boundary metabolite: m03108x.\n",
      "Adding exchange reaction EX_m03109x with default bounds for boundary metabolite: m03109x.\n",
      "Adding exchange reaction EX_uchol_x with default bounds for boundary metabolite: uchol_x.\n",
      "Adding exchange reaction EX_udca3s_x with default bounds for boundary metabolite: udca3s_x.\n",
      "Adding exchange reaction EX_valarggly_x with default bounds for boundary metabolite: valarggly_x.\n",
      "Adding exchange reaction EX_valhisasn_x with default bounds for boundary metabolite: valhisasn_x.\n",
      "Adding exchange reaction EX_valleuphe_x with default bounds for boundary metabolite: valleuphe_x.\n",
      "Adding exchange reaction EX_vallystyr_x with default bounds for boundary metabolite: vallystyr_x.\n",
      "Adding exchange reaction EX_valphearg_x with default bounds for boundary metabolite: valphearg_x.\n",
      "Adding exchange reaction EX_valprotrp_x with default bounds for boundary metabolite: valprotrp_x.\n",
      "Adding exchange reaction EX_valserarg_x with default bounds for boundary metabolite: valserarg_x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_valtrpphe_x with default bounds for boundary metabolite: valtrpphe_x.\n",
      "Adding exchange reaction EX_valtrpval_x with default bounds for boundary metabolite: valtrpval_x.\n",
      "Adding exchange reaction EX_valval_x with default bounds for boundary metabolite: valval_x.\n",
      "Adding exchange reaction EX_vanillac_x with default bounds for boundary metabolite: vanillac_x.\n",
      "Adding exchange reaction EX_vldl_hs_x with default bounds for boundary metabolite: vldl_hs_x.\n",
      "Adding exchange reaction EX_m00186x with default bounds for boundary metabolite: m00186x.\n",
      "Adding exchange reaction EX_acmpglut_x with default bounds for boundary metabolite: acmpglut_x.\n",
      "Adding exchange reaction EX_meracmp_x with default bounds for boundary metabolite: meracmp_x.\n",
      "Adding exchange reaction EX_acmp_x with default bounds for boundary metabolite: acmp_x.\n",
      "Adding exchange reaction EX_nfdac_x with default bounds for boundary metabolite: nfdac_x.\n",
      "Adding exchange reaction EX_m01268x with default bounds for boundary metabolite: m01268x.\n",
      "Adding exchange reaction EX_m01291x with default bounds for boundary metabolite: m01291x.\n",
      "Adding exchange reaction EX_m01313x with default bounds for boundary metabolite: m01313x.\n",
      "Adding exchange reaction EX_m01314x with default bounds for boundary metabolite: m01314x.\n",
      "Adding exchange reaction EX_allop_x with default bounds for boundary metabolite: allop_x.\n",
      "Adding exchange reaction EX_m01315x with default bounds for boundary metabolite: m01315x.\n",
      "Adding exchange reaction EX_m01321x with default bounds for boundary metabolite: m01321x.\n",
      "Adding exchange reaction EX_m01332x with default bounds for boundary metabolite: m01332x.\n",
      "Adding exchange reaction EX_m01335x with default bounds for boundary metabolite: m01335x.\n",
      "Adding exchange reaction EX_m01337x with default bounds for boundary metabolite: m01337x.\n",
      "Adding exchange reaction EX_m01342x with default bounds for boundary metabolite: m01342x.\n",
      "Adding exchange reaction EX_m01343x with default bounds for boundary metabolite: m01343x.\n",
      "Adding exchange reaction EX_m01345x with default bounds for boundary metabolite: m01345x.\n",
      "Adding exchange reaction EX_m01347x with default bounds for boundary metabolite: m01347x.\n",
      "Adding exchange reaction EX_m01348x with default bounds for boundary metabolite: m01348x.\n",
      "Adding exchange reaction EX_m01350x with default bounds for boundary metabolite: m01350x.\n",
      "Adding exchange reaction EX_m01351x with default bounds for boundary metabolite: m01351x.\n",
      "Adding exchange reaction EX_m01353x with default bounds for boundary metabolite: m01353x.\n",
      "Adding exchange reaction EX_m01354x with default bounds for boundary metabolite: m01354x.\n",
      "Adding exchange reaction EX_m01355x with default bounds for boundary metabolite: m01355x.\n",
      "Adding exchange reaction EX_m01359x with default bounds for boundary metabolite: m01359x.\n",
      "Adding exchange reaction EX_m01362x with default bounds for boundary metabolite: m01362x.\n",
      "Adding exchange reaction EX_m01366x with default bounds for boundary metabolite: m01366x.\n",
      "Adding exchange reaction EX_atvacid_x with default bounds for boundary metabolite: atvacid_x.\n",
      "Adding exchange reaction EX_atvlac_x with default bounds for boundary metabolite: atvlac_x.\n",
      "Adding exchange reaction EX_m01372x with default bounds for boundary metabolite: m01372x.\n",
      "Adding exchange reaction EX_m01373x with default bounds for boundary metabolite: m01373x.\n",
      "Adding exchange reaction EX_m01380x with default bounds for boundary metabolite: m01380x.\n",
      "Adding exchange reaction EX_m01387x with default bounds for boundary metabolite: m01387x.\n",
      "Adding exchange reaction EX_m01386x with default bounds for boundary metabolite: m01386x.\n",
      "Adding exchange reaction EX_m01392x with default bounds for boundary metabolite: m01392x.\n",
      "Adding exchange reaction EX_m01395x with default bounds for boundary metabolite: m01395x.\n",
      "Adding exchange reaction EX_m01399x with default bounds for boundary metabolite: m01399x.\n",
      "Adding exchange reaction EX_m01417x with default bounds for boundary metabolite: m01417x.\n",
      "Adding exchange reaction EX_m01421x with default bounds for boundary metabolite: m01421x.\n",
      "Adding exchange reaction EX_crglz_x with default bounds for boundary metabolite: crglz_x.\n",
      "Adding exchange reaction EX_m01423x with default bounds for boundary metabolite: m01423x.\n",
      "Adding exchange reaction EX_m01430x with default bounds for boundary metabolite: m01430x.\n",
      "Adding exchange reaction EX_m01431x with default bounds for boundary metabolite: m01431x.\n",
      "Adding exchange reaction EX_crvs_x with default bounds for boundary metabolite: crvs_x.\n",
      "Adding exchange reaction EX_crvsm1_x with default bounds for boundary metabolite: crvsm1_x.\n",
      "Adding exchange reaction EX_crvsm23_x with default bounds for boundary metabolite: crvsm23_x.\n",
      "Adding exchange reaction EX_crvsm24_x with default bounds for boundary metabolite: crvsm24_x.\n",
      "Adding exchange reaction EX_m01432x with default bounds for boundary metabolite: m01432x.\n",
      "Adding exchange reaction EX_m01439x with default bounds for boundary metabolite: m01439x.\n",
      "Adding exchange reaction EX_m01446x with default bounds for boundary metabolite: m01446x.\n",
      "Adding exchange reaction EX_m01447x with default bounds for boundary metabolite: m01447x.\n",
      "Adding exchange reaction EX_m01448x with default bounds for boundary metabolite: m01448x.\n",
      "Adding exchange reaction EX_m01504x with default bounds for boundary metabolite: m01504x.\n",
      "Adding exchange reaction EX_m01512x with default bounds for boundary metabolite: m01512x.\n",
      "Adding exchange reaction EX_m01571x with default bounds for boundary metabolite: m01571x.\n",
      "Adding exchange reaction EX_m01580x with default bounds for boundary metabolite: m01580x.\n",
      "Adding exchange reaction EX_m01582x with default bounds for boundary metabolite: m01582x.\n",
      "Adding exchange reaction EX_m01583x with default bounds for boundary metabolite: m01583x.\n",
      "Adding exchange reaction EX_m01584x with default bounds for boundary metabolite: m01584x.\n",
      "Adding exchange reaction EX_m01585x with default bounds for boundary metabolite: m01585x.\n",
      "Adding exchange reaction EX_m01600x with default bounds for boundary metabolite: m01600x.\n",
      "Adding exchange reaction EX_m01601x with default bounds for boundary metabolite: m01601x.\n",
      "Adding exchange reaction EX_m01605x with default bounds for boundary metabolite: m01605x.\n",
      "Adding exchange reaction EX_m01609x with default bounds for boundary metabolite: m01609x.\n",
      "Adding exchange reaction EX_m01616x with default bounds for boundary metabolite: m01616x.\n",
      "Adding exchange reaction EX_m01620x with default bounds for boundary metabolite: m01620x.\n",
      "Adding exchange reaction EX_csa_x with default bounds for boundary metabolite: csa_x.\n",
      "Adding exchange reaction EX_csasulp_x with default bounds for boundary metabolite: csasulp_x.\n",
      "Adding exchange reaction EX_m01627x with default bounds for boundary metabolite: m01627x.\n",
      "Adding exchange reaction EX_cysacmp_x with default bounds for boundary metabolite: cysacmp_x.\n",
      "Adding exchange reaction EX_m01644x with default bounds for boundary metabolite: m01644x.\n",
      "Adding exchange reaction EX_m01686x with default bounds for boundary metabolite: m01686x.\n",
      "Adding exchange reaction EX_m01688x with default bounds for boundary metabolite: m01688x.\n",
      "Adding exchange reaction EX_m01747x with default bounds for boundary metabolite: m01747x.\n",
      "Adding exchange reaction EX_m01752x with default bounds for boundary metabolite: m01752x.\n",
      "Adding exchange reaction EX_m01753x with default bounds for boundary metabolite: m01753x.\n",
      "Adding exchange reaction EX_m01651x with default bounds for boundary metabolite: m01651x.\n",
      "Adding exchange reaction EX_dhglz_x with default bounds for boundary metabolite: dhglz_x.\n",
      "Adding exchange reaction EX_m01660x with default bounds for boundary metabolite: m01660x.\n",
      "Adding exchange reaction EX_deoxfvs_x with default bounds for boundary metabolite: deoxfvs_x.\n",
      "Adding exchange reaction EX_m01674x with default bounds for boundary metabolite: m01674x.\n",
      "Adding exchange reaction EX_fvstet_x with default bounds for boundary metabolite: fvstet_x.\n",
      "Adding exchange reaction EX_fvstetglu_x with default bounds for boundary metabolite: fvstetglu_x.\n",
      "Adding exchange reaction EX_profvs_x with default bounds for boundary metabolite: profvs_x.\n",
      "Adding exchange reaction EX_dspvs_x with default bounds for boundary metabolite: dspvs_x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_m01675x with default bounds for boundary metabolite: m01675x.\n",
      "Adding exchange reaction EX_m01696x with default bounds for boundary metabolite: m01696x.\n",
      "Adding exchange reaction EX_m01698x with default bounds for boundary metabolite: m01698x.\n",
      "Adding exchange reaction EX_m01705x with default bounds for boundary metabolite: m01705x.\n",
      "Adding exchange reaction EX_m01708x with default bounds for boundary metabolite: m01708x.\n",
      "Adding exchange reaction EX_m01767x with default bounds for boundary metabolite: m01767x.\n",
      "Adding exchange reaction EX_m01771x with default bounds for boundary metabolite: m01771x.\n",
      "Adding exchange reaction EX_m01773x with default bounds for boundary metabolite: m01773x.\n",
      "Adding exchange reaction EX_m01778x with default bounds for boundary metabolite: m01778x.\n",
      "Adding exchange reaction EX_m01790x with default bounds for boundary metabolite: m01790x.\n",
      "Adding exchange reaction EX_m01798x with default bounds for boundary metabolite: m01798x.\n",
      "Adding exchange reaction EX_m01820x with default bounds for boundary metabolite: m01820x.\n",
      "Adding exchange reaction EX_m01807x with default bounds for boundary metabolite: m01807x.\n",
      "Adding exchange reaction EX_m01819x with default bounds for boundary metabolite: m01819x.\n",
      "Adding exchange reaction EX_m01827x with default bounds for boundary metabolite: m01827x.\n",
      "Adding exchange reaction EX_fvs_x with default bounds for boundary metabolite: fvs_x.\n",
      "Adding exchange reaction EX_m01831x with default bounds for boundary metabolite: m01831x.\n",
      "Adding exchange reaction EX_m01839x with default bounds for boundary metabolite: m01839x.\n",
      "Adding exchange reaction EX_m01841x with default bounds for boundary metabolite: m01841x.\n",
      "Adding exchange reaction EX_m01862x with default bounds for boundary metabolite: m01862x.\n",
      "Adding exchange reaction EX_m01909x with default bounds for boundary metabolite: m01909x.\n",
      "Adding exchange reaction EX_m01924x with default bounds for boundary metabolite: m01924x.\n",
      "Adding exchange reaction EX_m01922x with default bounds for boundary metabolite: m01922x.\n",
      "Adding exchange reaction EX_m01923x with default bounds for boundary metabolite: m01923x.\n",
      "Adding exchange reaction EX_m01925x with default bounds for boundary metabolite: m01925x.\n",
      "Adding exchange reaction EX_m01926x with default bounds for boundary metabolite: m01926x.\n",
      "Adding exchange reaction EX_m01927x with default bounds for boundary metabolite: m01927x.\n",
      "Adding exchange reaction EX_m01932x with default bounds for boundary metabolite: m01932x.\n",
      "Adding exchange reaction EX_m01954x with default bounds for boundary metabolite: m01954x.\n",
      "Adding exchange reaction EX_m01967x with default bounds for boundary metabolite: m01967x.\n",
      "Adding exchange reaction EX_m01972x with default bounds for boundary metabolite: m01972x.\n",
      "Adding exchange reaction EX_m01973x with default bounds for boundary metabolite: m01973x.\n",
      "Adding exchange reaction EX_glc3meacp_x with default bounds for boundary metabolite: glc3meacp_x.\n",
      "Adding exchange reaction EX_gtacmp_x with default bounds for boundary metabolite: gtacmp_x.\n",
      "Adding exchange reaction EX_m01980x with default bounds for boundary metabolite: m01980x.\n",
      "Adding exchange reaction EX_m01981x with default bounds for boundary metabolite: m01981x.\n",
      "Adding exchange reaction EX_m01984x with default bounds for boundary metabolite: m01984x.\n",
      "Adding exchange reaction EX_m01991x with default bounds for boundary metabolite: m01991x.\n",
      "Adding exchange reaction EX_m01992x with default bounds for boundary metabolite: m01992x.\n",
      "Adding exchange reaction EX_m01998x with default bounds for boundary metabolite: m01998x.\n",
      "Adding exchange reaction EX_m02000x with default bounds for boundary metabolite: m02000x.\n",
      "Adding exchange reaction EX_m02004x with default bounds for boundary metabolite: m02004x.\n",
      "Adding exchange reaction EX_m02007x with default bounds for boundary metabolite: m02007x.\n",
      "Adding exchange reaction EX_m02035x with default bounds for boundary metabolite: m02035x.\n",
      "Adding exchange reaction EX_m02036x with default bounds for boundary metabolite: m02036x.\n",
      "Adding exchange reaction EX_m02044x with default bounds for boundary metabolite: m02044x.\n",
      "Adding exchange reaction EX_m02053x with default bounds for boundary metabolite: m02053x.\n",
      "Adding exchange reaction EX_m02096x with default bounds for boundary metabolite: m02096x.\n",
      "Adding exchange reaction EX_m02117x with default bounds for boundary metabolite: m02117x.\n",
      "Adding exchange reaction EX_m02122x with default bounds for boundary metabolite: m02122x.\n",
      "Adding exchange reaction EX_m02123x with default bounds for boundary metabolite: m02123x.\n",
      "Adding exchange reaction EX_m02124x with default bounds for boundary metabolite: m02124x.\n",
      "Adding exchange reaction EX_m02133x with default bounds for boundary metabolite: m02133x.\n",
      "Adding exchange reaction EX_m02134x with default bounds for boundary metabolite: m02134x.\n",
      "Adding exchange reaction EX_m02135x with default bounds for boundary metabolite: m02135x.\n",
      "Adding exchange reaction EX_m02142x with default bounds for boundary metabolite: m02142x.\n",
      "Adding exchange reaction EX_nfdoh_x with default bounds for boundary metabolite: nfdoh_x.\n",
      "Adding exchange reaction EX_m02156x with default bounds for boundary metabolite: m02156x.\n",
      "Adding exchange reaction EX_m02157x with default bounds for boundary metabolite: m02157x.\n",
      "Adding exchange reaction EX_m02158x with default bounds for boundary metabolite: m02158x.\n",
      "Adding exchange reaction EX_ibupgluc_x with default bounds for boundary metabolite: ibupgluc_x.\n",
      "Adding exchange reaction EX_ibup_R_x with default bounds for boundary metabolite: ibup_R_x.\n",
      "Adding exchange reaction EX_ibup_S_x with default bounds for boundary metabolite: ibup_S_x.\n",
      "Adding exchange reaction EX_m02166x with default bounds for boundary metabolite: m02166x.\n",
      "Adding exchange reaction EX_m02169x with default bounds for boundary metabolite: m02169x.\n",
      "Adding exchange reaction EX_m02173x with default bounds for boundary metabolite: m02173x.\n",
      "Adding exchange reaction EX_m02181x with default bounds for boundary metabolite: m02181x.\n",
      "Adding exchange reaction EX_m02183x with default bounds for boundary metabolite: m02183x.\n",
      "Adding exchange reaction EX_m02185x with default bounds for boundary metabolite: m02185x.\n",
      "Adding exchange reaction EX_m02190x with default bounds for boundary metabolite: m02190x.\n",
      "Adding exchange reaction EX_m02318x with default bounds for boundary metabolite: m02318x.\n",
      "Adding exchange reaction EX_m02314x with default bounds for boundary metabolite: m02314x.\n",
      "Adding exchange reaction EX_m02315x with default bounds for boundary metabolite: m02315x.\n",
      "Adding exchange reaction EX_m02316x with default bounds for boundary metabolite: m02316x.\n",
      "Adding exchange reaction EX_m02317x with default bounds for boundary metabolite: m02317x.\n",
      "Adding exchange reaction EX_m02319x with default bounds for boundary metabolite: m02319x.\n",
      "Adding exchange reaction EX_nfdlac_x with default bounds for boundary metabolite: nfdlac_x.\n",
      "Adding exchange reaction EX_m02333x with default bounds for boundary metabolite: m02333x.\n",
      "Adding exchange reaction EX_m02336x with default bounds for boundary metabolite: m02336x.\n",
      "Adding exchange reaction EX_m02343x with default bounds for boundary metabolite: m02343x.\n",
      "Adding exchange reaction EX_m02344x with default bounds for boundary metabolite: m02344x.\n",
      "Adding exchange reaction EX_m02365x with default bounds for boundary metabolite: m02365x.\n",
      "Adding exchange reaction EX_m02385x with default bounds for boundary metabolite: m02385x.\n",
      "Adding exchange reaction EX_m02388x with default bounds for boundary metabolite: m02388x.\n",
      "Adding exchange reaction EX_m02393x with default bounds for boundary metabolite: m02393x.\n",
      "Adding exchange reaction EX_m02395x with default bounds for boundary metabolite: m02395x.\n",
      "Adding exchange reaction EX_m02396x with default bounds for boundary metabolite: m02396x.\n",
      "Adding exchange reaction EX_m02402x with default bounds for boundary metabolite: m02402x.\n",
      "Adding exchange reaction EX_lvst_x with default bounds for boundary metabolite: lvst_x.\n",
      "Adding exchange reaction EX_lvstacid_x with default bounds for boundary metabolite: lvstacid_x.\n",
      "Adding exchange reaction EX_m02429x with default bounds for boundary metabolite: m02429x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_m02439x with default bounds for boundary metabolite: m02439x.\n",
      "Adding exchange reaction EX_m02441x with default bounds for boundary metabolite: m02441x.\n",
      "Adding exchange reaction EX_m02444x with default bounds for boundary metabolite: m02444x.\n",
      "Adding exchange reaction EX_m02446x with default bounds for boundary metabolite: m02446x.\n",
      "Adding exchange reaction EX_m02447x with default bounds for boundary metabolite: m02447x.\n",
      "Adding exchange reaction EX_m02448x with default bounds for boundary metabolite: m02448x.\n",
      "Adding exchange reaction EX_m02449x with default bounds for boundary metabolite: m02449x.\n",
      "Adding exchange reaction EX_m02451x with default bounds for boundary metabolite: m02451x.\n",
      "Adding exchange reaction EX_m02456x with default bounds for boundary metabolite: m02456x.\n",
      "Adding exchange reaction EX_m02457x with default bounds for boundary metabolite: m02457x.\n",
      "Adding exchange reaction EX_m02460x with default bounds for boundary metabolite: m02460x.\n",
      "Adding exchange reaction EX_m02467x with default bounds for boundary metabolite: m02467x.\n",
      "Adding exchange reaction EX_m02473x with default bounds for boundary metabolite: m02473x.\n",
      "Adding exchange reaction EX_m02479x with default bounds for boundary metabolite: m02479x.\n",
      "Adding exchange reaction EX_m02494x with default bounds for boundary metabolite: m02494x.\n",
      "Adding exchange reaction EX_m02563x with default bounds for boundary metabolite: m02563x.\n",
      "Adding exchange reaction EX_m02562x with default bounds for boundary metabolite: m02562x.\n",
      "Adding exchange reaction EX_m02564x with default bounds for boundary metabolite: m02564x.\n",
      "Adding exchange reaction EX_m02570x with default bounds for boundary metabolite: m02570x.\n",
      "Adding exchange reaction EX_m02569x with default bounds for boundary metabolite: m02569x.\n",
      "Adding exchange reaction EX_m02585x with default bounds for boundary metabolite: m02585x.\n",
      "Adding exchange reaction EX_m02591x with default bounds for boundary metabolite: m02591x.\n",
      "Adding exchange reaction EX_m02613x with default bounds for boundary metabolite: m02613x.\n",
      "Adding exchange reaction EX_m02622x with default bounds for boundary metabolite: m02622x.\n",
      "Adding exchange reaction EX_m02639x with default bounds for boundary metabolite: m02639x.\n",
      "Adding exchange reaction EX_m02646x with default bounds for boundary metabolite: m02646x.\n",
      "Adding exchange reaction EX_m02648x with default bounds for boundary metabolite: m02648x.\n",
      "Adding exchange reaction EX_m02660x with default bounds for boundary metabolite: m02660x.\n",
      "Adding exchange reaction EX_oxyp_x with default bounds for boundary metabolite: oxyp_x.\n",
      "Adding exchange reaction EX_oxy1rb_x with default bounds for boundary metabolite: oxy1rb_x.\n",
      "Adding exchange reaction EX_oxy7rb_x with default bounds for boundary metabolite: oxy7rb_x.\n",
      "Adding exchange reaction EX_m02674x with default bounds for boundary metabolite: m02674x.\n",
      "Adding exchange reaction EX_m02675x with default bounds for boundary metabolite: m02675x.\n",
      "Adding exchange reaction EX_m02676x with default bounds for boundary metabolite: m02676x.\n",
      "Adding exchange reaction EX_m02679x with default bounds for boundary metabolite: m02679x.\n",
      "Adding exchange reaction EX_m02690x with default bounds for boundary metabolite: m02690x.\n",
      "Adding exchange reaction EX_m02714x with default bounds for boundary metabolite: m02714x.\n",
      "Adding exchange reaction EX_m02720x with default bounds for boundary metabolite: m02720x.\n",
      "Adding exchange reaction EX_m02725x with default bounds for boundary metabolite: m02725x.\n",
      "Adding exchange reaction EX_m02733x with default bounds for boundary metabolite: m02733x.\n",
      "Adding exchange reaction EX_m02738x with default bounds for boundary metabolite: m02738x.\n",
      "Adding exchange reaction EX_m02741x with default bounds for boundary metabolite: m02741x.\n",
      "Adding exchange reaction EX_m02745x with default bounds for boundary metabolite: m02745x.\n",
      "Adding exchange reaction EX_m02752x with default bounds for boundary metabolite: m02752x.\n",
      "Adding exchange reaction EX_ptvst_x with default bounds for boundary metabolite: ptvst_x.\n",
      "Adding exchange reaction EX_ptvstm13_x with default bounds for boundary metabolite: ptvstm13_x.\n",
      "Adding exchange reaction EX_ptvstm3_x with default bounds for boundary metabolite: ptvstm3_x.\n",
      "Adding exchange reaction EX_ptvstlac_x with default bounds for boundary metabolite: ptvstlac_x.\n",
      "Adding exchange reaction EX_m02753x with default bounds for boundary metabolite: m02753x.\n",
      "Adding exchange reaction EX_m02756x with default bounds for boundary metabolite: m02756x.\n",
      "Adding exchange reaction EX_pvs_x with default bounds for boundary metabolite: pvs_x.\n",
      "Adding exchange reaction EX_pvsgluc_x with default bounds for boundary metabolite: pvsgluc_x.\n",
      "Adding exchange reaction EX_m02763x with default bounds for boundary metabolite: m02763x.\n",
      "Adding exchange reaction EX_m02762x with default bounds for boundary metabolite: m02762x.\n",
      "Adding exchange reaction EX_m02766x with default bounds for boundary metabolite: m02766x.\n",
      "Adding exchange reaction EX_m02771x with default bounds for boundary metabolite: m02771x.\n",
      "Adding exchange reaction EX_m02776x with default bounds for boundary metabolite: m02776x.\n",
      "Adding exchange reaction EX_m02777x with default bounds for boundary metabolite: m02777x.\n",
      "Adding exchange reaction EX_m02778x with default bounds for boundary metabolite: m02778x.\n",
      "Adding exchange reaction EX_m02779x with default bounds for boundary metabolite: m02779x.\n",
      "Adding exchange reaction EX_m02780x with default bounds for boundary metabolite: m02780x.\n",
      "Adding exchange reaction EX_m02781x with default bounds for boundary metabolite: m02781x.\n",
      "Adding exchange reaction EX_m02782x with default bounds for boundary metabolite: m02782x.\n",
      "Adding exchange reaction EX_m02784x with default bounds for boundary metabolite: m02784x.\n",
      "Adding exchange reaction EX_m02787x with default bounds for boundary metabolite: m02787x.\n",
      "Adding exchange reaction EX_m02788x with default bounds for boundary metabolite: m02788x.\n",
      "Adding exchange reaction EX_m02790x with default bounds for boundary metabolite: m02790x.\n",
      "Adding exchange reaction EX_m02792x with default bounds for boundary metabolite: m02792x.\n",
      "Adding exchange reaction EX_m02794x with default bounds for boundary metabolite: m02794x.\n",
      "Adding exchange reaction EX_m02795x with default bounds for boundary metabolite: m02795x.\n",
      "Adding exchange reaction EX_m02796x with default bounds for boundary metabolite: m02796x.\n",
      "Adding exchange reaction EX_m02802x with default bounds for boundary metabolite: m02802x.\n",
      "Adding exchange reaction EX_m02803x with default bounds for boundary metabolite: m02803x.\n",
      "Adding exchange reaction EX_m02805x with default bounds for boundary metabolite: m02805x.\n",
      "Adding exchange reaction EX_m02812x with default bounds for boundary metabolite: m02812x.\n",
      "Adding exchange reaction EX_m02821x with default bounds for boundary metabolite: m02821x.\n",
      "Adding exchange reaction EX_m02822x with default bounds for boundary metabolite: m02822x.\n",
      "Adding exchange reaction EX_m02823x with default bounds for boundary metabolite: m02823x.\n",
      "Adding exchange reaction EX_m02832x with default bounds for boundary metabolite: m02832x.\n",
      "Adding exchange reaction EX_m02837x with default bounds for boundary metabolite: m02837x.\n",
      "Adding exchange reaction EX_m02839x with default bounds for boundary metabolite: m02839x.\n",
      "Adding exchange reaction EX_rsv_x with default bounds for boundary metabolite: rsv_x.\n",
      "Adding exchange reaction EX_rsvlac_x with default bounds for boundary metabolite: rsvlac_x.\n",
      "Adding exchange reaction EX_m02868x with default bounds for boundary metabolite: m02868x.\n",
      "Adding exchange reaction EX_m02875x with default bounds for boundary metabolite: m02875x.\n",
      "Adding exchange reaction EX_m02891x with default bounds for boundary metabolite: m02891x.\n",
      "Adding exchange reaction EX_smvacid_x with default bounds for boundary metabolite: smvacid_x.\n",
      "Adding exchange reaction EX_smv_x with default bounds for boundary metabolite: smv_x.\n",
      "Adding exchange reaction EX_m02912x with default bounds for boundary metabolite: m02912x.\n",
      "Adding exchange reaction EX_m02914x with default bounds for boundary metabolite: m02914x.\n",
      "Adding exchange reaction EX_m02923x with default bounds for boundary metabolite: m02923x.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_m02920x with default bounds for boundary metabolite: m02920x.\n",
      "Adding exchange reaction EX_m02921x with default bounds for boundary metabolite: m02921x.\n",
      "Adding exchange reaction EX_m02922x with default bounds for boundary metabolite: m02922x.\n",
      "Adding exchange reaction EX_m02925x with default bounds for boundary metabolite: m02925x.\n",
      "Adding exchange reaction EX_m02927x with default bounds for boundary metabolite: m02927x.\n",
      "Adding exchange reaction EX_m02929x with default bounds for boundary metabolite: m02929x.\n",
      "Adding exchange reaction EX_m02933x with default bounds for boundary metabolite: m02933x.\n",
      "Adding exchange reaction EX_m02938x with default bounds for boundary metabolite: m02938x.\n",
      "Adding exchange reaction EX_m02939x with default bounds for boundary metabolite: m02939x.\n",
      "Adding exchange reaction EX_m02940x with default bounds for boundary metabolite: m02940x.\n",
      "Adding exchange reaction EX_m02942x with default bounds for boundary metabolite: m02942x.\n",
      "Adding exchange reaction EX_m02950x with default bounds for boundary metabolite: m02950x.\n",
      "Adding exchange reaction EX_m02951x with default bounds for boundary metabolite: m02951x.\n",
      "Adding exchange reaction EX_m02952x with default bounds for boundary metabolite: m02952x.\n",
      "Adding exchange reaction EX_stacmp_x with default bounds for boundary metabolite: stacmp_x.\n",
      "Adding exchange reaction EX_s3meacmp_x with default bounds for boundary metabolite: s3meacmp_x.\n",
      "Adding exchange reaction EX_sulpacmp_x with default bounds for boundary metabolite: sulpacmp_x.\n",
      "Adding exchange reaction EX_tauribup_S_x with default bounds for boundary metabolite: tauribup_S_x.\n",
      "Adding exchange reaction EX_m02964x with default bounds for boundary metabolite: m02964x.\n",
      "Adding exchange reaction EX_m02965x with default bounds for boundary metabolite: m02965x.\n",
      "Adding exchange reaction EX_m02966x with default bounds for boundary metabolite: m02966x.\n",
      "Adding exchange reaction EX_m02974x with default bounds for boundary metabolite: m02974x.\n",
      "Adding exchange reaction EX_m02978x with default bounds for boundary metabolite: m02978x.\n",
      "Adding exchange reaction EX_tsacmgluc_x with default bounds for boundary metabolite: tsacmgluc_x.\n",
      "Adding exchange reaction EX_tsacmsul_x with default bounds for boundary metabolite: tsacmsul_x.\n",
      "Adding exchange reaction EX_thrfvs_x with default bounds for boundary metabolite: thrfvs_x.\n",
      "Adding exchange reaction EX_m02992x with default bounds for boundary metabolite: m02992x.\n",
      "Adding exchange reaction EX_m02995x with default bounds for boundary metabolite: m02995x.\n",
      "Adding exchange reaction EX_m03037x with default bounds for boundary metabolite: m03037x.\n",
      "Adding exchange reaction EX_tlacfvs_x with default bounds for boundary metabolite: tlacfvs_x.\n",
      "Adding exchange reaction EX_m03045x with default bounds for boundary metabolite: m03045x.\n",
      "Adding exchange reaction EX_m03051x with default bounds for boundary metabolite: m03051x.\n",
      "Adding exchange reaction EX_tripvs_x with default bounds for boundary metabolite: tripvs_x.\n",
      "Adding exchange reaction EX_m03088x with default bounds for boundary metabolite: m03088x.\n",
      "Adding exchange reaction EX_m03099x with default bounds for boundary metabolite: m03099x.\n",
      "Adding exchange reaction EX_m03102x with default bounds for boundary metabolite: m03102x.\n",
      "Adding exchange reaction EX_m03103x with default bounds for boundary metabolite: m03103x.\n",
      "Adding exchange reaction EX_m03124x with default bounds for boundary metabolite: m03124x.\n",
      "Adding exchange reaction EX_m03125x with default bounds for boundary metabolite: m03125x.\n",
      "Adding exchange reaction EX_m03127x with default bounds for boundary metabolite: m03127x.\n",
      "Adding exchange reaction EX_m03129x with default bounds for boundary metabolite: m03129x.\n",
      "Adding exchange reaction EX_m03136x with default bounds for boundary metabolite: m03136x.\n",
      "Adding exchange reaction EX_m03148x with default bounds for boundary metabolite: m03148x.\n",
      "Adding exchange reaction EX_m03149x with default bounds for boundary metabolite: m03149x.\n",
      "Adding exchange reaction EX_m03150x with default bounds for boundary metabolite: m03150x.\n",
      "Adding exchange reaction EX_m03153x with default bounds for boundary metabolite: m03153x.\n",
      "Adding exchange reaction EX_m00591x with default bounds for boundary metabolite: m00591x.\n",
      "Adding exchange reaction EX_m01435x with default bounds for boundary metabolite: m01435x.\n",
      "Adding exchange reaction EX_m02328x with default bounds for boundary metabolite: m02328x.\n",
      "Adding exchange reaction EX_chylo_hs_x with default bounds for boundary metabolite: chylo_hs_x.\n",
      "Adding exchange reaction EX_m10005x with default bounds for boundary metabolite: m10005x.\n",
      "Adding exchange reaction EX_m10006x with default bounds for boundary metabolite: m10006x.\n",
      "Adding exchange reaction EX_m10011x with default bounds for boundary metabolite: m10011x.\n",
      "Adding exchange reaction EX_m00235x with default bounds for boundary metabolite: m00235x.\n",
      "Adding exchange reaction EX_m02838x with default bounds for boundary metabolite: m02838x.\n",
      "Adding exchange reaction EX_m10021x with default bounds for boundary metabolite: m10021x.\n",
      "Adding exchange reaction EX_m10022x with default bounds for boundary metabolite: m10022x.\n",
      "Adding exchange reaction EX_m10023x with default bounds for boundary metabolite: m10023x.\n",
      "Adding exchange reaction EX_m10024x with default bounds for boundary metabolite: m10024x.\n",
      "Adding exchange reaction EX_m10025x with default bounds for boundary metabolite: m10025x.\n",
      "Adding exchange reaction EX_m10026x with default bounds for boundary metabolite: m10026x.\n",
      "Adding exchange reaction EX_m10027x with default bounds for boundary metabolite: m10027x.\n",
      "Adding exchange reaction EX_m10028x with default bounds for boundary metabolite: m10028x.\n",
      "Adding exchange reaction EX_m10029x with default bounds for boundary metabolite: m10029x.\n",
      "Adding exchange reaction EX_m10030x with default bounds for boundary metabolite: m10030x.\n",
      "Adding exchange reaction EX_m10031x with default bounds for boundary metabolite: m10031x.\n",
      "Adding exchange reaction EX_m10032x with default bounds for boundary metabolite: m10032x.\n",
      "Adding exchange reaction EX_m10033x with default bounds for boundary metabolite: m10033x.\n",
      "Adding exchange reaction EX_m10034x with default bounds for boundary metabolite: m10034x.\n",
      "Adding exchange reaction EX_m10041x with default bounds for boundary metabolite: m10041x.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07fbf4184f8d0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>10073</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>14770</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>145</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass_human - 1.0*biomass_human_reverse_fb2f2</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>Cytosol, Extracellular, Lysosome, Endoplasmic reticulum, Mitochondria, Peroxisome, Golgi apparatus, Nucleus, Boundary, Inner mitochondria</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model  at 0x7fbf4184f8d0>"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "xmlFile = 'Human-GEM/ModelFiles/xml/HumanGEM.xml'\n",
    "excelFile = 'Human-GEM/ModelFiles/xlsx/HumanGEM.xlsx'\n",
    "\n",
    "model2 = cobra.io.read_sbml_model(xmlFile)\n",
    "\n",
    "model2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Metabolite identifier</strong></td><td>m00043c</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>(2E)-eicosenoyl-CoA</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07fbf4352fef0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Formula</strong></td><td>C41H68N7O17P3S</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartment</strong></td><td>c</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>In 2 reaction(s)</strong></td><td>\n",
       "                    HMR_2203, HMR_2204</td>\n",
       "            </tr>\n",
       "        </table>"
      ],
      "text/plain": [
       "<Metabolite m00043c at 0x7fbf4352fef0>"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model2.metabolites[99]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['__class__',\n",
       " '__delattr__',\n",
       " '__dict__',\n",
       " '__dir__',\n",
       " '__doc__',\n",
       " '__eq__',\n",
       " '__format__',\n",
       " '__ge__',\n",
       " '__getattribute__',\n",
       " '__getstate__',\n",
       " '__gt__',\n",
       " '__hash__',\n",
       " '__init__',\n",
       " '__init_subclass__',\n",
       " '__le__',\n",
       " '__lt__',\n",
       " '__module__',\n",
       " '__ne__',\n",
       " '__new__',\n",
       " '__reduce__',\n",
       " '__reduce_ex__',\n",
       " '__repr__',\n",
       " '__setattr__',\n",
       " '__sizeof__',\n",
       " '__str__',\n",
       " '__subclasshook__',\n",
       " '__weakref__',\n",
       " '_annotation',\n",
       " '_bound',\n",
       " '_id',\n",
       " '_model',\n",
       " '_reaction',\n",
       " '_repr_html_',\n",
       " '_set_id_with_model',\n",
       " 'annotation',\n",
       " 'charge',\n",
       " 'compartment',\n",
       " 'constraint',\n",
       " 'copy',\n",
       " 'elements',\n",
       " 'formula',\n",
       " 'formula_weight',\n",
       " 'id',\n",
       " 'model',\n",
       " 'name',\n",
       " 'notes',\n",
       " 'reactions',\n",
       " 'remove_from_model',\n",
       " 'shadow_price',\n",
       " 'summary',\n",
       " 'y']"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dir(model2.metabolites[99])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['m00103c', 'm00043c', 'C41H68N7O17P3S', {'sbo': 'SBO:0000247'}]"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def at_metabolite2compound(M):\n",
    "    # convert cobra Metabolite to metDataModel Compound\n",
    "    Cpd = Compound()\n",
    "    Cpd.src_id = M.id\n",
    "    Cpd.id = M.id           #[2:8]\n",
    "    Cpd.name = M.name\n",
    "    Cpd.charge = M.charge\n",
    "    Cpd.charged_formula = M.formula\n",
    "    Cpd.db_ids = M.annotation\n",
    "    return Cpd\n",
    "\n",
    "[at_metabolite2compound(model2.metabolites[199]).id,\n",
    " at_metabolite2compound(model2.metabolites[99]).src_id,\n",
    " at_metabolite2compound(model2.metabolites[99]).charged_formula,\n",
    " at_metabolite2compound(model2.metabolites[99]).db_ids,\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total,  10073 10073\n",
      "real,  10073 2680 10073\n"
     ]
    }
   ],
   "source": [
    "# list of Compounds\n",
    "atCpds = []\n",
    "anno2 = {}\n",
    "for M in model2.metabolites:\n",
    "    anno2[M.id] = M.annotation\n",
    "    atCpds.append(at_metabolite2compound(M))\n",
    "    \n",
    "print(\"total, \", len(atCpds), len(anno2))\n",
    "\n",
    "unique = set([C.id for C in atCpds])\n",
    "formula = set([C.charged_formula for C in atCpds if C.charged_formula])\n",
    "# a little tricker to test on dictionaries\n",
    "annotation = [v for v in anno2.values() if v]\n",
    "\n",
    "print(\"real, \", len(unique), len(formula), len(annotation))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['asnphecys_c',\n",
       " 'm03155s',\n",
       " 'pcholar_hs_c',\n",
       " 'm01819x',\n",
       " 'c12dc_p',\n",
       " 'm00333m',\n",
       " 'm02349s',\n",
       " 'm00630m',\n",
       " '35dhpvs_s']"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(unique)[:9]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**The ATLA format is not all compatible with cobra**\n",
    "\n",
    "Do something else for now:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dict_keys(['mets', 'metsNoComp', 'metBiGGID', 'metKEGGID', 'metHMDBID', 'metChEBIID', 'metPubChemID', 'metLipidMapsID', 'metEHMNID', 'metHepatoNET1ID', 'metRecon3DID', 'metMetaNetXID'])\n"
     ]
    }
   ],
   "source": [
    "#\n",
    "# Will check the JSON data in the repo\n",
    "# shuzhao@canyon:~/projects/Azimuth/model_porting$ ls Human-GEM/data/annotation/\n",
    "# humanGEMMetAssoc.JSON  humanGEMRxnAssoc.JSON\n",
    "\n",
    "import json\n",
    "jm = json.load(open('Human-GEM/data/annotation/humanGEMMetAssoc.JSON'))\n",
    "\n",
    "print(jm.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['m00001c', 'm00001s', 'm00002c', 'm00002s', 'm00003c']\n",
      "['m00001', 'm00001', 'm00002', 'm00002', 'm00003']\n",
      "['carveol', 'carveol', 'appnn', 'appnn', '']\n",
      "['C00964', 'C00964', 'C09880', 'C09880', '']\n",
      "['', '', 'HMDB06525', 'HMDB06525', '']\n",
      "['CHEBI:15389', 'CHEBI:15389', 'CHEBI:36740', 'CHEBI:36740', '']\n",
      "['', '', '6654', '6654', '']\n",
      "['', '', '', '', 'LMFA01030283']\n",
      "['', '', '', '', '']\n",
      "['', '', '', '', '']\n",
      "['carveol', 'carveol', 'appnn', 'appnn', 'M00003']\n",
      "['MNXM45735', 'MNXM45735', 'MNXM163755', 'MNXM163755', 'MNXM150165; MNXM27815']\n"
     ]
    }
   ],
   "source": [
    "for k,v in jm.items(): \n",
    "    print( v[:5] )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mets 10138\n",
      "metsNoComp 4168\n",
      "metBiGGID 2347\n",
      "metKEGGID 1647\n",
      "metHMDBID 713\n",
      "metChEBIID 1247\n",
      "metPubChemID 1382\n",
      "metLipidMapsID 487\n",
      "metEHMNID 712\n",
      "metHepatoNET1ID 773\n",
      "metRecon3DID 5089\n",
      "metMetaNetXID 2972\n"
     ]
    }
   ],
   "source": [
    "# How many unique IDs in each?\n",
    "for k,v in jm.items(): \n",
    "    print(k, len(set(v)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4168\n",
      "1411\n"
     ]
    }
   ],
   "source": [
    "# there is 4168 metsNoComp\n",
    "\n",
    "# check HMDB or PubChem\n",
    "HP = {}\n",
    "for ii in range(10138):\n",
    "    HP[jm['metsNoComp'][ii]] = jm['metHMDBID'][ii] or jm['metPubChemID'][ii]\n",
    "    \n",
    "print(len(HP))\n",
    "print(len(set([x for x in HP.values() if x])))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Numbers in ATLAS\n",
    "\n",
    "len(unique) 4168\n",
    "\n",
    "len(formula) 2680\n",
    "\n",
    "len(annotation) = 1411 # HMDB or PubChem"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## E. coli\n",
    "\n",
    "http://bigg.ucsd.edu/models/\n",
    "\n",
    "Downloads last updated Oct 31, 2019 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "ecoli = 'EColi-iJO1366/iJO1366.xml'\n",
    "\n",
    "model3 = cobra.io.read_sbml_model(ecoli)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iJO1366</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07fbf384ab128</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1805</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>2583</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*BIOMASS_Ec_iJO1366_core_53p95M - 1.0*BIOMASS_Ec_iJO1366_core_53p95M_reverse_5c8b1</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular space, periplasm</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iJO1366 at 0x7fbf384ab128>"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Metabolite identifier</strong></td><td>2agpe181_c</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>2-Acyl-sn-glycero-3-phosphoethanolamine (n-C18:1)</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07fbf38565f28</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Formula</strong></td><td>C23H46NO7P1</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartment</strong></td><td>c</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>In 4 reaction(s)</strong></td><td>\n",
       "                    2AGPE181tipp, LPLIPAL2E181, LPLIPAL2ATE181, 2AGPEAT181</td>\n",
       "            </tr>\n",
       "        </table>"
      ],
      "text/plain": [
       "<Metabolite 2agpe181_c at 0x7fbf38565f28>"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model3.metabolites[99]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['2agpe181',\n",
       " '2agpe181_c',\n",
       " 'C23H46NO7P1',\n",
       " {'sbo': 'SBO:0000247',\n",
       "  'bigg.metabolite': '2agpe181',\n",
       "  'metanetx.chemical': 'MNXM3449'}]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def metabolite3compound(M):\n",
    "    # convert cobra Metabolite to metDataModel Compound\n",
    "    Cpd = Compound()\n",
    "    Cpd.src_id = M.id\n",
    "    Cpd.id = M.id.split(\"_\")[0]\n",
    "    Cpd.name = M.name\n",
    "    Cpd.charge = M.charge\n",
    "    Cpd.charged_formula = M.formula\n",
    "    Cpd.db_ids = M.annotation\n",
    "    return Cpd\n",
    "\n",
    "[metabolite3compound(model3.metabolites[99]).id,\n",
    " metabolite3compound(model3.metabolites[99]).src_id,\n",
    " metabolite3compound(model3.metabolites[99]).charged_formula,\n",
    " metabolite3compound(model3.metabolites[99]).db_ids,\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total,  1805 1100\n",
      "real,  1100 911 1100\n"
     ]
    }
   ],
   "source": [
    "# list of Compounds\n",
    "myCpds3 = []\n",
    "anno3 = {}\n",
    "for M in model3.metabolites:\n",
    "    anno3[M.id.split(\"_\")[0]] = M.annotation\n",
    "    myCpds3.append(metabolite3compound(M))\n",
    "    \n",
    "print(\"total, \", len(myCpds3), len(anno3))\n",
    "\n",
    "unique = set([C.id for C in myCpds3])\n",
    "formula = set([C.charged_formula for C in myCpds3 if C.charged_formula])\n",
    "# a little tricker to test on dictionaries\n",
    "annotation = [v for v in anno3.values() if 'metanetx' in str(v)]\n",
    "\n",
    "print(\"real, \", len(unique), len(formula), len(annotation))\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Numbers in E. coli iJO1366\n",
    "\n",
    "len(unique) 1100\n",
    "len(formula) 911\n",
    "len(annotation) = 1100 ?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## WormJam model\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'' is not a valid SBML 'SId'.\n",
      "No objective in listOfObjectives\n",
      "No objective coefficients in model. Unclear what should be optimized\n"
     ]
    }
   ],
   "source": [
    "worm = 'worm/WormJam-GEM-20190101_L3.xml'\n",
    "\n",
    "model4 = cobra.io.read_sbml_model(worm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07fbf317d02b0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>2834</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>3530</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>114</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>Cytosol, Mitochondrion, Nucleus, Extracellular, Mitochondrial Inner Membrane</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model  at 0x7fbf317d02b0>"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Metabolite identifier</strong></td><td>g3m8masn_c</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>(alpha-D-Glucosyl)3-(alpha-D-mannosyl)8-beta-D-...</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07fbf38f20240</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Formula</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartment</strong></td><td>c</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>In 4 reaction(s)</strong></td><td>\n",
       "                    2_4_1_119_RXN_c, RC05979, RC05976, 3_2_1_106_RXN_c</td>\n",
       "            </tr>\n",
       "        </table>"
      ],
      "text/plain": [
       "<Metabolite g3m8masn_c at 0x7fbf38f20240>"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model4.metabolites[9]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2834\n",
      "1271\n"
     ]
    }
   ],
   "source": [
    "metabolites4 = [M.id for M in model4.metabolites]\n",
    "print(len(metabolites4))\n",
    "\n",
    "metabolites4 = set([x.split('_')[0] for x in metabolites4])\n",
    "print(len(metabolites4))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Numbers in WormJam\n",
    "\n",
    "len(unique) 1271 \n",
    "\n",
    "formula and annotation yet unclear\n",
    "\n",
    "needed to pull from elsewhere"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Numbers in Drosophila\n",
    "\n",
    "\n",
    "https://www.nature.com/articles/s41598-019-53532-4\n",
    "\n",
    "Schönborn, J.W., Jehrke, L., Mettler-Altmann, T. et al. FlySilico: Flux balance modeling of Drosophila larval growth and resource allocation. Sci Rep 9, 17156 (2019). https://doi.org/10.1038/s41598-019-53532-4\n",
    "\n",
    "This is a curated model, all metabolites having public IDs.\n",
    "\n",
    "number of unique metabolites = 203\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Title page\n",
      "reactions\n",
      "metabolites\n"
     ]
    }
   ],
   "source": [
    "fly = 'Drosophia/41598_2019_53532_MOESM2_ESM.xls'\n",
    "xlsxData = xlrd.open_workbook(fly)\n",
    "sheets = xlsxData.sheets()\n",
    "for x in sheets: print(x.name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Abbreviation', 'officialname', 'formula(neutral)', 'formula', 'Charge', 'Compartment', 'KEGG ID', 'PubChem ID', 'ChEBI ID', 'InChI string', 'Smiles', 'HMDB', 'Notes']\n",
      "['13dpg[c]', '3-Phospho-D-glyceroyl phosphate', 'C3H8O10P2', 'C3H4O10P2', -4.0, 'c', 'C00236', 3535.0, 16001.0, '1S/C3H8O10P2/c4-2(1-12-14(6,7)8)3(5)13-15(9,10)11/h2,4H,1H2,(H2,6,7,8)(H2,9,10,11)/t2-/m1/s1', 'O[C@H](COP(O)(O)=O)C(=O)OP(O)(O)=O', '', '']\n",
      "['2pg[c]', 'D-Glycerate 2-phosphate', 'C3H7O7P', 'C3H4O7P', -3.0, 'c', 'C00631', 3904.0, 17835.0, '1S/C3H7O7P/c4-1-2(3(5)6)10-11(7,8)9/h2,4H,1H2,(H,5,6)(H2,7,8,9)/t2-/m1/s1', 'OC[C@@H](OP(O)(O)=O)C(O)=O', 'HMDB03391', '']\n",
      "['3pg[c]', '3-Phospho-D-glycerate', 'C3H7O7P', 'C3H4O7P', -3.0, 'c', 'C00197', 3497.0, 17794.0, '1S/C3H7O7P/c4-2(3(5)6)1-10-11(7,8)9/h2,4H,1H2,(H,5,6)(H2,7,8,9)/p-3/t2-/m1/s1', 'C(OP(=O)([O-])[O-])C(O)C(=O)[O-]', '', '']\n",
      "['6pgc[c]', '6-Phospho-D-gluconate', 'C6H13O10P', 'C6H10O10P', -3.0, 'c', 'C00345', 3638.0, 48928.0, '1S/C6H13O10P/c7-2(1-16-17(13,14)15)3(8)4(9)5(10)6(11)12/h2-5,7-10H,1H2,(H,11,12)(H2,13,14,15)/t2-,3-,4+,5-/m1/s1', 'O[C@H](COP(O)(O)=O)[C@@H](O)[C@H](O)[C@@H](O)C(O)=O', 'HMDB0001316', '']\n"
     ]
    }
   ],
   "source": [
    "for ii in range(5):\n",
    "    print(sheets[2].row_values(ii))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "293 203\n"
     ]
    }
   ],
   "source": [
    "fly_m = [sheets[2].row_values(ii)[0] for ii in range(1, sheets[2].nrows)]\n",
    "\n",
    "fly_m_unique = set([x.split(\"[\")[0] for x in fly_m])\n",
    "\n",
    "print( len(fly_m), len(fly_m_unique ))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**I needed the above information to make some quick figures.**\n",
    "\n",
    "Now getting to the export of models to our own format. \n",
    "\n",
    "Use RECON3D as example. \n",
    "Continued on a separate notebook - metabolicModel_RECON3D_20210510.json."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
